{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SRP121892\n",
    "\n",
    "**paper:** [PMID: 29456143](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11098552/) - Neocortical association cell types in the forebrain of birds and alligators, 2018\n",
    "\n",
    "**date, curator:** 2024-10-10, Sara Carsanaro\n",
    "\n",
    "**notes**\n",
    "- emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect\n",
    "- infoOrgan was supplemented by the paper, originally each just said brain and I added the brain regions specified in the paper\n",
    "- paper doesn't specify protocol but does say RNA Selection is polyA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### annotation summary\n",
    "run this after annotation is complete"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_f0870\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_f0870_level0_col0\" class=\"col_heading level0 col0\" >infoOrgan</th>\n",
       "      <th id=\"T_f0870_level0_col1\" class=\"col_heading level0 col1\" >anatId</th>\n",
       "      <th id=\"T_f0870_level0_col2\" class=\"col_heading level0 col2\" >anatName</th>\n",
       "      <th id=\"T_f0870_level0_col3\" class=\"col_heading level0 col3\" >anatAnnotationStatus</th>\n",
       "      <th id=\"T_f0870_level0_col4\" class=\"col_heading level0 col4\" >comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_f0870_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_f0870_row0_col0\" class=\"data row0 col0\" >brain - arcopallium</td>\n",
       "      <td id=\"T_f0870_row0_col1\" class=\"data row0 col1\" >UBERON:0007350</td>\n",
       "      <td id=\"T_f0870_row0_col2\" class=\"data row0 col2\" >arcopallium</td>\n",
       "      <td id=\"T_f0870_row0_col3\" class=\"data row0 col3\" >perfect match</td>\n",
       "      <td id=\"T_f0870_row0_col4\" class=\"data row0 col4\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f0870_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_f0870_row1_col0\" class=\"data row1 col0\" >brain - basorostralis</td>\n",
       "      <td id=\"T_f0870_row1_col1\" class=\"data row1 col1\" >UBERON:0000203</td>\n",
       "      <td id=\"T_f0870_row1_col2\" class=\"data row1 col2\" >pallium</td>\n",
       "      <td id=\"T_f0870_row1_col3\" class=\"data row1 col3\" >missing child term</td>\n",
       "      <td id=\"T_f0870_row1_col4\" class=\"data row1 col4\" >nucleus basorostralis of the pallium (PMID: 15116397), annotated as pallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f0870_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_f0870_row2_col0\" class=\"data row2 col0\" >brain - entopallium</td>\n",
       "      <td id=\"T_f0870_row2_col1\" class=\"data row2 col1\" >UBERON:0014759</td>\n",
       "      <td id=\"T_f0870_row2_col2\" class=\"data row2 col2\" >entopallium</td>\n",
       "      <td id=\"T_f0870_row2_col3\" class=\"data row2 col3\" >perfect match</td>\n",
       "      <td id=\"T_f0870_row2_col4\" class=\"data row2 col4\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f0870_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_f0870_row3_col0\" class=\"data row3 col0\" >brain - Field L</td>\n",
       "      <td id=\"T_f0870_row3_col1\" class=\"data row3 col1\" >UBERON:0007334</td>\n",
       "      <td id=\"T_f0870_row3_col2\" class=\"data row3 col2\" >nidopallium</td>\n",
       "      <td id=\"T_f0870_row3_col3\" class=\"data row3 col3\" >missing child term</td>\n",
       "      <td id=\"T_f0870_row3_col4\" class=\"data row3 col4\" >Field L is the nidopallial region containing the primary auditory thalamorecipient zone (PMID: 15116397), annotated as nidopallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f0870_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_f0870_row4_col0\" class=\"data row4 col0\" >brain - HA (hyperpallium apicale)</td>\n",
       "      <td id=\"T_f0870_row4_col1\" class=\"data row4 col1\" >UBERON:0014757</td>\n",
       "      <td id=\"T_f0870_row4_col2\" class=\"data row4 col2\" >hyperpallium apicale</td>\n",
       "      <td id=\"T_f0870_row4_col3\" class=\"data row4 col3\" >perfect match</td>\n",
       "      <td id=\"T_f0870_row4_col4\" class=\"data row4 col4\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f0870_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_f0870_row5_col0\" class=\"data row5 col0\" >brain - anterior mesopallium</td>\n",
       "      <td id=\"T_f0870_row5_col1\" class=\"data row5 col1\" >UBERON:0007349</td>\n",
       "      <td id=\"T_f0870_row5_col2\" class=\"data row5 col2\" >mesopallium</td>\n",
       "      <td id=\"T_f0870_row5_col3\" class=\"data row5 col3\" >missing child term</td>\n",
       "      <td id=\"T_f0870_row5_col4\" class=\"data row5 col4\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_f0870_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_f0870_row6_col0\" class=\"data row6 col0\" >brain - posterior nidopallium</td>\n",
       "      <td id=\"T_f0870_row6_col1\" class=\"data row6 col1\" >UBERON:0007334</td>\n",
       "      <td id=\"T_f0870_row6_col2\" class=\"data row6 col2\" >nidopallium</td>\n",
       "      <td id=\"T_f0870_row6_col3\" class=\"data row6 col3\" >missing child term</td>\n",
       "      <td id=\"T_f0870_row6_col4\" class=\"data row6 col4\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "anat_summary = library[['infoOrgan', 'anatId', 'anatName', 'anatAnnotationStatus', 'comment']]\n",
    "unique_anat = anat_summary.drop_duplicates()\n",
    "display_df(unique_anat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_86e10\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_86e10_level0_col0\" class=\"col_heading level0 col0\" >infoStage</th>\n",
       "      <th id=\"T_86e10_level0_col1\" class=\"col_heading level0 col1\" >stageId</th>\n",
       "      <th id=\"T_86e10_level0_col2\" class=\"col_heading level0 col2\" >stageName</th>\n",
       "      <th id=\"T_86e10_level0_col3\" class=\"col_heading level0 col3\" >stageAnnotationStatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_86e10_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_86e10_row0_col0\" class=\"data row0 col0\" >Embryonic day 14</td>\n",
       "      <td id=\"T_86e10_row0_col1\" class=\"data row0 col1\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_86e10_row0_col2\" class=\"data row0 col2\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_86e10_row0_col3\" class=\"data row0 col3\" >perfect match</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dev_summary = library[['infoStage', 'stageId', 'stageName', 'stageAnnotationStatus']]\n",
    "unique_dev = dev_summary.drop_duplicates()\n",
    "display_df(unique_dev)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### set variables, import packages, define functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "experiment_id = \"SRP121892\"\n",
    "\n",
    "path_to_create_exp_script = \"/Users/scarsana/Desktop/git/scRNA-Seq/scripts/Create_ExpLib_tables.py\" \n",
    "experiment_type = \"bulk\"\n",
    "\n",
    "path_to_output_main = \"/Users/scarsana/Desktop/git/expression-annotations/Notebooks/bulk/\" \n",
    "path_to_output = \"{}{}/\".format(path_to_output_main, experiment_id)\n",
    "library_path_from_script = \"{}RNASeqLibrary_{}.tsv\".format(path_to_output, experiment_id)\n",
    "experiment_path_from_script = \"{}RNASeqExperiment_{}.tsv\".format(path_to_output, experiment_id)\n",
    "library_to_add_path = \"{}complete_RNASeqLibrary_{}.tsv\".format(path_to_output, experiment_id)\n",
    "experiment_to_add_path = \"{}complete_RNASeqExperiment_{}.tsv\".format(path_to_output, experiment_id)\n",
    "script_file = \"{}.ipynb\".format(experiment_id)\n",
    "commit_message_exp = '\"adding annotated bulk experiment {}\"'.format(experiment_id)\n",
    "commit_message_py = '\"adding annotation files for {} to notebook folder\"'.format(experiment_id)\n",
    "\n",
    "\n",
    "## to add to git\n",
    "path_to_git_annotations = \"/Users/scarsana/Desktop/git/expression-annotations/RNA_Seq/\"\n",
    "git_library_path = \"{}RNASeqLibrary.tsv\".format(path_to_git_annotations)\n",
    "git_experiment_path = \"{}RNASeqExperiment.tsv\".format(path_to_git_annotations)\n",
    "\n",
    "library_cols = ['#libraryId', 'experimentId', 'platform', 'SRSId', 'anatId', 'anatName', 'stageId', 'stageName', 'url_GSM', 'infoOrgan', 'infoStage', 'anatAnnotationStatus', 'anatBiologicalStatus', 'stageAnnotationStatus', 'sex', 'strain', 'genotype', 'speciesId', 'protocol', 'protocolType', 'RNASelection', 'globin_reduction', 'replicate', 'lib_name', 'sampleName', 'sampleAge_value', 'sampleAge_unit', 'PATOid', 'PATOname','comment', 'condition', 'physiologicalStatus', 'annotatorId', 'lastModificationDate']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.simplefilter(action='ignore', category=FutureWarning)\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from IPython.display import display, HTML\n",
    "import os\n",
    "import csv\n",
    "\n",
    "# displays df with the scrollbar next to the DataFrame\n",
    "def display_df(df):\n",
    "    pd.set_option(\"display.max_rows\", None)\n",
    "    pd.set_option(\"display.max_columns\", None)\n",
    "    display(HTML(\"<div style='height: 300px; overflow: auto; width: fit-content'>\" +\n",
    "        df.style.to_html(index=False) + \"</div>\"))\n",
    "\n",
    "# function that compares two columns in a dataframe and tells you which ones are not equal (case insensitive)\n",
    "def compare_columns(df, col1, col2, return_col):\n",
    "    compare_return = df[col1].str.lower() != df[col2].str.lower()  \n",
    "    df.loc[compare_return, return_col] \n",
    "    if not any(compare_return):\n",
    "        print(\"The two columns are equal (case insensitive)\")\n",
    "    else:\n",
    "        print(\"The following rows are not equal: \")\n",
    "        print(df.loc[compare_return, return_col])\n",
    "\n",
    "# fixes formatting of file to match libreoffice settings/historic file format\n",
    "def update_format(path):\n",
    "    with open(path, 'r') as file:\n",
    "        filedata = file.read()\n",
    "    # Replace the target string\n",
    "    filedata = filedata.replace(\"\\t\\\"\\\"\", \"\\t\")\n",
    "    # Write the file out again\n",
    "    with open(path, 'w') as file:\n",
    "        file.write(filedata)\n",
    "\n",
    "# checks for duplicate values in a specific column and prints those values + the corresponding library id\n",
    "def dup_check(df, column):\n",
    "    duplicateCheck = df.duplicated(subset=[column], keep=False)\n",
    "    duplicateCheck.sort_values(inplace=True)\n",
    "    if duplicateCheck.unique().any() == False:\n",
    "        print(\"no duplicate values in \" + column)\n",
    "    elif duplicateCheck.unique().any() == True and column != '#libraryId':\n",
    "        dups = df[duplicateCheck].loc[:,['#libraryId', column]]\n",
    "        df_dups = pd.DataFrame(dups)\n",
    "        df_dups.sort_values(inplace=True, by=column)\n",
    "        print(df_dups)\n",
    "    elif duplicateCheck.unique().any() == True and column == '#libraryId':\n",
    "        print(df[duplicateCheck].loc[:,['#libraryId']])\n",
    "\n",
    "# prints all unique values in a specific column\n",
    "def unique_sorted(df, column):\n",
    "    unique = df[column].unique()\n",
    "    unique.sort()\n",
    "    print(unique)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/scarsana/Desktop/git/scRNA-Seq/scripts/Create_ExpLib_tables.py:97: SyntaxWarning: invalid escape sequence '\\('\n",
      "  all_protoc = [w.replace('(', '\\(') for w in all_protoc]\n",
      "/Users/scarsana/Desktop/git/scRNA-Seq/scripts/Create_ExpLib_tables.py:98: SyntaxWarning: invalid escape sequence '\\)'\n",
      "  all_protoc = [w.replace(')', '\\)') for w in all_protoc] \n",
      "Be patient, it may take a few minutes.\n",
      "curl: (22) The requested URL returned error: 500\n",
      "\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m curl command failed ( Thu Oct 10 15:55:51 CEST 2024 ) with: 22\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "\u001b[34mhttps://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi -d id=29456143&dbfrom=pubmed&cmd=prlinks&tool=edirect&edirect=16.2&edirect_os=Darwin&email=scarsana%40SORGE42778\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "HTTP/1.1 500 Internal Server Error\n",
      "\u001b[m\u001b[31m\u001b[1m\u001b[7m WARNING: \u001b[m\u001b[31m\u001b[1m FAILURE ( Thu Oct 10 15:55:50 CEST 2024 )\u001b[m\n",
      "\u001b[34mnquire -url https://eutils.ncbi.nlm.nih.gov/entrez/eutils/ elink.fcgi -id 29456143 -dbfrom pubmed -cmd prlinks -tool edirect -edirect 16.2 -edirect_os Darwin -email scarsana@SORGE42778\n",
      "\u001b[m\u001b[31mEMPTY RESULT\u001b[m\n",
      "\u001b[34mSECOND ATTEMPT\u001b[m\n",
      "curl: (22) The requested URL returned error: 500\n",
      "\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m curl command failed ( Thu Oct 10 15:56:14 CEST 2024 ) with: 22\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "\u001b[34mhttps://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi -d id=29456143&dbfrom=pubmed&cmd=prlinks&tool=edirect&edirect=16.2&edirect_os=Darwin&email=scarsana%40SORGE42778\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "HTTP/1.1 500 Internal Server Error\n",
      "\u001b[m\u001b[31m\u001b[1m\u001b[7m WARNING: \u001b[m\u001b[31m\u001b[1m FAILURE ( Thu Oct 10 15:56:13 CEST 2024 )\u001b[m\n",
      "\u001b[34mnquire -url https://eutils.ncbi.nlm.nih.gov/entrez/eutils/ elink.fcgi -id 29456143 -dbfrom pubmed -cmd prlinks -tool edirect -edirect 16.2 -edirect_os Darwin -email scarsana@SORGE42778\n",
      "\u001b[m\u001b[31mEMPTY RESULT\u001b[m\n",
      "\u001b[34mLAST ATTEMPT\u001b[m\n",
      "curl: (22) The requested URL returned error: 500\n",
      "\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m curl command failed ( Thu Oct 10 15:56:35 CEST 2024 ) with: 22\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "\u001b[34mhttps://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi -d id=29456143&dbfrom=pubmed&cmd=prlinks&tool=edirect&edirect=16.2&edirect_os=Darwin&email=scarsana%40SORGE42778\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "HTTP/1.1 500 Internal Server Error\n",
      "\u001b[m\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[31m\u001b[1m FAILURE ( Thu Oct 10 15:56:35 CEST 2024 )\u001b[m\n",
      "\u001b[34mnquire -url https://eutils.ncbi.nlm.nih.gov/entrez/eutils/ elink.fcgi -id 29456143 -dbfrom pubmed -cmd prlinks -tool edirect -edirect 16.2 -edirect_os Darwin -email scarsana@SORGE42778\n",
      "\u001b[m\u001b[31mEMPTY RESULT\u001b[m\n",
      "\u001b[34mQUERY FAILURE\u001b[m\n",
      "curl: (22) The requested URL returned error: 500\n",
      "\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m curl command failed ( Thu Oct 10 15:56:57 CEST 2024 ) with: 22\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "\u001b[34mhttps://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi -d id=29456143&dbfrom=pubmed&cmd=prlinks&tool=edirect&edirect=16.2&edirect_os=Darwin&email=scarsana%40SORGE42778\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "HTTP/1.1 500 Internal Server Error\n",
      "100%|█████████████████████████████████████████████| 7/7 [00:10<00:00,  1.57s/it]\n",
      "curl: (22) The requested URL returned error: 500\n",
      "\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m curl command failed ( Thu Oct 10 15:57:43 CEST 2024 ) with: 22\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "\u001b[34mhttps://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi -d id=4663954%2C4663953%2C4663952%2C4663951%2C4663950%2C4663949%2C4663948&WebEnv=MCID_6707dd555b414ea75f0f3720&dbfrom=sra&db=biosample&cmd=neighbor_history&linkname=sra_biosample&tool=edirect&edirect=16.2&edirect_os=Darwin&email=scarsana%40SORGE42778\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "HTTP/1.1 500 Internal Server Error\n",
      "\u001b[m\u001b[31m\u001b[1m\u001b[7m WARNING: \u001b[m\u001b[31m\u001b[1m FAILURE ( Thu Oct 10 15:57:42 CEST 2024 )\u001b[m\n",
      "\u001b[34mnquire -url https://eutils.ncbi.nlm.nih.gov/entrez/eutils/ elink.fcgi -id 4663954,4663953,4663952,4663951,4663950,4663949,4663948 -WebEnv MCID_6707dd555b414ea75f0f3720 -dbfrom sra -db biosample -cmd neighbor_history -linkname sra_biosample -tool edirect -edirect 16.2 -edirect_os Darwin -email scarsana@SORGE42778\n",
      "\u001b[m\u001b[31mEMPTY RESULT\u001b[m\n",
      "\u001b[34mSECOND ATTEMPT\u001b[m\n",
      "curl: (22) The requested URL returned error: 500\n",
      "\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m curl command failed ( Thu Oct 10 15:58:05 CEST 2024 ) with: 22\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "\u001b[34mhttps://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi -d id=4663954%2C4663953%2C4663952%2C4663951%2C4663950%2C4663949%2C4663948&WebEnv=MCID_6707dd555b414ea75f0f3720&dbfrom=sra&db=biosample&cmd=neighbor_history&linkname=sra_biosample&tool=edirect&edirect=16.2&edirect_os=Darwin&email=scarsana%40SORGE42778\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "HTTP/1.1 500 Internal Server Error\n",
      "\u001b[m\u001b[31m\u001b[1m\u001b[7m WARNING: \u001b[m\u001b[31m\u001b[1m FAILURE ( Thu Oct 10 15:58:04 CEST 2024 )\u001b[m\n",
      "\u001b[34mnquire -url https://eutils.ncbi.nlm.nih.gov/entrez/eutils/ elink.fcgi -id 4663954,4663953,4663952,4663951,4663950,4663949,4663948 -WebEnv MCID_6707dd555b414ea75f0f3720 -dbfrom sra -db biosample -cmd neighbor_history -linkname sra_biosample -tool edirect -edirect 16.2 -edirect_os Darwin -email scarsana@SORGE42778\n",
      "\u001b[m\u001b[31mEMPTY RESULT\u001b[m\n",
      "\u001b[34mLAST ATTEMPT\u001b[m\n",
      "curl: (22) The requested URL returned error: 500\n",
      "\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m curl command failed ( Thu Oct 10 15:58:27 CEST 2024 ) with: 22\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "\u001b[34mhttps://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi -d id=4663954%2C4663953%2C4663952%2C4663951%2C4663950%2C4663949%2C4663948&WebEnv=MCID_6707dd555b414ea75f0f3720&dbfrom=sra&db=biosample&cmd=neighbor_history&linkname=sra_biosample&tool=edirect&edirect=16.2&edirect_os=Darwin&email=scarsana%40SORGE42778\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "HTTP/1.1 500 Internal Server Error\n",
      "\u001b[m\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[31m\u001b[1m FAILURE ( Thu Oct 10 15:58:26 CEST 2024 )\u001b[m\n",
      "\u001b[34mnquire -url https://eutils.ncbi.nlm.nih.gov/entrez/eutils/ elink.fcgi -id 4663954,4663953,4663952,4663951,4663950,4663949,4663948 -WebEnv MCID_6707dd555b414ea75f0f3720 -dbfrom sra -db biosample -cmd neighbor_history -linkname sra_biosample -tool edirect -edirect 16.2 -edirect_os Darwin -email scarsana@SORGE42778\n",
      "\u001b[m\u001b[31mEMPTY RESULT\u001b[m\n",
      "\u001b[34mQUERY FAILURE\u001b[m\n",
      "curl: (22) The requested URL returned error: 500\n",
      "\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\u001b[31m\u001b[1m curl command failed ( Thu Oct 10 15:58:48 CEST 2024 ) with: 22\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "\u001b[34mhttps://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi -d id=4663954%2C4663953%2C4663952%2C4663951%2C4663950%2C4663949%2C4663948&WebEnv=MCID_6707dd555b414ea75f0f3720&dbfrom=sra&db=biosample&cmd=neighbor_history&linkname=sra_biosample&tool=edirect&edirect=16.2&edirect_os=Darwin&email=scarsana%40SORGE42778\u001b[m\u001b[?7h\u001b[4l\u001b>\u001b7\u001b[r\u001b[?1;3;4;6l\u001b8\n",
      "HTTP/1.1 500 Internal Server Error\n",
      "\u001b[m\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[31m\u001b[1m ELink failure\u001b[m\n",
      "\u001b[m\u001b[31m\u001b[1m\u001b[7m ERROR: \u001b[m\u001b[31m\u001b[1m Missing -db argument\u001b[m\n",
      "7 samples dont have attributes, try to find them somewhere else\n",
      "100%|█████████████████████████████████████████████| 7/7 [00:23<00:00,  3.30s/it]\n",
      "0 samples dont have attributes\n"
     ]
    }
   ],
   "source": [
    "! python3 $path_to_create_exp_script $experiment_id $path_to_output $experiment_type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### library annnotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_96fe1\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_96fe1_level0_col0\" class=\"col_heading level0 col0\" >#libraryId</th>\n",
       "      <th id=\"T_96fe1_level0_col1\" class=\"col_heading level0 col1\" >experimentId</th>\n",
       "      <th id=\"T_96fe1_level0_col2\" class=\"col_heading level0 col2\" >platform</th>\n",
       "      <th id=\"T_96fe1_level0_col3\" class=\"col_heading level0 col3\" >SRSId</th>\n",
       "      <th id=\"T_96fe1_level0_col4\" class=\"col_heading level0 col4\" >anatId</th>\n",
       "      <th id=\"T_96fe1_level0_col5\" class=\"col_heading level0 col5\" >anatName</th>\n",
       "      <th id=\"T_96fe1_level0_col6\" class=\"col_heading level0 col6\" >stageId</th>\n",
       "      <th id=\"T_96fe1_level0_col7\" class=\"col_heading level0 col7\" >stageName</th>\n",
       "      <th id=\"T_96fe1_level0_col8\" class=\"col_heading level0 col8\" >url_GSM</th>\n",
       "      <th id=\"T_96fe1_level0_col9\" class=\"col_heading level0 col9\" >infoOrgan</th>\n",
       "      <th id=\"T_96fe1_level0_col10\" class=\"col_heading level0 col10\" >infoStage</th>\n",
       "      <th id=\"T_96fe1_level0_col11\" class=\"col_heading level0 col11\" >anatAnnotationStatus</th>\n",
       "      <th id=\"T_96fe1_level0_col12\" class=\"col_heading level0 col12\" >anatBiologicalStatus</th>\n",
       "      <th id=\"T_96fe1_level0_col13\" class=\"col_heading level0 col13\" >stageAnnotationStatus</th>\n",
       "      <th id=\"T_96fe1_level0_col14\" class=\"col_heading level0 col14\" >sex</th>\n",
       "      <th id=\"T_96fe1_level0_col15\" class=\"col_heading level0 col15\" >strain</th>\n",
       "      <th id=\"T_96fe1_level0_col16\" class=\"col_heading level0 col16\" >genotype</th>\n",
       "      <th id=\"T_96fe1_level0_col17\" class=\"col_heading level0 col17\" >speciesId</th>\n",
       "      <th id=\"T_96fe1_level0_col18\" class=\"col_heading level0 col18\" >protocol</th>\n",
       "      <th id=\"T_96fe1_level0_col19\" class=\"col_heading level0 col19\" >protocolType</th>\n",
       "      <th id=\"T_96fe1_level0_col20\" class=\"col_heading level0 col20\" >RNASelection</th>\n",
       "      <th id=\"T_96fe1_level0_col21\" class=\"col_heading level0 col21\" >globin_reduction</th>\n",
       "      <th id=\"T_96fe1_level0_col22\" class=\"col_heading level0 col22\" >replicate</th>\n",
       "      <th id=\"T_96fe1_level0_col23\" class=\"col_heading level0 col23\" >lib_name</th>\n",
       "      <th id=\"T_96fe1_level0_col24\" class=\"col_heading level0 col24\" >sampleName</th>\n",
       "      <th id=\"T_96fe1_level0_col25\" class=\"col_heading level0 col25\" >sampleAge_value</th>\n",
       "      <th id=\"T_96fe1_level0_col26\" class=\"col_heading level0 col26\" >sampleAge_unit</th>\n",
       "      <th id=\"T_96fe1_level0_col27\" class=\"col_heading level0 col27\" >PATOid</th>\n",
       "      <th id=\"T_96fe1_level0_col28\" class=\"col_heading level0 col28\" >PATOname</th>\n",
       "      <th id=\"T_96fe1_level0_col29\" class=\"col_heading level0 col29\" >comment</th>\n",
       "      <th id=\"T_96fe1_level0_col30\" class=\"col_heading level0 col30\" >condition</th>\n",
       "      <th id=\"T_96fe1_level0_col31\" class=\"col_heading level0 col31\" >physiologicalStatus</th>\n",
       "      <th id=\"T_96fe1_level0_col32\" class=\"col_heading level0 col32\" >annotatorId</th>\n",
       "      <th id=\"T_96fe1_level0_col33\" class=\"col_heading level0 col33\" >lastModificationDate</th>\n",
       "      <th id=\"T_96fe1_level0_col34\" class=\"col_heading level0 col34\" >library_contruction_protocol</th>\n",
       "      <th id=\"T_96fe1_level0_col35\" class=\"col_heading level0 col35\" >source_qc</th>\n",
       "      <th id=\"T_96fe1_level0_col36\" class=\"col_heading level0 col36\" >lib_name_2</th>\n",
       "      <th id=\"T_96fe1_level0_col37\" class=\"col_heading level0 col37\" >lib_name_3</th>\n",
       "      <th id=\"T_96fe1_level0_col38\" class=\"col_heading level0 col38\" >source_name</th>\n",
       "      <th id=\"T_96fe1_level0_col39\" class=\"col_heading level0 col39\" >individual</th>\n",
       "      <th id=\"T_96fe1_level0_col40\" class=\"col_heading level0 col40\" >infoStage_2</th>\n",
       "      <th id=\"T_96fe1_level0_col41\" class=\"col_heading level0 col41\" >infoStage_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_96fe1_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_96fe1_row0_col0\" class=\"data row0 col0\" >SRX3334825</td>\n",
       "      <td id=\"T_96fe1_row0_col1\" class=\"data row0 col1\" >SRP121892</td>\n",
       "      <td id=\"T_96fe1_row0_col2\" class=\"data row0 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_96fe1_row0_col3\" class=\"data row0 col3\" >SRS2637233</td>\n",
       "      <td id=\"T_96fe1_row0_col4\" class=\"data row0 col4\" >UBERON:0007350</td>\n",
       "      <td id=\"T_96fe1_row0_col5\" class=\"data row0 col5\" >arcopallium</td>\n",
       "      <td id=\"T_96fe1_row0_col6\" class=\"data row0 col6\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col7\" class=\"data row0 col7\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col9\" class=\"data row0 col9\" >brain - arcopallium</td>\n",
       "      <td id=\"T_96fe1_row0_col10\" class=\"data row0 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_96fe1_row0_col11\" class=\"data row0 col11\" >perfect match</td>\n",
       "      <td id=\"T_96fe1_row0_col12\" class=\"data row0 col12\" >not documented</td>\n",
       "      <td id=\"T_96fe1_row0_col13\" class=\"data row0 col13\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col14\" class=\"data row0 col14\" >not collected</td>\n",
       "      <td id=\"T_96fe1_row0_col15\" class=\"data row0 col15\" >White leghorn</td>\n",
       "      <td id=\"T_96fe1_row0_col16\" class=\"data row0 col16\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col17\" class=\"data row0 col17\" >9031</td>\n",
       "      <td id=\"T_96fe1_row0_col18\" class=\"data row0 col18\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col19\" class=\"data row0 col19\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col20\" class=\"data row0 col20\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col21\" class=\"data row0 col21\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col22\" class=\"data row0 col22\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col23\" class=\"data row0 col23\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_96fe1_row0_col24\" class=\"data row0 col24\" >SAMN07839988</td>\n",
       "      <td id=\"T_96fe1_row0_col25\" class=\"data row0 col25\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col26\" class=\"data row0 col26\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col27\" class=\"data row0 col27\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col28\" class=\"data row0 col28\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col29\" class=\"data row0 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_96fe1_row0_col30\" class=\"data row0 col30\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col31\" class=\"data row0 col31\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col32\" class=\"data row0 col32\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col33\" class=\"data row0 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_96fe1_row0_col34\" class=\"data row0 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_96fe1_row0_col35\" class=\"data row0 col35\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col36\" class=\"data row0 col36\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_96fe1_row0_col37\" class=\"data row0 col37\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col38\" class=\"data row0 col38\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col39\" class=\"data row0 col39\" ></td>\n",
       "      <td id=\"T_96fe1_row0_col40\" class=\"data row0 col40\" >E14</td>\n",
       "      <td id=\"T_96fe1_row0_col41\" class=\"data row0 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96fe1_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_96fe1_row1_col0\" class=\"data row1 col0\" >SRX3334824</td>\n",
       "      <td id=\"T_96fe1_row1_col1\" class=\"data row1 col1\" >SRP121892</td>\n",
       "      <td id=\"T_96fe1_row1_col2\" class=\"data row1 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_96fe1_row1_col3\" class=\"data row1 col3\" >SRS2637232</td>\n",
       "      <td id=\"T_96fe1_row1_col4\" class=\"data row1 col4\" >UBERON:0000203</td>\n",
       "      <td id=\"T_96fe1_row1_col5\" class=\"data row1 col5\" >pallium</td>\n",
       "      <td id=\"T_96fe1_row1_col6\" class=\"data row1 col6\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col7\" class=\"data row1 col7\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col8\" class=\"data row1 col8\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col9\" class=\"data row1 col9\" >brain - basorostralis</td>\n",
       "      <td id=\"T_96fe1_row1_col10\" class=\"data row1 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_96fe1_row1_col11\" class=\"data row1 col11\" >missing child term</td>\n",
       "      <td id=\"T_96fe1_row1_col12\" class=\"data row1 col12\" >not documented</td>\n",
       "      <td id=\"T_96fe1_row1_col13\" class=\"data row1 col13\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col14\" class=\"data row1 col14\" >not collected</td>\n",
       "      <td id=\"T_96fe1_row1_col15\" class=\"data row1 col15\" >White leghorn</td>\n",
       "      <td id=\"T_96fe1_row1_col16\" class=\"data row1 col16\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col17\" class=\"data row1 col17\" >9031</td>\n",
       "      <td id=\"T_96fe1_row1_col18\" class=\"data row1 col18\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col19\" class=\"data row1 col19\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col20\" class=\"data row1 col20\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col21\" class=\"data row1 col21\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col22\" class=\"data row1 col22\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col23\" class=\"data row1 col23\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_96fe1_row1_col24\" class=\"data row1 col24\" >SAMN07839989</td>\n",
       "      <td id=\"T_96fe1_row1_col25\" class=\"data row1 col25\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col26\" class=\"data row1 col26\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col27\" class=\"data row1 col27\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col28\" class=\"data row1 col28\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col29\" class=\"data row1 col29\" >nucleus basorostralis of the pallium (PMID: 15116397), annotated as pallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_96fe1_row1_col30\" class=\"data row1 col30\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col31\" class=\"data row1 col31\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col32\" class=\"data row1 col32\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col33\" class=\"data row1 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_96fe1_row1_col34\" class=\"data row1 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_96fe1_row1_col35\" class=\"data row1 col35\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col36\" class=\"data row1 col36\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_96fe1_row1_col37\" class=\"data row1 col37\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col38\" class=\"data row1 col38\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col39\" class=\"data row1 col39\" ></td>\n",
       "      <td id=\"T_96fe1_row1_col40\" class=\"data row1 col40\" >E15</td>\n",
       "      <td id=\"T_96fe1_row1_col41\" class=\"data row1 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96fe1_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_96fe1_row2_col0\" class=\"data row2 col0\" >SRX3334823</td>\n",
       "      <td id=\"T_96fe1_row2_col1\" class=\"data row2 col1\" >SRP121892</td>\n",
       "      <td id=\"T_96fe1_row2_col2\" class=\"data row2 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_96fe1_row2_col3\" class=\"data row2 col3\" >SRS2637229</td>\n",
       "      <td id=\"T_96fe1_row2_col4\" class=\"data row2 col4\" >UBERON:0014759</td>\n",
       "      <td id=\"T_96fe1_row2_col5\" class=\"data row2 col5\" >entopallium</td>\n",
       "      <td id=\"T_96fe1_row2_col6\" class=\"data row2 col6\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col7\" class=\"data row2 col7\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col8\" class=\"data row2 col8\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col9\" class=\"data row2 col9\" >brain - entopallium</td>\n",
       "      <td id=\"T_96fe1_row2_col10\" class=\"data row2 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_96fe1_row2_col11\" class=\"data row2 col11\" >perfect match</td>\n",
       "      <td id=\"T_96fe1_row2_col12\" class=\"data row2 col12\" >not documented</td>\n",
       "      <td id=\"T_96fe1_row2_col13\" class=\"data row2 col13\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col14\" class=\"data row2 col14\" >not collected</td>\n",
       "      <td id=\"T_96fe1_row2_col15\" class=\"data row2 col15\" >White leghorn</td>\n",
       "      <td id=\"T_96fe1_row2_col16\" class=\"data row2 col16\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col17\" class=\"data row2 col17\" >9031</td>\n",
       "      <td id=\"T_96fe1_row2_col18\" class=\"data row2 col18\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col19\" class=\"data row2 col19\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col20\" class=\"data row2 col20\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col21\" class=\"data row2 col21\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col22\" class=\"data row2 col22\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col23\" class=\"data row2 col23\" >Gga_entopallium</td>\n",
       "      <td id=\"T_96fe1_row2_col24\" class=\"data row2 col24\" >SAMN07839990</td>\n",
       "      <td id=\"T_96fe1_row2_col25\" class=\"data row2 col25\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col26\" class=\"data row2 col26\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col27\" class=\"data row2 col27\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col28\" class=\"data row2 col28\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col29\" class=\"data row2 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_96fe1_row2_col30\" class=\"data row2 col30\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col31\" class=\"data row2 col31\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col32\" class=\"data row2 col32\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col33\" class=\"data row2 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_96fe1_row2_col34\" class=\"data row2 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_96fe1_row2_col35\" class=\"data row2 col35\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col36\" class=\"data row2 col36\" >Gga_entopallium</td>\n",
       "      <td id=\"T_96fe1_row2_col37\" class=\"data row2 col37\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col38\" class=\"data row2 col38\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col39\" class=\"data row2 col39\" ></td>\n",
       "      <td id=\"T_96fe1_row2_col40\" class=\"data row2 col40\" >E16</td>\n",
       "      <td id=\"T_96fe1_row2_col41\" class=\"data row2 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96fe1_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_96fe1_row3_col0\" class=\"data row3 col0\" >SRX3334822</td>\n",
       "      <td id=\"T_96fe1_row3_col1\" class=\"data row3 col1\" >SRP121892</td>\n",
       "      <td id=\"T_96fe1_row3_col2\" class=\"data row3 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_96fe1_row3_col3\" class=\"data row3 col3\" >SRS2637228</td>\n",
       "      <td id=\"T_96fe1_row3_col4\" class=\"data row3 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_96fe1_row3_col5\" class=\"data row3 col5\" >nidopallium</td>\n",
       "      <td id=\"T_96fe1_row3_col6\" class=\"data row3 col6\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col7\" class=\"data row3 col7\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col8\" class=\"data row3 col8\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col9\" class=\"data row3 col9\" >brain - Field L</td>\n",
       "      <td id=\"T_96fe1_row3_col10\" class=\"data row3 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_96fe1_row3_col11\" class=\"data row3 col11\" >missing child term</td>\n",
       "      <td id=\"T_96fe1_row3_col12\" class=\"data row3 col12\" >not documented</td>\n",
       "      <td id=\"T_96fe1_row3_col13\" class=\"data row3 col13\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col14\" class=\"data row3 col14\" >not collected</td>\n",
       "      <td id=\"T_96fe1_row3_col15\" class=\"data row3 col15\" >White leghorn</td>\n",
       "      <td id=\"T_96fe1_row3_col16\" class=\"data row3 col16\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col17\" class=\"data row3 col17\" >9031</td>\n",
       "      <td id=\"T_96fe1_row3_col18\" class=\"data row3 col18\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col19\" class=\"data row3 col19\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col20\" class=\"data row3 col20\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col21\" class=\"data row3 col21\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col22\" class=\"data row3 col22\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col23\" class=\"data row3 col23\" >Gga_FieldL</td>\n",
       "      <td id=\"T_96fe1_row3_col24\" class=\"data row3 col24\" >SAMN07839991</td>\n",
       "      <td id=\"T_96fe1_row3_col25\" class=\"data row3 col25\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col26\" class=\"data row3 col26\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col27\" class=\"data row3 col27\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col28\" class=\"data row3 col28\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col29\" class=\"data row3 col29\" >Field L is the nidopallial region containing the primary auditory thalamorecipient zone (PMID: 15116397), annotated as nidopallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_96fe1_row3_col30\" class=\"data row3 col30\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col31\" class=\"data row3 col31\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col32\" class=\"data row3 col32\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col33\" class=\"data row3 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_96fe1_row3_col34\" class=\"data row3 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_96fe1_row3_col35\" class=\"data row3 col35\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col36\" class=\"data row3 col36\" >Gga_FieldL</td>\n",
       "      <td id=\"T_96fe1_row3_col37\" class=\"data row3 col37\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col38\" class=\"data row3 col38\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col39\" class=\"data row3 col39\" ></td>\n",
       "      <td id=\"T_96fe1_row3_col40\" class=\"data row3 col40\" >E17</td>\n",
       "      <td id=\"T_96fe1_row3_col41\" class=\"data row3 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96fe1_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_96fe1_row4_col0\" class=\"data row4 col0\" >SRX3334821</td>\n",
       "      <td id=\"T_96fe1_row4_col1\" class=\"data row4 col1\" >SRP121892</td>\n",
       "      <td id=\"T_96fe1_row4_col2\" class=\"data row4 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_96fe1_row4_col3\" class=\"data row4 col3\" >SRS2637227</td>\n",
       "      <td id=\"T_96fe1_row4_col4\" class=\"data row4 col4\" >UBERON:0014757</td>\n",
       "      <td id=\"T_96fe1_row4_col5\" class=\"data row4 col5\" >hyperpallium apicale</td>\n",
       "      <td id=\"T_96fe1_row4_col6\" class=\"data row4 col6\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col7\" class=\"data row4 col7\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col8\" class=\"data row4 col8\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col9\" class=\"data row4 col9\" >brain - HA (hyperpallium apicale)</td>\n",
       "      <td id=\"T_96fe1_row4_col10\" class=\"data row4 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_96fe1_row4_col11\" class=\"data row4 col11\" >perfect match</td>\n",
       "      <td id=\"T_96fe1_row4_col12\" class=\"data row4 col12\" >not documented</td>\n",
       "      <td id=\"T_96fe1_row4_col13\" class=\"data row4 col13\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col14\" class=\"data row4 col14\" >not collected</td>\n",
       "      <td id=\"T_96fe1_row4_col15\" class=\"data row4 col15\" >White leghorn</td>\n",
       "      <td id=\"T_96fe1_row4_col16\" class=\"data row4 col16\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col17\" class=\"data row4 col17\" >9031</td>\n",
       "      <td id=\"T_96fe1_row4_col18\" class=\"data row4 col18\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col19\" class=\"data row4 col19\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col20\" class=\"data row4 col20\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col21\" class=\"data row4 col21\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col22\" class=\"data row4 col22\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col23\" class=\"data row4 col23\" >Gga_HA</td>\n",
       "      <td id=\"T_96fe1_row4_col24\" class=\"data row4 col24\" >SAMN07839992</td>\n",
       "      <td id=\"T_96fe1_row4_col25\" class=\"data row4 col25\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col26\" class=\"data row4 col26\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col27\" class=\"data row4 col27\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col28\" class=\"data row4 col28\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col29\" class=\"data row4 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_96fe1_row4_col30\" class=\"data row4 col30\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col31\" class=\"data row4 col31\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col32\" class=\"data row4 col32\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col33\" class=\"data row4 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_96fe1_row4_col34\" class=\"data row4 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_96fe1_row4_col35\" class=\"data row4 col35\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col36\" class=\"data row4 col36\" >Gga_HA</td>\n",
       "      <td id=\"T_96fe1_row4_col37\" class=\"data row4 col37\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col38\" class=\"data row4 col38\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col39\" class=\"data row4 col39\" ></td>\n",
       "      <td id=\"T_96fe1_row4_col40\" class=\"data row4 col40\" >E18</td>\n",
       "      <td id=\"T_96fe1_row4_col41\" class=\"data row4 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96fe1_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_96fe1_row5_col0\" class=\"data row5 col0\" >SRX3334820</td>\n",
       "      <td id=\"T_96fe1_row5_col1\" class=\"data row5 col1\" >SRP121892</td>\n",
       "      <td id=\"T_96fe1_row5_col2\" class=\"data row5 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_96fe1_row5_col3\" class=\"data row5 col3\" >SRS2637226</td>\n",
       "      <td id=\"T_96fe1_row5_col4\" class=\"data row5 col4\" >UBERON:0007349</td>\n",
       "      <td id=\"T_96fe1_row5_col5\" class=\"data row5 col5\" >mesopallium</td>\n",
       "      <td id=\"T_96fe1_row5_col6\" class=\"data row5 col6\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col7\" class=\"data row5 col7\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col8\" class=\"data row5 col8\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col9\" class=\"data row5 col9\" >brain - anterior mesopallium</td>\n",
       "      <td id=\"T_96fe1_row5_col10\" class=\"data row5 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_96fe1_row5_col11\" class=\"data row5 col11\" >missing child term</td>\n",
       "      <td id=\"T_96fe1_row5_col12\" class=\"data row5 col12\" >not documented</td>\n",
       "      <td id=\"T_96fe1_row5_col13\" class=\"data row5 col13\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col14\" class=\"data row5 col14\" >not collected</td>\n",
       "      <td id=\"T_96fe1_row5_col15\" class=\"data row5 col15\" >White leghorn</td>\n",
       "      <td id=\"T_96fe1_row5_col16\" class=\"data row5 col16\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col17\" class=\"data row5 col17\" >9031</td>\n",
       "      <td id=\"T_96fe1_row5_col18\" class=\"data row5 col18\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col19\" class=\"data row5 col19\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col20\" class=\"data row5 col20\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col21\" class=\"data row5 col21\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col22\" class=\"data row5 col22\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col23\" class=\"data row5 col23\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_96fe1_row5_col24\" class=\"data row5 col24\" >SAMN07839993</td>\n",
       "      <td id=\"T_96fe1_row5_col25\" class=\"data row5 col25\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col26\" class=\"data row5 col26\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col27\" class=\"data row5 col27\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col28\" class=\"data row5 col28\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col29\" class=\"data row5 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_96fe1_row5_col30\" class=\"data row5 col30\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col31\" class=\"data row5 col31\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col32\" class=\"data row5 col32\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col33\" class=\"data row5 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_96fe1_row5_col34\" class=\"data row5 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_96fe1_row5_col35\" class=\"data row5 col35\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col36\" class=\"data row5 col36\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_96fe1_row5_col37\" class=\"data row5 col37\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col38\" class=\"data row5 col38\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col39\" class=\"data row5 col39\" ></td>\n",
       "      <td id=\"T_96fe1_row5_col40\" class=\"data row5 col40\" >E19</td>\n",
       "      <td id=\"T_96fe1_row5_col41\" class=\"data row5 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_96fe1_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_96fe1_row6_col0\" class=\"data row6 col0\" >SRX3334819</td>\n",
       "      <td id=\"T_96fe1_row6_col1\" class=\"data row6 col1\" >SRP121892</td>\n",
       "      <td id=\"T_96fe1_row6_col2\" class=\"data row6 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_96fe1_row6_col3\" class=\"data row6 col3\" >SRS2637225</td>\n",
       "      <td id=\"T_96fe1_row6_col4\" class=\"data row6 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_96fe1_row6_col5\" class=\"data row6 col5\" >nidopallium</td>\n",
       "      <td id=\"T_96fe1_row6_col6\" class=\"data row6 col6\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col7\" class=\"data row6 col7\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col8\" class=\"data row6 col8\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col9\" class=\"data row6 col9\" >brain - posterior nidopallium</td>\n",
       "      <td id=\"T_96fe1_row6_col10\" class=\"data row6 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_96fe1_row6_col11\" class=\"data row6 col11\" >missing child term</td>\n",
       "      <td id=\"T_96fe1_row6_col12\" class=\"data row6 col12\" >not documented</td>\n",
       "      <td id=\"T_96fe1_row6_col13\" class=\"data row6 col13\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col14\" class=\"data row6 col14\" >not collected</td>\n",
       "      <td id=\"T_96fe1_row6_col15\" class=\"data row6 col15\" >White leghorn</td>\n",
       "      <td id=\"T_96fe1_row6_col16\" class=\"data row6 col16\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col17\" class=\"data row6 col17\" >9031</td>\n",
       "      <td id=\"T_96fe1_row6_col18\" class=\"data row6 col18\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col19\" class=\"data row6 col19\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col20\" class=\"data row6 col20\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col21\" class=\"data row6 col21\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col22\" class=\"data row6 col22\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col23\" class=\"data row6 col23\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_96fe1_row6_col24\" class=\"data row6 col24\" >SAMN07839994</td>\n",
       "      <td id=\"T_96fe1_row6_col25\" class=\"data row6 col25\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col26\" class=\"data row6 col26\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col27\" class=\"data row6 col27\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col28\" class=\"data row6 col28\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col29\" class=\"data row6 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_96fe1_row6_col30\" class=\"data row6 col30\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col31\" class=\"data row6 col31\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col32\" class=\"data row6 col32\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col33\" class=\"data row6 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_96fe1_row6_col34\" class=\"data row6 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_96fe1_row6_col35\" class=\"data row6 col35\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col36\" class=\"data row6 col36\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_96fe1_row6_col37\" class=\"data row6 col37\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col38\" class=\"data row6 col38\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col39\" class=\"data row6 col39\" ></td>\n",
       "      <td id=\"T_96fe1_row6_col40\" class=\"data row6 col40\" >E20</td>\n",
       "      <td id=\"T_96fe1_row6_col41\" class=\"data row6 col41\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library = pd.read_csv(library_path_from_script, sep='\\t', index_col=False, keep_default_na=False, na_values=['NULL','null', 'nan','NaN'], dtype=object)\n",
    "display_df(library)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### anatomical entity\n",
    "* [uberon ols](https://www.ebi.ac.uk/ols4/ontologies/uberon)\n",
    "* infoOrgan was supplemented by the paper, originally each just said brain and I added the brain regions specified in the paper\n",
    "* manually did anatId, anatName, anatAnnotationStatus, and anatBiologicalStatus - see comments for extra details"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['brain - Field L' 'brain - HA (hyperpallium apicale)'\n",
      " 'brain - anterior mesopallium' 'brain - arcopallium'\n",
      " 'brain - basorostralis' 'brain - entopallium'\n",
      " 'brain - posterior nidopallium']\n"
     ]
    }
   ],
   "source": [
    "unique_sorted(library, \"infoOrgan\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_0e78d\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_0e78d_level0_col0\" class=\"col_heading level0 col0\" >#libraryId</th>\n",
       "      <th id=\"T_0e78d_level0_col1\" class=\"col_heading level0 col1\" >experimentId</th>\n",
       "      <th id=\"T_0e78d_level0_col2\" class=\"col_heading level0 col2\" >platform</th>\n",
       "      <th id=\"T_0e78d_level0_col3\" class=\"col_heading level0 col3\" >SRSId</th>\n",
       "      <th id=\"T_0e78d_level0_col4\" class=\"col_heading level0 col4\" >anatId</th>\n",
       "      <th id=\"T_0e78d_level0_col5\" class=\"col_heading level0 col5\" >anatName</th>\n",
       "      <th id=\"T_0e78d_level0_col6\" class=\"col_heading level0 col6\" >stageId</th>\n",
       "      <th id=\"T_0e78d_level0_col7\" class=\"col_heading level0 col7\" >stageName</th>\n",
       "      <th id=\"T_0e78d_level0_col8\" class=\"col_heading level0 col8\" >url_GSM</th>\n",
       "      <th id=\"T_0e78d_level0_col9\" class=\"col_heading level0 col9\" >infoOrgan</th>\n",
       "      <th id=\"T_0e78d_level0_col10\" class=\"col_heading level0 col10\" >infoStage</th>\n",
       "      <th id=\"T_0e78d_level0_col11\" class=\"col_heading level0 col11\" >anatAnnotationStatus</th>\n",
       "      <th id=\"T_0e78d_level0_col12\" class=\"col_heading level0 col12\" >anatBiologicalStatus</th>\n",
       "      <th id=\"T_0e78d_level0_col13\" class=\"col_heading level0 col13\" >stageAnnotationStatus</th>\n",
       "      <th id=\"T_0e78d_level0_col14\" class=\"col_heading level0 col14\" >sex</th>\n",
       "      <th id=\"T_0e78d_level0_col15\" class=\"col_heading level0 col15\" >strain</th>\n",
       "      <th id=\"T_0e78d_level0_col16\" class=\"col_heading level0 col16\" >genotype</th>\n",
       "      <th id=\"T_0e78d_level0_col17\" class=\"col_heading level0 col17\" >speciesId</th>\n",
       "      <th id=\"T_0e78d_level0_col18\" class=\"col_heading level0 col18\" >protocol</th>\n",
       "      <th id=\"T_0e78d_level0_col19\" class=\"col_heading level0 col19\" >protocolType</th>\n",
       "      <th id=\"T_0e78d_level0_col20\" class=\"col_heading level0 col20\" >RNASelection</th>\n",
       "      <th id=\"T_0e78d_level0_col21\" class=\"col_heading level0 col21\" >globin_reduction</th>\n",
       "      <th id=\"T_0e78d_level0_col22\" class=\"col_heading level0 col22\" >replicate</th>\n",
       "      <th id=\"T_0e78d_level0_col23\" class=\"col_heading level0 col23\" >lib_name</th>\n",
       "      <th id=\"T_0e78d_level0_col24\" class=\"col_heading level0 col24\" >sampleName</th>\n",
       "      <th id=\"T_0e78d_level0_col25\" class=\"col_heading level0 col25\" >sampleAge_value</th>\n",
       "      <th id=\"T_0e78d_level0_col26\" class=\"col_heading level0 col26\" >sampleAge_unit</th>\n",
       "      <th id=\"T_0e78d_level0_col27\" class=\"col_heading level0 col27\" >PATOid</th>\n",
       "      <th id=\"T_0e78d_level0_col28\" class=\"col_heading level0 col28\" >PATOname</th>\n",
       "      <th id=\"T_0e78d_level0_col29\" class=\"col_heading level0 col29\" >comment</th>\n",
       "      <th id=\"T_0e78d_level0_col30\" class=\"col_heading level0 col30\" >condition</th>\n",
       "      <th id=\"T_0e78d_level0_col31\" class=\"col_heading level0 col31\" >physiologicalStatus</th>\n",
       "      <th id=\"T_0e78d_level0_col32\" class=\"col_heading level0 col32\" >annotatorId</th>\n",
       "      <th id=\"T_0e78d_level0_col33\" class=\"col_heading level0 col33\" >lastModificationDate</th>\n",
       "      <th id=\"T_0e78d_level0_col34\" class=\"col_heading level0 col34\" >library_contruction_protocol</th>\n",
       "      <th id=\"T_0e78d_level0_col35\" class=\"col_heading level0 col35\" >source_qc</th>\n",
       "      <th id=\"T_0e78d_level0_col36\" class=\"col_heading level0 col36\" >lib_name_2</th>\n",
       "      <th id=\"T_0e78d_level0_col37\" class=\"col_heading level0 col37\" >lib_name_3</th>\n",
       "      <th id=\"T_0e78d_level0_col38\" class=\"col_heading level0 col38\" >source_name</th>\n",
       "      <th id=\"T_0e78d_level0_col39\" class=\"col_heading level0 col39\" >individual</th>\n",
       "      <th id=\"T_0e78d_level0_col40\" class=\"col_heading level0 col40\" >infoStage_2</th>\n",
       "      <th id=\"T_0e78d_level0_col41\" class=\"col_heading level0 col41\" >infoStage_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_0e78d_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_0e78d_row0_col0\" class=\"data row0 col0\" >SRX3334825</td>\n",
       "      <td id=\"T_0e78d_row0_col1\" class=\"data row0 col1\" >SRP121892</td>\n",
       "      <td id=\"T_0e78d_row0_col2\" class=\"data row0 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_0e78d_row0_col3\" class=\"data row0 col3\" >SRS2637233</td>\n",
       "      <td id=\"T_0e78d_row0_col4\" class=\"data row0 col4\" >UBERON:0007350</td>\n",
       "      <td id=\"T_0e78d_row0_col5\" class=\"data row0 col5\" >arcopallium</td>\n",
       "      <td id=\"T_0e78d_row0_col6\" class=\"data row0 col6\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col7\" class=\"data row0 col7\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col9\" class=\"data row0 col9\" >brain - arcopallium</td>\n",
       "      <td id=\"T_0e78d_row0_col10\" class=\"data row0 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_0e78d_row0_col11\" class=\"data row0 col11\" >perfect match</td>\n",
       "      <td id=\"T_0e78d_row0_col12\" class=\"data row0 col12\" >not documented</td>\n",
       "      <td id=\"T_0e78d_row0_col13\" class=\"data row0 col13\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col14\" class=\"data row0 col14\" >not collected</td>\n",
       "      <td id=\"T_0e78d_row0_col15\" class=\"data row0 col15\" >White leghorn</td>\n",
       "      <td id=\"T_0e78d_row0_col16\" class=\"data row0 col16\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col17\" class=\"data row0 col17\" >9031</td>\n",
       "      <td id=\"T_0e78d_row0_col18\" class=\"data row0 col18\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col19\" class=\"data row0 col19\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col20\" class=\"data row0 col20\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col21\" class=\"data row0 col21\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col22\" class=\"data row0 col22\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col23\" class=\"data row0 col23\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_0e78d_row0_col24\" class=\"data row0 col24\" >SAMN07839988</td>\n",
       "      <td id=\"T_0e78d_row0_col25\" class=\"data row0 col25\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col26\" class=\"data row0 col26\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col27\" class=\"data row0 col27\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col28\" class=\"data row0 col28\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col29\" class=\"data row0 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_0e78d_row0_col30\" class=\"data row0 col30\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col31\" class=\"data row0 col31\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col32\" class=\"data row0 col32\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col33\" class=\"data row0 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_0e78d_row0_col34\" class=\"data row0 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_0e78d_row0_col35\" class=\"data row0 col35\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col36\" class=\"data row0 col36\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_0e78d_row0_col37\" class=\"data row0 col37\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col38\" class=\"data row0 col38\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col39\" class=\"data row0 col39\" ></td>\n",
       "      <td id=\"T_0e78d_row0_col40\" class=\"data row0 col40\" >E14</td>\n",
       "      <td id=\"T_0e78d_row0_col41\" class=\"data row0 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0e78d_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_0e78d_row1_col0\" class=\"data row1 col0\" >SRX3334824</td>\n",
       "      <td id=\"T_0e78d_row1_col1\" class=\"data row1 col1\" >SRP121892</td>\n",
       "      <td id=\"T_0e78d_row1_col2\" class=\"data row1 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_0e78d_row1_col3\" class=\"data row1 col3\" >SRS2637232</td>\n",
       "      <td id=\"T_0e78d_row1_col4\" class=\"data row1 col4\" >UBERON:0000203</td>\n",
       "      <td id=\"T_0e78d_row1_col5\" class=\"data row1 col5\" >pallium</td>\n",
       "      <td id=\"T_0e78d_row1_col6\" class=\"data row1 col6\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col7\" class=\"data row1 col7\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col8\" class=\"data row1 col8\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col9\" class=\"data row1 col9\" >brain - basorostralis</td>\n",
       "      <td id=\"T_0e78d_row1_col10\" class=\"data row1 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_0e78d_row1_col11\" class=\"data row1 col11\" >missing child term</td>\n",
       "      <td id=\"T_0e78d_row1_col12\" class=\"data row1 col12\" >not documented</td>\n",
       "      <td id=\"T_0e78d_row1_col13\" class=\"data row1 col13\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col14\" class=\"data row1 col14\" >not collected</td>\n",
       "      <td id=\"T_0e78d_row1_col15\" class=\"data row1 col15\" >White leghorn</td>\n",
       "      <td id=\"T_0e78d_row1_col16\" class=\"data row1 col16\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col17\" class=\"data row1 col17\" >9031</td>\n",
       "      <td id=\"T_0e78d_row1_col18\" class=\"data row1 col18\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col19\" class=\"data row1 col19\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col20\" class=\"data row1 col20\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col21\" class=\"data row1 col21\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col22\" class=\"data row1 col22\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col23\" class=\"data row1 col23\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_0e78d_row1_col24\" class=\"data row1 col24\" >SAMN07839989</td>\n",
       "      <td id=\"T_0e78d_row1_col25\" class=\"data row1 col25\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col26\" class=\"data row1 col26\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col27\" class=\"data row1 col27\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col28\" class=\"data row1 col28\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col29\" class=\"data row1 col29\" >nucleus basorostralis of the pallium (PMID: 15116397), annotated as pallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_0e78d_row1_col30\" class=\"data row1 col30\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col31\" class=\"data row1 col31\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col32\" class=\"data row1 col32\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col33\" class=\"data row1 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_0e78d_row1_col34\" class=\"data row1 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_0e78d_row1_col35\" class=\"data row1 col35\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col36\" class=\"data row1 col36\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_0e78d_row1_col37\" class=\"data row1 col37\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col38\" class=\"data row1 col38\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col39\" class=\"data row1 col39\" ></td>\n",
       "      <td id=\"T_0e78d_row1_col40\" class=\"data row1 col40\" >E15</td>\n",
       "      <td id=\"T_0e78d_row1_col41\" class=\"data row1 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0e78d_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_0e78d_row2_col0\" class=\"data row2 col0\" >SRX3334823</td>\n",
       "      <td id=\"T_0e78d_row2_col1\" class=\"data row2 col1\" >SRP121892</td>\n",
       "      <td id=\"T_0e78d_row2_col2\" class=\"data row2 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_0e78d_row2_col3\" class=\"data row2 col3\" >SRS2637229</td>\n",
       "      <td id=\"T_0e78d_row2_col4\" class=\"data row2 col4\" >UBERON:0014759</td>\n",
       "      <td id=\"T_0e78d_row2_col5\" class=\"data row2 col5\" >entopallium</td>\n",
       "      <td id=\"T_0e78d_row2_col6\" class=\"data row2 col6\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col7\" class=\"data row2 col7\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col8\" class=\"data row2 col8\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col9\" class=\"data row2 col9\" >brain - entopallium</td>\n",
       "      <td id=\"T_0e78d_row2_col10\" class=\"data row2 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_0e78d_row2_col11\" class=\"data row2 col11\" >perfect match</td>\n",
       "      <td id=\"T_0e78d_row2_col12\" class=\"data row2 col12\" >not documented</td>\n",
       "      <td id=\"T_0e78d_row2_col13\" class=\"data row2 col13\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col14\" class=\"data row2 col14\" >not collected</td>\n",
       "      <td id=\"T_0e78d_row2_col15\" class=\"data row2 col15\" >White leghorn</td>\n",
       "      <td id=\"T_0e78d_row2_col16\" class=\"data row2 col16\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col17\" class=\"data row2 col17\" >9031</td>\n",
       "      <td id=\"T_0e78d_row2_col18\" class=\"data row2 col18\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col19\" class=\"data row2 col19\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col20\" class=\"data row2 col20\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col21\" class=\"data row2 col21\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col22\" class=\"data row2 col22\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col23\" class=\"data row2 col23\" >Gga_entopallium</td>\n",
       "      <td id=\"T_0e78d_row2_col24\" class=\"data row2 col24\" >SAMN07839990</td>\n",
       "      <td id=\"T_0e78d_row2_col25\" class=\"data row2 col25\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col26\" class=\"data row2 col26\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col27\" class=\"data row2 col27\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col28\" class=\"data row2 col28\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col29\" class=\"data row2 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_0e78d_row2_col30\" class=\"data row2 col30\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col31\" class=\"data row2 col31\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col32\" class=\"data row2 col32\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col33\" class=\"data row2 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_0e78d_row2_col34\" class=\"data row2 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_0e78d_row2_col35\" class=\"data row2 col35\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col36\" class=\"data row2 col36\" >Gga_entopallium</td>\n",
       "      <td id=\"T_0e78d_row2_col37\" class=\"data row2 col37\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col38\" class=\"data row2 col38\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col39\" class=\"data row2 col39\" ></td>\n",
       "      <td id=\"T_0e78d_row2_col40\" class=\"data row2 col40\" >E16</td>\n",
       "      <td id=\"T_0e78d_row2_col41\" class=\"data row2 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0e78d_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_0e78d_row3_col0\" class=\"data row3 col0\" >SRX3334822</td>\n",
       "      <td id=\"T_0e78d_row3_col1\" class=\"data row3 col1\" >SRP121892</td>\n",
       "      <td id=\"T_0e78d_row3_col2\" class=\"data row3 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_0e78d_row3_col3\" class=\"data row3 col3\" >SRS2637228</td>\n",
       "      <td id=\"T_0e78d_row3_col4\" class=\"data row3 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_0e78d_row3_col5\" class=\"data row3 col5\" >nidopallium</td>\n",
       "      <td id=\"T_0e78d_row3_col6\" class=\"data row3 col6\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col7\" class=\"data row3 col7\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col8\" class=\"data row3 col8\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col9\" class=\"data row3 col9\" >brain - Field L</td>\n",
       "      <td id=\"T_0e78d_row3_col10\" class=\"data row3 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_0e78d_row3_col11\" class=\"data row3 col11\" >missing child term</td>\n",
       "      <td id=\"T_0e78d_row3_col12\" class=\"data row3 col12\" >not documented</td>\n",
       "      <td id=\"T_0e78d_row3_col13\" class=\"data row3 col13\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col14\" class=\"data row3 col14\" >not collected</td>\n",
       "      <td id=\"T_0e78d_row3_col15\" class=\"data row3 col15\" >White leghorn</td>\n",
       "      <td id=\"T_0e78d_row3_col16\" class=\"data row3 col16\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col17\" class=\"data row3 col17\" >9031</td>\n",
       "      <td id=\"T_0e78d_row3_col18\" class=\"data row3 col18\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col19\" class=\"data row3 col19\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col20\" class=\"data row3 col20\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col21\" class=\"data row3 col21\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col22\" class=\"data row3 col22\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col23\" class=\"data row3 col23\" >Gga_FieldL</td>\n",
       "      <td id=\"T_0e78d_row3_col24\" class=\"data row3 col24\" >SAMN07839991</td>\n",
       "      <td id=\"T_0e78d_row3_col25\" class=\"data row3 col25\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col26\" class=\"data row3 col26\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col27\" class=\"data row3 col27\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col28\" class=\"data row3 col28\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col29\" class=\"data row3 col29\" >Field L is the nidopallial region containing the primary auditory thalamorecipient zone (PMID: 15116397), annotated as nidopallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_0e78d_row3_col30\" class=\"data row3 col30\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col31\" class=\"data row3 col31\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col32\" class=\"data row3 col32\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col33\" class=\"data row3 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_0e78d_row3_col34\" class=\"data row3 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_0e78d_row3_col35\" class=\"data row3 col35\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col36\" class=\"data row3 col36\" >Gga_FieldL</td>\n",
       "      <td id=\"T_0e78d_row3_col37\" class=\"data row3 col37\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col38\" class=\"data row3 col38\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col39\" class=\"data row3 col39\" ></td>\n",
       "      <td id=\"T_0e78d_row3_col40\" class=\"data row3 col40\" >E17</td>\n",
       "      <td id=\"T_0e78d_row3_col41\" class=\"data row3 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0e78d_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_0e78d_row4_col0\" class=\"data row4 col0\" >SRX3334821</td>\n",
       "      <td id=\"T_0e78d_row4_col1\" class=\"data row4 col1\" >SRP121892</td>\n",
       "      <td id=\"T_0e78d_row4_col2\" class=\"data row4 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_0e78d_row4_col3\" class=\"data row4 col3\" >SRS2637227</td>\n",
       "      <td id=\"T_0e78d_row4_col4\" class=\"data row4 col4\" >UBERON:0014757</td>\n",
       "      <td id=\"T_0e78d_row4_col5\" class=\"data row4 col5\" >hyperpallium apicale</td>\n",
       "      <td id=\"T_0e78d_row4_col6\" class=\"data row4 col6\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col7\" class=\"data row4 col7\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col8\" class=\"data row4 col8\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col9\" class=\"data row4 col9\" >brain - HA (hyperpallium apicale)</td>\n",
       "      <td id=\"T_0e78d_row4_col10\" class=\"data row4 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_0e78d_row4_col11\" class=\"data row4 col11\" >perfect match</td>\n",
       "      <td id=\"T_0e78d_row4_col12\" class=\"data row4 col12\" >not documented</td>\n",
       "      <td id=\"T_0e78d_row4_col13\" class=\"data row4 col13\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col14\" class=\"data row4 col14\" >not collected</td>\n",
       "      <td id=\"T_0e78d_row4_col15\" class=\"data row4 col15\" >White leghorn</td>\n",
       "      <td id=\"T_0e78d_row4_col16\" class=\"data row4 col16\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col17\" class=\"data row4 col17\" >9031</td>\n",
       "      <td id=\"T_0e78d_row4_col18\" class=\"data row4 col18\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col19\" class=\"data row4 col19\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col20\" class=\"data row4 col20\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col21\" class=\"data row4 col21\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col22\" class=\"data row4 col22\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col23\" class=\"data row4 col23\" >Gga_HA</td>\n",
       "      <td id=\"T_0e78d_row4_col24\" class=\"data row4 col24\" >SAMN07839992</td>\n",
       "      <td id=\"T_0e78d_row4_col25\" class=\"data row4 col25\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col26\" class=\"data row4 col26\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col27\" class=\"data row4 col27\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col28\" class=\"data row4 col28\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col29\" class=\"data row4 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_0e78d_row4_col30\" class=\"data row4 col30\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col31\" class=\"data row4 col31\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col32\" class=\"data row4 col32\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col33\" class=\"data row4 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_0e78d_row4_col34\" class=\"data row4 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_0e78d_row4_col35\" class=\"data row4 col35\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col36\" class=\"data row4 col36\" >Gga_HA</td>\n",
       "      <td id=\"T_0e78d_row4_col37\" class=\"data row4 col37\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col38\" class=\"data row4 col38\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col39\" class=\"data row4 col39\" ></td>\n",
       "      <td id=\"T_0e78d_row4_col40\" class=\"data row4 col40\" >E18</td>\n",
       "      <td id=\"T_0e78d_row4_col41\" class=\"data row4 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0e78d_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_0e78d_row5_col0\" class=\"data row5 col0\" >SRX3334820</td>\n",
       "      <td id=\"T_0e78d_row5_col1\" class=\"data row5 col1\" >SRP121892</td>\n",
       "      <td id=\"T_0e78d_row5_col2\" class=\"data row5 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_0e78d_row5_col3\" class=\"data row5 col3\" >SRS2637226</td>\n",
       "      <td id=\"T_0e78d_row5_col4\" class=\"data row5 col4\" >UBERON:0007349</td>\n",
       "      <td id=\"T_0e78d_row5_col5\" class=\"data row5 col5\" >mesopallium</td>\n",
       "      <td id=\"T_0e78d_row5_col6\" class=\"data row5 col6\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col7\" class=\"data row5 col7\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col8\" class=\"data row5 col8\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col9\" class=\"data row5 col9\" >brain - anterior mesopallium</td>\n",
       "      <td id=\"T_0e78d_row5_col10\" class=\"data row5 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_0e78d_row5_col11\" class=\"data row5 col11\" >missing child term</td>\n",
       "      <td id=\"T_0e78d_row5_col12\" class=\"data row5 col12\" >not documented</td>\n",
       "      <td id=\"T_0e78d_row5_col13\" class=\"data row5 col13\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col14\" class=\"data row5 col14\" >not collected</td>\n",
       "      <td id=\"T_0e78d_row5_col15\" class=\"data row5 col15\" >White leghorn</td>\n",
       "      <td id=\"T_0e78d_row5_col16\" class=\"data row5 col16\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col17\" class=\"data row5 col17\" >9031</td>\n",
       "      <td id=\"T_0e78d_row5_col18\" class=\"data row5 col18\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col19\" class=\"data row5 col19\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col20\" class=\"data row5 col20\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col21\" class=\"data row5 col21\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col22\" class=\"data row5 col22\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col23\" class=\"data row5 col23\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_0e78d_row5_col24\" class=\"data row5 col24\" >SAMN07839993</td>\n",
       "      <td id=\"T_0e78d_row5_col25\" class=\"data row5 col25\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col26\" class=\"data row5 col26\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col27\" class=\"data row5 col27\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col28\" class=\"data row5 col28\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col29\" class=\"data row5 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_0e78d_row5_col30\" class=\"data row5 col30\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col31\" class=\"data row5 col31\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col32\" class=\"data row5 col32\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col33\" class=\"data row5 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_0e78d_row5_col34\" class=\"data row5 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_0e78d_row5_col35\" class=\"data row5 col35\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col36\" class=\"data row5 col36\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_0e78d_row5_col37\" class=\"data row5 col37\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col38\" class=\"data row5 col38\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col39\" class=\"data row5 col39\" ></td>\n",
       "      <td id=\"T_0e78d_row5_col40\" class=\"data row5 col40\" >E19</td>\n",
       "      <td id=\"T_0e78d_row5_col41\" class=\"data row5 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_0e78d_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_0e78d_row6_col0\" class=\"data row6 col0\" >SRX3334819</td>\n",
       "      <td id=\"T_0e78d_row6_col1\" class=\"data row6 col1\" >SRP121892</td>\n",
       "      <td id=\"T_0e78d_row6_col2\" class=\"data row6 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_0e78d_row6_col3\" class=\"data row6 col3\" >SRS2637225</td>\n",
       "      <td id=\"T_0e78d_row6_col4\" class=\"data row6 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_0e78d_row6_col5\" class=\"data row6 col5\" >nidopallium</td>\n",
       "      <td id=\"T_0e78d_row6_col6\" class=\"data row6 col6\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col7\" class=\"data row6 col7\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col8\" class=\"data row6 col8\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col9\" class=\"data row6 col9\" >brain - posterior nidopallium</td>\n",
       "      <td id=\"T_0e78d_row6_col10\" class=\"data row6 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_0e78d_row6_col11\" class=\"data row6 col11\" >missing child term</td>\n",
       "      <td id=\"T_0e78d_row6_col12\" class=\"data row6 col12\" >not documented</td>\n",
       "      <td id=\"T_0e78d_row6_col13\" class=\"data row6 col13\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col14\" class=\"data row6 col14\" >not collected</td>\n",
       "      <td id=\"T_0e78d_row6_col15\" class=\"data row6 col15\" >White leghorn</td>\n",
       "      <td id=\"T_0e78d_row6_col16\" class=\"data row6 col16\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col17\" class=\"data row6 col17\" >9031</td>\n",
       "      <td id=\"T_0e78d_row6_col18\" class=\"data row6 col18\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col19\" class=\"data row6 col19\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col20\" class=\"data row6 col20\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col21\" class=\"data row6 col21\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col22\" class=\"data row6 col22\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col23\" class=\"data row6 col23\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_0e78d_row6_col24\" class=\"data row6 col24\" >SAMN07839994</td>\n",
       "      <td id=\"T_0e78d_row6_col25\" class=\"data row6 col25\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col26\" class=\"data row6 col26\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col27\" class=\"data row6 col27\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col28\" class=\"data row6 col28\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col29\" class=\"data row6 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_0e78d_row6_col30\" class=\"data row6 col30\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col31\" class=\"data row6 col31\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col32\" class=\"data row6 col32\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col33\" class=\"data row6 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_0e78d_row6_col34\" class=\"data row6 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_0e78d_row6_col35\" class=\"data row6 col35\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col36\" class=\"data row6 col36\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_0e78d_row6_col37\" class=\"data row6 col37\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col38\" class=\"data row6 col38\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col39\" class=\"data row6 col39\" ></td>\n",
       "      <td id=\"T_0e78d_row6_col40\" class=\"data row6 col40\" >E20</td>\n",
       "      <td id=\"T_0e78d_row6_col41\" class=\"data row6 col41\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# view manual annotations\n",
    "display_df(library)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### stage\n",
    "- [species specific developmental ontologies](https://github.com/obophenotype/developmental-stage-ontologies/tree/master/src)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Embryonic day 14']\n"
     ]
    }
   ],
   "source": [
    "unique_sorted(library, \"infoStage\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_7b997\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_7b997_level0_col0\" class=\"col_heading level0 col0\" >#libraryId</th>\n",
       "      <th id=\"T_7b997_level0_col1\" class=\"col_heading level0 col1\" >experimentId</th>\n",
       "      <th id=\"T_7b997_level0_col2\" class=\"col_heading level0 col2\" >platform</th>\n",
       "      <th id=\"T_7b997_level0_col3\" class=\"col_heading level0 col3\" >SRSId</th>\n",
       "      <th id=\"T_7b997_level0_col4\" class=\"col_heading level0 col4\" >anatId</th>\n",
       "      <th id=\"T_7b997_level0_col5\" class=\"col_heading level0 col5\" >anatName</th>\n",
       "      <th id=\"T_7b997_level0_col6\" class=\"col_heading level0 col6\" >stageId</th>\n",
       "      <th id=\"T_7b997_level0_col7\" class=\"col_heading level0 col7\" >stageName</th>\n",
       "      <th id=\"T_7b997_level0_col8\" class=\"col_heading level0 col8\" >url_GSM</th>\n",
       "      <th id=\"T_7b997_level0_col9\" class=\"col_heading level0 col9\" >infoOrgan</th>\n",
       "      <th id=\"T_7b997_level0_col10\" class=\"col_heading level0 col10\" >infoStage</th>\n",
       "      <th id=\"T_7b997_level0_col11\" class=\"col_heading level0 col11\" >anatAnnotationStatus</th>\n",
       "      <th id=\"T_7b997_level0_col12\" class=\"col_heading level0 col12\" >anatBiologicalStatus</th>\n",
       "      <th id=\"T_7b997_level0_col13\" class=\"col_heading level0 col13\" >stageAnnotationStatus</th>\n",
       "      <th id=\"T_7b997_level0_col14\" class=\"col_heading level0 col14\" >sex</th>\n",
       "      <th id=\"T_7b997_level0_col15\" class=\"col_heading level0 col15\" >strain</th>\n",
       "      <th id=\"T_7b997_level0_col16\" class=\"col_heading level0 col16\" >genotype</th>\n",
       "      <th id=\"T_7b997_level0_col17\" class=\"col_heading level0 col17\" >speciesId</th>\n",
       "      <th id=\"T_7b997_level0_col18\" class=\"col_heading level0 col18\" >protocol</th>\n",
       "      <th id=\"T_7b997_level0_col19\" class=\"col_heading level0 col19\" >protocolType</th>\n",
       "      <th id=\"T_7b997_level0_col20\" class=\"col_heading level0 col20\" >RNASelection</th>\n",
       "      <th id=\"T_7b997_level0_col21\" class=\"col_heading level0 col21\" >globin_reduction</th>\n",
       "      <th id=\"T_7b997_level0_col22\" class=\"col_heading level0 col22\" >replicate</th>\n",
       "      <th id=\"T_7b997_level0_col23\" class=\"col_heading level0 col23\" >lib_name</th>\n",
       "      <th id=\"T_7b997_level0_col24\" class=\"col_heading level0 col24\" >sampleName</th>\n",
       "      <th id=\"T_7b997_level0_col25\" class=\"col_heading level0 col25\" >sampleAge_value</th>\n",
       "      <th id=\"T_7b997_level0_col26\" class=\"col_heading level0 col26\" >sampleAge_unit</th>\n",
       "      <th id=\"T_7b997_level0_col27\" class=\"col_heading level0 col27\" >PATOid</th>\n",
       "      <th id=\"T_7b997_level0_col28\" class=\"col_heading level0 col28\" >PATOname</th>\n",
       "      <th id=\"T_7b997_level0_col29\" class=\"col_heading level0 col29\" >comment</th>\n",
       "      <th id=\"T_7b997_level0_col30\" class=\"col_heading level0 col30\" >condition</th>\n",
       "      <th id=\"T_7b997_level0_col31\" class=\"col_heading level0 col31\" >physiologicalStatus</th>\n",
       "      <th id=\"T_7b997_level0_col32\" class=\"col_heading level0 col32\" >annotatorId</th>\n",
       "      <th id=\"T_7b997_level0_col33\" class=\"col_heading level0 col33\" >lastModificationDate</th>\n",
       "      <th id=\"T_7b997_level0_col34\" class=\"col_heading level0 col34\" >library_contruction_protocol</th>\n",
       "      <th id=\"T_7b997_level0_col35\" class=\"col_heading level0 col35\" >source_qc</th>\n",
       "      <th id=\"T_7b997_level0_col36\" class=\"col_heading level0 col36\" >lib_name_2</th>\n",
       "      <th id=\"T_7b997_level0_col37\" class=\"col_heading level0 col37\" >lib_name_3</th>\n",
       "      <th id=\"T_7b997_level0_col38\" class=\"col_heading level0 col38\" >source_name</th>\n",
       "      <th id=\"T_7b997_level0_col39\" class=\"col_heading level0 col39\" >individual</th>\n",
       "      <th id=\"T_7b997_level0_col40\" class=\"col_heading level0 col40\" >infoStage_2</th>\n",
       "      <th id=\"T_7b997_level0_col41\" class=\"col_heading level0 col41\" >infoStage_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_7b997_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_7b997_row0_col0\" class=\"data row0 col0\" >SRX3334825</td>\n",
       "      <td id=\"T_7b997_row0_col1\" class=\"data row0 col1\" >SRP121892</td>\n",
       "      <td id=\"T_7b997_row0_col2\" class=\"data row0 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_7b997_row0_col3\" class=\"data row0 col3\" >SRS2637233</td>\n",
       "      <td id=\"T_7b997_row0_col4\" class=\"data row0 col4\" >UBERON:0007350</td>\n",
       "      <td id=\"T_7b997_row0_col5\" class=\"data row0 col5\" >arcopallium</td>\n",
       "      <td id=\"T_7b997_row0_col6\" class=\"data row0 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_7b997_row0_col7\" class=\"data row0 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_7b997_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_7b997_row0_col9\" class=\"data row0 col9\" >brain - arcopallium</td>\n",
       "      <td id=\"T_7b997_row0_col10\" class=\"data row0 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_7b997_row0_col11\" class=\"data row0 col11\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row0_col12\" class=\"data row0 col12\" >not documented</td>\n",
       "      <td id=\"T_7b997_row0_col13\" class=\"data row0 col13\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row0_col14\" class=\"data row0 col14\" >not collected</td>\n",
       "      <td id=\"T_7b997_row0_col15\" class=\"data row0 col15\" >White leghorn</td>\n",
       "      <td id=\"T_7b997_row0_col16\" class=\"data row0 col16\" ></td>\n",
       "      <td id=\"T_7b997_row0_col17\" class=\"data row0 col17\" >9031</td>\n",
       "      <td id=\"T_7b997_row0_col18\" class=\"data row0 col18\" ></td>\n",
       "      <td id=\"T_7b997_row0_col19\" class=\"data row0 col19\" ></td>\n",
       "      <td id=\"T_7b997_row0_col20\" class=\"data row0 col20\" ></td>\n",
       "      <td id=\"T_7b997_row0_col21\" class=\"data row0 col21\" ></td>\n",
       "      <td id=\"T_7b997_row0_col22\" class=\"data row0 col22\" ></td>\n",
       "      <td id=\"T_7b997_row0_col23\" class=\"data row0 col23\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_7b997_row0_col24\" class=\"data row0 col24\" >SAMN07839988</td>\n",
       "      <td id=\"T_7b997_row0_col25\" class=\"data row0 col25\" ></td>\n",
       "      <td id=\"T_7b997_row0_col26\" class=\"data row0 col26\" ></td>\n",
       "      <td id=\"T_7b997_row0_col27\" class=\"data row0 col27\" ></td>\n",
       "      <td id=\"T_7b997_row0_col28\" class=\"data row0 col28\" ></td>\n",
       "      <td id=\"T_7b997_row0_col29\" class=\"data row0 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_7b997_row0_col30\" class=\"data row0 col30\" ></td>\n",
       "      <td id=\"T_7b997_row0_col31\" class=\"data row0 col31\" ></td>\n",
       "      <td id=\"T_7b997_row0_col32\" class=\"data row0 col32\" ></td>\n",
       "      <td id=\"T_7b997_row0_col33\" class=\"data row0 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_7b997_row0_col34\" class=\"data row0 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_7b997_row0_col35\" class=\"data row0 col35\" ></td>\n",
       "      <td id=\"T_7b997_row0_col36\" class=\"data row0 col36\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_7b997_row0_col37\" class=\"data row0 col37\" ></td>\n",
       "      <td id=\"T_7b997_row0_col38\" class=\"data row0 col38\" ></td>\n",
       "      <td id=\"T_7b997_row0_col39\" class=\"data row0 col39\" ></td>\n",
       "      <td id=\"T_7b997_row0_col40\" class=\"data row0 col40\" >E14</td>\n",
       "      <td id=\"T_7b997_row0_col41\" class=\"data row0 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_7b997_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_7b997_row1_col0\" class=\"data row1 col0\" >SRX3334824</td>\n",
       "      <td id=\"T_7b997_row1_col1\" class=\"data row1 col1\" >SRP121892</td>\n",
       "      <td id=\"T_7b997_row1_col2\" class=\"data row1 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_7b997_row1_col3\" class=\"data row1 col3\" >SRS2637232</td>\n",
       "      <td id=\"T_7b997_row1_col4\" class=\"data row1 col4\" >UBERON:0000203</td>\n",
       "      <td id=\"T_7b997_row1_col5\" class=\"data row1 col5\" >pallium</td>\n",
       "      <td id=\"T_7b997_row1_col6\" class=\"data row1 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_7b997_row1_col7\" class=\"data row1 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_7b997_row1_col8\" class=\"data row1 col8\" ></td>\n",
       "      <td id=\"T_7b997_row1_col9\" class=\"data row1 col9\" >brain - basorostralis</td>\n",
       "      <td id=\"T_7b997_row1_col10\" class=\"data row1 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_7b997_row1_col11\" class=\"data row1 col11\" >missing child term</td>\n",
       "      <td id=\"T_7b997_row1_col12\" class=\"data row1 col12\" >not documented</td>\n",
       "      <td id=\"T_7b997_row1_col13\" class=\"data row1 col13\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row1_col14\" class=\"data row1 col14\" >not collected</td>\n",
       "      <td id=\"T_7b997_row1_col15\" class=\"data row1 col15\" >White leghorn</td>\n",
       "      <td id=\"T_7b997_row1_col16\" class=\"data row1 col16\" ></td>\n",
       "      <td id=\"T_7b997_row1_col17\" class=\"data row1 col17\" >9031</td>\n",
       "      <td id=\"T_7b997_row1_col18\" class=\"data row1 col18\" ></td>\n",
       "      <td id=\"T_7b997_row1_col19\" class=\"data row1 col19\" ></td>\n",
       "      <td id=\"T_7b997_row1_col20\" class=\"data row1 col20\" ></td>\n",
       "      <td id=\"T_7b997_row1_col21\" class=\"data row1 col21\" ></td>\n",
       "      <td id=\"T_7b997_row1_col22\" class=\"data row1 col22\" ></td>\n",
       "      <td id=\"T_7b997_row1_col23\" class=\"data row1 col23\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_7b997_row1_col24\" class=\"data row1 col24\" >SAMN07839989</td>\n",
       "      <td id=\"T_7b997_row1_col25\" class=\"data row1 col25\" ></td>\n",
       "      <td id=\"T_7b997_row1_col26\" class=\"data row1 col26\" ></td>\n",
       "      <td id=\"T_7b997_row1_col27\" class=\"data row1 col27\" ></td>\n",
       "      <td id=\"T_7b997_row1_col28\" class=\"data row1 col28\" ></td>\n",
       "      <td id=\"T_7b997_row1_col29\" class=\"data row1 col29\" >nucleus basorostralis of the pallium (PMID: 15116397), annotated as pallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_7b997_row1_col30\" class=\"data row1 col30\" ></td>\n",
       "      <td id=\"T_7b997_row1_col31\" class=\"data row1 col31\" ></td>\n",
       "      <td id=\"T_7b997_row1_col32\" class=\"data row1 col32\" ></td>\n",
       "      <td id=\"T_7b997_row1_col33\" class=\"data row1 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_7b997_row1_col34\" class=\"data row1 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_7b997_row1_col35\" class=\"data row1 col35\" ></td>\n",
       "      <td id=\"T_7b997_row1_col36\" class=\"data row1 col36\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_7b997_row1_col37\" class=\"data row1 col37\" ></td>\n",
       "      <td id=\"T_7b997_row1_col38\" class=\"data row1 col38\" ></td>\n",
       "      <td id=\"T_7b997_row1_col39\" class=\"data row1 col39\" ></td>\n",
       "      <td id=\"T_7b997_row1_col40\" class=\"data row1 col40\" >E15</td>\n",
       "      <td id=\"T_7b997_row1_col41\" class=\"data row1 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_7b997_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_7b997_row2_col0\" class=\"data row2 col0\" >SRX3334823</td>\n",
       "      <td id=\"T_7b997_row2_col1\" class=\"data row2 col1\" >SRP121892</td>\n",
       "      <td id=\"T_7b997_row2_col2\" class=\"data row2 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_7b997_row2_col3\" class=\"data row2 col3\" >SRS2637229</td>\n",
       "      <td id=\"T_7b997_row2_col4\" class=\"data row2 col4\" >UBERON:0014759</td>\n",
       "      <td id=\"T_7b997_row2_col5\" class=\"data row2 col5\" >entopallium</td>\n",
       "      <td id=\"T_7b997_row2_col6\" class=\"data row2 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_7b997_row2_col7\" class=\"data row2 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_7b997_row2_col8\" class=\"data row2 col8\" ></td>\n",
       "      <td id=\"T_7b997_row2_col9\" class=\"data row2 col9\" >brain - entopallium</td>\n",
       "      <td id=\"T_7b997_row2_col10\" class=\"data row2 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_7b997_row2_col11\" class=\"data row2 col11\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row2_col12\" class=\"data row2 col12\" >not documented</td>\n",
       "      <td id=\"T_7b997_row2_col13\" class=\"data row2 col13\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row2_col14\" class=\"data row2 col14\" >not collected</td>\n",
       "      <td id=\"T_7b997_row2_col15\" class=\"data row2 col15\" >White leghorn</td>\n",
       "      <td id=\"T_7b997_row2_col16\" class=\"data row2 col16\" ></td>\n",
       "      <td id=\"T_7b997_row2_col17\" class=\"data row2 col17\" >9031</td>\n",
       "      <td id=\"T_7b997_row2_col18\" class=\"data row2 col18\" ></td>\n",
       "      <td id=\"T_7b997_row2_col19\" class=\"data row2 col19\" ></td>\n",
       "      <td id=\"T_7b997_row2_col20\" class=\"data row2 col20\" ></td>\n",
       "      <td id=\"T_7b997_row2_col21\" class=\"data row2 col21\" ></td>\n",
       "      <td id=\"T_7b997_row2_col22\" class=\"data row2 col22\" ></td>\n",
       "      <td id=\"T_7b997_row2_col23\" class=\"data row2 col23\" >Gga_entopallium</td>\n",
       "      <td id=\"T_7b997_row2_col24\" class=\"data row2 col24\" >SAMN07839990</td>\n",
       "      <td id=\"T_7b997_row2_col25\" class=\"data row2 col25\" ></td>\n",
       "      <td id=\"T_7b997_row2_col26\" class=\"data row2 col26\" ></td>\n",
       "      <td id=\"T_7b997_row2_col27\" class=\"data row2 col27\" ></td>\n",
       "      <td id=\"T_7b997_row2_col28\" class=\"data row2 col28\" ></td>\n",
       "      <td id=\"T_7b997_row2_col29\" class=\"data row2 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_7b997_row2_col30\" class=\"data row2 col30\" ></td>\n",
       "      <td id=\"T_7b997_row2_col31\" class=\"data row2 col31\" ></td>\n",
       "      <td id=\"T_7b997_row2_col32\" class=\"data row2 col32\" ></td>\n",
       "      <td id=\"T_7b997_row2_col33\" class=\"data row2 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_7b997_row2_col34\" class=\"data row2 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_7b997_row2_col35\" class=\"data row2 col35\" ></td>\n",
       "      <td id=\"T_7b997_row2_col36\" class=\"data row2 col36\" >Gga_entopallium</td>\n",
       "      <td id=\"T_7b997_row2_col37\" class=\"data row2 col37\" ></td>\n",
       "      <td id=\"T_7b997_row2_col38\" class=\"data row2 col38\" ></td>\n",
       "      <td id=\"T_7b997_row2_col39\" class=\"data row2 col39\" ></td>\n",
       "      <td id=\"T_7b997_row2_col40\" class=\"data row2 col40\" >E16</td>\n",
       "      <td id=\"T_7b997_row2_col41\" class=\"data row2 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_7b997_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_7b997_row3_col0\" class=\"data row3 col0\" >SRX3334822</td>\n",
       "      <td id=\"T_7b997_row3_col1\" class=\"data row3 col1\" >SRP121892</td>\n",
       "      <td id=\"T_7b997_row3_col2\" class=\"data row3 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_7b997_row3_col3\" class=\"data row3 col3\" >SRS2637228</td>\n",
       "      <td id=\"T_7b997_row3_col4\" class=\"data row3 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_7b997_row3_col5\" class=\"data row3 col5\" >nidopallium</td>\n",
       "      <td id=\"T_7b997_row3_col6\" class=\"data row3 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_7b997_row3_col7\" class=\"data row3 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_7b997_row3_col8\" class=\"data row3 col8\" ></td>\n",
       "      <td id=\"T_7b997_row3_col9\" class=\"data row3 col9\" >brain - Field L</td>\n",
       "      <td id=\"T_7b997_row3_col10\" class=\"data row3 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_7b997_row3_col11\" class=\"data row3 col11\" >missing child term</td>\n",
       "      <td id=\"T_7b997_row3_col12\" class=\"data row3 col12\" >not documented</td>\n",
       "      <td id=\"T_7b997_row3_col13\" class=\"data row3 col13\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row3_col14\" class=\"data row3 col14\" >not collected</td>\n",
       "      <td id=\"T_7b997_row3_col15\" class=\"data row3 col15\" >White leghorn</td>\n",
       "      <td id=\"T_7b997_row3_col16\" class=\"data row3 col16\" ></td>\n",
       "      <td id=\"T_7b997_row3_col17\" class=\"data row3 col17\" >9031</td>\n",
       "      <td id=\"T_7b997_row3_col18\" class=\"data row3 col18\" ></td>\n",
       "      <td id=\"T_7b997_row3_col19\" class=\"data row3 col19\" ></td>\n",
       "      <td id=\"T_7b997_row3_col20\" class=\"data row3 col20\" ></td>\n",
       "      <td id=\"T_7b997_row3_col21\" class=\"data row3 col21\" ></td>\n",
       "      <td id=\"T_7b997_row3_col22\" class=\"data row3 col22\" ></td>\n",
       "      <td id=\"T_7b997_row3_col23\" class=\"data row3 col23\" >Gga_FieldL</td>\n",
       "      <td id=\"T_7b997_row3_col24\" class=\"data row3 col24\" >SAMN07839991</td>\n",
       "      <td id=\"T_7b997_row3_col25\" class=\"data row3 col25\" ></td>\n",
       "      <td id=\"T_7b997_row3_col26\" class=\"data row3 col26\" ></td>\n",
       "      <td id=\"T_7b997_row3_col27\" class=\"data row3 col27\" ></td>\n",
       "      <td id=\"T_7b997_row3_col28\" class=\"data row3 col28\" ></td>\n",
       "      <td id=\"T_7b997_row3_col29\" class=\"data row3 col29\" >Field L is the nidopallial region containing the primary auditory thalamorecipient zone (PMID: 15116397), annotated as nidopallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_7b997_row3_col30\" class=\"data row3 col30\" ></td>\n",
       "      <td id=\"T_7b997_row3_col31\" class=\"data row3 col31\" ></td>\n",
       "      <td id=\"T_7b997_row3_col32\" class=\"data row3 col32\" ></td>\n",
       "      <td id=\"T_7b997_row3_col33\" class=\"data row3 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_7b997_row3_col34\" class=\"data row3 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_7b997_row3_col35\" class=\"data row3 col35\" ></td>\n",
       "      <td id=\"T_7b997_row3_col36\" class=\"data row3 col36\" >Gga_FieldL</td>\n",
       "      <td id=\"T_7b997_row3_col37\" class=\"data row3 col37\" ></td>\n",
       "      <td id=\"T_7b997_row3_col38\" class=\"data row3 col38\" ></td>\n",
       "      <td id=\"T_7b997_row3_col39\" class=\"data row3 col39\" ></td>\n",
       "      <td id=\"T_7b997_row3_col40\" class=\"data row3 col40\" >E17</td>\n",
       "      <td id=\"T_7b997_row3_col41\" class=\"data row3 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_7b997_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_7b997_row4_col0\" class=\"data row4 col0\" >SRX3334821</td>\n",
       "      <td id=\"T_7b997_row4_col1\" class=\"data row4 col1\" >SRP121892</td>\n",
       "      <td id=\"T_7b997_row4_col2\" class=\"data row4 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_7b997_row4_col3\" class=\"data row4 col3\" >SRS2637227</td>\n",
       "      <td id=\"T_7b997_row4_col4\" class=\"data row4 col4\" >UBERON:0014757</td>\n",
       "      <td id=\"T_7b997_row4_col5\" class=\"data row4 col5\" >hyperpallium apicale</td>\n",
       "      <td id=\"T_7b997_row4_col6\" class=\"data row4 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_7b997_row4_col7\" class=\"data row4 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_7b997_row4_col8\" class=\"data row4 col8\" ></td>\n",
       "      <td id=\"T_7b997_row4_col9\" class=\"data row4 col9\" >brain - HA (hyperpallium apicale)</td>\n",
       "      <td id=\"T_7b997_row4_col10\" class=\"data row4 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_7b997_row4_col11\" class=\"data row4 col11\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row4_col12\" class=\"data row4 col12\" >not documented</td>\n",
       "      <td id=\"T_7b997_row4_col13\" class=\"data row4 col13\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row4_col14\" class=\"data row4 col14\" >not collected</td>\n",
       "      <td id=\"T_7b997_row4_col15\" class=\"data row4 col15\" >White leghorn</td>\n",
       "      <td id=\"T_7b997_row4_col16\" class=\"data row4 col16\" ></td>\n",
       "      <td id=\"T_7b997_row4_col17\" class=\"data row4 col17\" >9031</td>\n",
       "      <td id=\"T_7b997_row4_col18\" class=\"data row4 col18\" ></td>\n",
       "      <td id=\"T_7b997_row4_col19\" class=\"data row4 col19\" ></td>\n",
       "      <td id=\"T_7b997_row4_col20\" class=\"data row4 col20\" ></td>\n",
       "      <td id=\"T_7b997_row4_col21\" class=\"data row4 col21\" ></td>\n",
       "      <td id=\"T_7b997_row4_col22\" class=\"data row4 col22\" ></td>\n",
       "      <td id=\"T_7b997_row4_col23\" class=\"data row4 col23\" >Gga_HA</td>\n",
       "      <td id=\"T_7b997_row4_col24\" class=\"data row4 col24\" >SAMN07839992</td>\n",
       "      <td id=\"T_7b997_row4_col25\" class=\"data row4 col25\" ></td>\n",
       "      <td id=\"T_7b997_row4_col26\" class=\"data row4 col26\" ></td>\n",
       "      <td id=\"T_7b997_row4_col27\" class=\"data row4 col27\" ></td>\n",
       "      <td id=\"T_7b997_row4_col28\" class=\"data row4 col28\" ></td>\n",
       "      <td id=\"T_7b997_row4_col29\" class=\"data row4 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_7b997_row4_col30\" class=\"data row4 col30\" ></td>\n",
       "      <td id=\"T_7b997_row4_col31\" class=\"data row4 col31\" ></td>\n",
       "      <td id=\"T_7b997_row4_col32\" class=\"data row4 col32\" ></td>\n",
       "      <td id=\"T_7b997_row4_col33\" class=\"data row4 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_7b997_row4_col34\" class=\"data row4 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_7b997_row4_col35\" class=\"data row4 col35\" ></td>\n",
       "      <td id=\"T_7b997_row4_col36\" class=\"data row4 col36\" >Gga_HA</td>\n",
       "      <td id=\"T_7b997_row4_col37\" class=\"data row4 col37\" ></td>\n",
       "      <td id=\"T_7b997_row4_col38\" class=\"data row4 col38\" ></td>\n",
       "      <td id=\"T_7b997_row4_col39\" class=\"data row4 col39\" ></td>\n",
       "      <td id=\"T_7b997_row4_col40\" class=\"data row4 col40\" >E18</td>\n",
       "      <td id=\"T_7b997_row4_col41\" class=\"data row4 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_7b997_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_7b997_row5_col0\" class=\"data row5 col0\" >SRX3334820</td>\n",
       "      <td id=\"T_7b997_row5_col1\" class=\"data row5 col1\" >SRP121892</td>\n",
       "      <td id=\"T_7b997_row5_col2\" class=\"data row5 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_7b997_row5_col3\" class=\"data row5 col3\" >SRS2637226</td>\n",
       "      <td id=\"T_7b997_row5_col4\" class=\"data row5 col4\" >UBERON:0007349</td>\n",
       "      <td id=\"T_7b997_row5_col5\" class=\"data row5 col5\" >mesopallium</td>\n",
       "      <td id=\"T_7b997_row5_col6\" class=\"data row5 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_7b997_row5_col7\" class=\"data row5 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_7b997_row5_col8\" class=\"data row5 col8\" ></td>\n",
       "      <td id=\"T_7b997_row5_col9\" class=\"data row5 col9\" >brain - anterior mesopallium</td>\n",
       "      <td id=\"T_7b997_row5_col10\" class=\"data row5 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_7b997_row5_col11\" class=\"data row5 col11\" >missing child term</td>\n",
       "      <td id=\"T_7b997_row5_col12\" class=\"data row5 col12\" >not documented</td>\n",
       "      <td id=\"T_7b997_row5_col13\" class=\"data row5 col13\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row5_col14\" class=\"data row5 col14\" >not collected</td>\n",
       "      <td id=\"T_7b997_row5_col15\" class=\"data row5 col15\" >White leghorn</td>\n",
       "      <td id=\"T_7b997_row5_col16\" class=\"data row5 col16\" ></td>\n",
       "      <td id=\"T_7b997_row5_col17\" class=\"data row5 col17\" >9031</td>\n",
       "      <td id=\"T_7b997_row5_col18\" class=\"data row5 col18\" ></td>\n",
       "      <td id=\"T_7b997_row5_col19\" class=\"data row5 col19\" ></td>\n",
       "      <td id=\"T_7b997_row5_col20\" class=\"data row5 col20\" ></td>\n",
       "      <td id=\"T_7b997_row5_col21\" class=\"data row5 col21\" ></td>\n",
       "      <td id=\"T_7b997_row5_col22\" class=\"data row5 col22\" ></td>\n",
       "      <td id=\"T_7b997_row5_col23\" class=\"data row5 col23\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_7b997_row5_col24\" class=\"data row5 col24\" >SAMN07839993</td>\n",
       "      <td id=\"T_7b997_row5_col25\" class=\"data row5 col25\" ></td>\n",
       "      <td id=\"T_7b997_row5_col26\" class=\"data row5 col26\" ></td>\n",
       "      <td id=\"T_7b997_row5_col27\" class=\"data row5 col27\" ></td>\n",
       "      <td id=\"T_7b997_row5_col28\" class=\"data row5 col28\" ></td>\n",
       "      <td id=\"T_7b997_row5_col29\" class=\"data row5 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_7b997_row5_col30\" class=\"data row5 col30\" ></td>\n",
       "      <td id=\"T_7b997_row5_col31\" class=\"data row5 col31\" ></td>\n",
       "      <td id=\"T_7b997_row5_col32\" class=\"data row5 col32\" ></td>\n",
       "      <td id=\"T_7b997_row5_col33\" class=\"data row5 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_7b997_row5_col34\" class=\"data row5 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_7b997_row5_col35\" class=\"data row5 col35\" ></td>\n",
       "      <td id=\"T_7b997_row5_col36\" class=\"data row5 col36\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_7b997_row5_col37\" class=\"data row5 col37\" ></td>\n",
       "      <td id=\"T_7b997_row5_col38\" class=\"data row5 col38\" ></td>\n",
       "      <td id=\"T_7b997_row5_col39\" class=\"data row5 col39\" ></td>\n",
       "      <td id=\"T_7b997_row5_col40\" class=\"data row5 col40\" >E19</td>\n",
       "      <td id=\"T_7b997_row5_col41\" class=\"data row5 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_7b997_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_7b997_row6_col0\" class=\"data row6 col0\" >SRX3334819</td>\n",
       "      <td id=\"T_7b997_row6_col1\" class=\"data row6 col1\" >SRP121892</td>\n",
       "      <td id=\"T_7b997_row6_col2\" class=\"data row6 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_7b997_row6_col3\" class=\"data row6 col3\" >SRS2637225</td>\n",
       "      <td id=\"T_7b997_row6_col4\" class=\"data row6 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_7b997_row6_col5\" class=\"data row6 col5\" >nidopallium</td>\n",
       "      <td id=\"T_7b997_row6_col6\" class=\"data row6 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_7b997_row6_col7\" class=\"data row6 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_7b997_row6_col8\" class=\"data row6 col8\" ></td>\n",
       "      <td id=\"T_7b997_row6_col9\" class=\"data row6 col9\" >brain - posterior nidopallium</td>\n",
       "      <td id=\"T_7b997_row6_col10\" class=\"data row6 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_7b997_row6_col11\" class=\"data row6 col11\" >missing child term</td>\n",
       "      <td id=\"T_7b997_row6_col12\" class=\"data row6 col12\" >not documented</td>\n",
       "      <td id=\"T_7b997_row6_col13\" class=\"data row6 col13\" >perfect match</td>\n",
       "      <td id=\"T_7b997_row6_col14\" class=\"data row6 col14\" >not collected</td>\n",
       "      <td id=\"T_7b997_row6_col15\" class=\"data row6 col15\" >White leghorn</td>\n",
       "      <td id=\"T_7b997_row6_col16\" class=\"data row6 col16\" ></td>\n",
       "      <td id=\"T_7b997_row6_col17\" class=\"data row6 col17\" >9031</td>\n",
       "      <td id=\"T_7b997_row6_col18\" class=\"data row6 col18\" ></td>\n",
       "      <td id=\"T_7b997_row6_col19\" class=\"data row6 col19\" ></td>\n",
       "      <td id=\"T_7b997_row6_col20\" class=\"data row6 col20\" ></td>\n",
       "      <td id=\"T_7b997_row6_col21\" class=\"data row6 col21\" ></td>\n",
       "      <td id=\"T_7b997_row6_col22\" class=\"data row6 col22\" ></td>\n",
       "      <td id=\"T_7b997_row6_col23\" class=\"data row6 col23\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_7b997_row6_col24\" class=\"data row6 col24\" >SAMN07839994</td>\n",
       "      <td id=\"T_7b997_row6_col25\" class=\"data row6 col25\" ></td>\n",
       "      <td id=\"T_7b997_row6_col26\" class=\"data row6 col26\" ></td>\n",
       "      <td id=\"T_7b997_row6_col27\" class=\"data row6 col27\" ></td>\n",
       "      <td id=\"T_7b997_row6_col28\" class=\"data row6 col28\" ></td>\n",
       "      <td id=\"T_7b997_row6_col29\" class=\"data row6 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_7b997_row6_col30\" class=\"data row6 col30\" ></td>\n",
       "      <td id=\"T_7b997_row6_col31\" class=\"data row6 col31\" ></td>\n",
       "      <td id=\"T_7b997_row6_col32\" class=\"data row6 col32\" ></td>\n",
       "      <td id=\"T_7b997_row6_col33\" class=\"data row6 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_7b997_row6_col34\" class=\"data row6 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_7b997_row6_col35\" class=\"data row6 col35\" ></td>\n",
       "      <td id=\"T_7b997_row6_col36\" class=\"data row6 col36\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_7b997_row6_col37\" class=\"data row6 col37\" ></td>\n",
       "      <td id=\"T_7b997_row6_col38\" class=\"data row6 col38\" ></td>\n",
       "      <td id=\"T_7b997_row6_col39\" class=\"data row6 col39\" ></td>\n",
       "      <td id=\"T_7b997_row6_col40\" class=\"data row6 col40\" >E20</td>\n",
       "      <td id=\"T_7b997_row6_col41\" class=\"data row6 col41\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# all\n",
    "library.loc[:,'stageId'] = 'GgalDv:0000054'\n",
    "library.loc[:,'stageName'] = 'Hamburger Hamilton stage 40'\n",
    "# Usually obtained after 14.0 days of incubation.\n",
    "# perfect match, missing child term, other\n",
    "library.loc[:,'stageAnnotationStatus'] = 'perfect match'\n",
    "\n",
    "\n",
    "# view\n",
    "display_df(library)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### sex, strain, genotype, speciesId\n",
    "- uniprot [strain list](https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/docs/strains)\n",
    "- uniprot [species list](https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/docs/speclist)\n",
    "- bgee [strain mapping](https://gitlab.sib.swiss/Bgee/expression-annotations/-/tree/develop/Strains?ref_type=heads)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_85e53\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_85e53_level0_col0\" class=\"col_heading level0 col0\" >#libraryId</th>\n",
       "      <th id=\"T_85e53_level0_col1\" class=\"col_heading level0 col1\" >experimentId</th>\n",
       "      <th id=\"T_85e53_level0_col2\" class=\"col_heading level0 col2\" >platform</th>\n",
       "      <th id=\"T_85e53_level0_col3\" class=\"col_heading level0 col3\" >SRSId</th>\n",
       "      <th id=\"T_85e53_level0_col4\" class=\"col_heading level0 col4\" >anatId</th>\n",
       "      <th id=\"T_85e53_level0_col5\" class=\"col_heading level0 col5\" >anatName</th>\n",
       "      <th id=\"T_85e53_level0_col6\" class=\"col_heading level0 col6\" >stageId</th>\n",
       "      <th id=\"T_85e53_level0_col7\" class=\"col_heading level0 col7\" >stageName</th>\n",
       "      <th id=\"T_85e53_level0_col8\" class=\"col_heading level0 col8\" >url_GSM</th>\n",
       "      <th id=\"T_85e53_level0_col9\" class=\"col_heading level0 col9\" >infoOrgan</th>\n",
       "      <th id=\"T_85e53_level0_col10\" class=\"col_heading level0 col10\" >infoStage</th>\n",
       "      <th id=\"T_85e53_level0_col11\" class=\"col_heading level0 col11\" >anatAnnotationStatus</th>\n",
       "      <th id=\"T_85e53_level0_col12\" class=\"col_heading level0 col12\" >anatBiologicalStatus</th>\n",
       "      <th id=\"T_85e53_level0_col13\" class=\"col_heading level0 col13\" >stageAnnotationStatus</th>\n",
       "      <th id=\"T_85e53_level0_col14\" class=\"col_heading level0 col14\" >sex</th>\n",
       "      <th id=\"T_85e53_level0_col15\" class=\"col_heading level0 col15\" >strain</th>\n",
       "      <th id=\"T_85e53_level0_col16\" class=\"col_heading level0 col16\" >genotype</th>\n",
       "      <th id=\"T_85e53_level0_col17\" class=\"col_heading level0 col17\" >speciesId</th>\n",
       "      <th id=\"T_85e53_level0_col18\" class=\"col_heading level0 col18\" >protocol</th>\n",
       "      <th id=\"T_85e53_level0_col19\" class=\"col_heading level0 col19\" >protocolType</th>\n",
       "      <th id=\"T_85e53_level0_col20\" class=\"col_heading level0 col20\" >RNASelection</th>\n",
       "      <th id=\"T_85e53_level0_col21\" class=\"col_heading level0 col21\" >globin_reduction</th>\n",
       "      <th id=\"T_85e53_level0_col22\" class=\"col_heading level0 col22\" >replicate</th>\n",
       "      <th id=\"T_85e53_level0_col23\" class=\"col_heading level0 col23\" >lib_name</th>\n",
       "      <th id=\"T_85e53_level0_col24\" class=\"col_heading level0 col24\" >sampleName</th>\n",
       "      <th id=\"T_85e53_level0_col25\" class=\"col_heading level0 col25\" >sampleAge_value</th>\n",
       "      <th id=\"T_85e53_level0_col26\" class=\"col_heading level0 col26\" >sampleAge_unit</th>\n",
       "      <th id=\"T_85e53_level0_col27\" class=\"col_heading level0 col27\" >PATOid</th>\n",
       "      <th id=\"T_85e53_level0_col28\" class=\"col_heading level0 col28\" >PATOname</th>\n",
       "      <th id=\"T_85e53_level0_col29\" class=\"col_heading level0 col29\" >comment</th>\n",
       "      <th id=\"T_85e53_level0_col30\" class=\"col_heading level0 col30\" >condition</th>\n",
       "      <th id=\"T_85e53_level0_col31\" class=\"col_heading level0 col31\" >physiologicalStatus</th>\n",
       "      <th id=\"T_85e53_level0_col32\" class=\"col_heading level0 col32\" >annotatorId</th>\n",
       "      <th id=\"T_85e53_level0_col33\" class=\"col_heading level0 col33\" >lastModificationDate</th>\n",
       "      <th id=\"T_85e53_level0_col34\" class=\"col_heading level0 col34\" >library_contruction_protocol</th>\n",
       "      <th id=\"T_85e53_level0_col35\" class=\"col_heading level0 col35\" >source_qc</th>\n",
       "      <th id=\"T_85e53_level0_col36\" class=\"col_heading level0 col36\" >lib_name_2</th>\n",
       "      <th id=\"T_85e53_level0_col37\" class=\"col_heading level0 col37\" >lib_name_3</th>\n",
       "      <th id=\"T_85e53_level0_col38\" class=\"col_heading level0 col38\" >source_name</th>\n",
       "      <th id=\"T_85e53_level0_col39\" class=\"col_heading level0 col39\" >individual</th>\n",
       "      <th id=\"T_85e53_level0_col40\" class=\"col_heading level0 col40\" >infoStage_2</th>\n",
       "      <th id=\"T_85e53_level0_col41\" class=\"col_heading level0 col41\" >infoStage_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_85e53_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_85e53_row0_col0\" class=\"data row0 col0\" >SRX3334825</td>\n",
       "      <td id=\"T_85e53_row0_col1\" class=\"data row0 col1\" >SRP121892</td>\n",
       "      <td id=\"T_85e53_row0_col2\" class=\"data row0 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_85e53_row0_col3\" class=\"data row0 col3\" >SRS2637233</td>\n",
       "      <td id=\"T_85e53_row0_col4\" class=\"data row0 col4\" >UBERON:0007350</td>\n",
       "      <td id=\"T_85e53_row0_col5\" class=\"data row0 col5\" >arcopallium</td>\n",
       "      <td id=\"T_85e53_row0_col6\" class=\"data row0 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_85e53_row0_col7\" class=\"data row0 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_85e53_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_85e53_row0_col9\" class=\"data row0 col9\" >brain - arcopallium</td>\n",
       "      <td id=\"T_85e53_row0_col10\" class=\"data row0 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_85e53_row0_col11\" class=\"data row0 col11\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row0_col12\" class=\"data row0 col12\" >not documented</td>\n",
       "      <td id=\"T_85e53_row0_col13\" class=\"data row0 col13\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row0_col14\" class=\"data row0 col14\" >NA</td>\n",
       "      <td id=\"T_85e53_row0_col15\" class=\"data row0 col15\" >White leghorn</td>\n",
       "      <td id=\"T_85e53_row0_col16\" class=\"data row0 col16\" ></td>\n",
       "      <td id=\"T_85e53_row0_col17\" class=\"data row0 col17\" >9031</td>\n",
       "      <td id=\"T_85e53_row0_col18\" class=\"data row0 col18\" ></td>\n",
       "      <td id=\"T_85e53_row0_col19\" class=\"data row0 col19\" ></td>\n",
       "      <td id=\"T_85e53_row0_col20\" class=\"data row0 col20\" ></td>\n",
       "      <td id=\"T_85e53_row0_col21\" class=\"data row0 col21\" ></td>\n",
       "      <td id=\"T_85e53_row0_col22\" class=\"data row0 col22\" ></td>\n",
       "      <td id=\"T_85e53_row0_col23\" class=\"data row0 col23\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_85e53_row0_col24\" class=\"data row0 col24\" >SAMN07839988</td>\n",
       "      <td id=\"T_85e53_row0_col25\" class=\"data row0 col25\" ></td>\n",
       "      <td id=\"T_85e53_row0_col26\" class=\"data row0 col26\" ></td>\n",
       "      <td id=\"T_85e53_row0_col27\" class=\"data row0 col27\" ></td>\n",
       "      <td id=\"T_85e53_row0_col28\" class=\"data row0 col28\" ></td>\n",
       "      <td id=\"T_85e53_row0_col29\" class=\"data row0 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_85e53_row0_col30\" class=\"data row0 col30\" ></td>\n",
       "      <td id=\"T_85e53_row0_col31\" class=\"data row0 col31\" ></td>\n",
       "      <td id=\"T_85e53_row0_col32\" class=\"data row0 col32\" ></td>\n",
       "      <td id=\"T_85e53_row0_col33\" class=\"data row0 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_85e53_row0_col34\" class=\"data row0 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_85e53_row0_col35\" class=\"data row0 col35\" ></td>\n",
       "      <td id=\"T_85e53_row0_col36\" class=\"data row0 col36\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_85e53_row0_col37\" class=\"data row0 col37\" ></td>\n",
       "      <td id=\"T_85e53_row0_col38\" class=\"data row0 col38\" ></td>\n",
       "      <td id=\"T_85e53_row0_col39\" class=\"data row0 col39\" ></td>\n",
       "      <td id=\"T_85e53_row0_col40\" class=\"data row0 col40\" >E14</td>\n",
       "      <td id=\"T_85e53_row0_col41\" class=\"data row0 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_85e53_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_85e53_row1_col0\" class=\"data row1 col0\" >SRX3334824</td>\n",
       "      <td id=\"T_85e53_row1_col1\" class=\"data row1 col1\" >SRP121892</td>\n",
       "      <td id=\"T_85e53_row1_col2\" class=\"data row1 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_85e53_row1_col3\" class=\"data row1 col3\" >SRS2637232</td>\n",
       "      <td id=\"T_85e53_row1_col4\" class=\"data row1 col4\" >UBERON:0000203</td>\n",
       "      <td id=\"T_85e53_row1_col5\" class=\"data row1 col5\" >pallium</td>\n",
       "      <td id=\"T_85e53_row1_col6\" class=\"data row1 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_85e53_row1_col7\" class=\"data row1 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_85e53_row1_col8\" class=\"data row1 col8\" ></td>\n",
       "      <td id=\"T_85e53_row1_col9\" class=\"data row1 col9\" >brain - basorostralis</td>\n",
       "      <td id=\"T_85e53_row1_col10\" class=\"data row1 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_85e53_row1_col11\" class=\"data row1 col11\" >missing child term</td>\n",
       "      <td id=\"T_85e53_row1_col12\" class=\"data row1 col12\" >not documented</td>\n",
       "      <td id=\"T_85e53_row1_col13\" class=\"data row1 col13\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row1_col14\" class=\"data row1 col14\" >NA</td>\n",
       "      <td id=\"T_85e53_row1_col15\" class=\"data row1 col15\" >White leghorn</td>\n",
       "      <td id=\"T_85e53_row1_col16\" class=\"data row1 col16\" ></td>\n",
       "      <td id=\"T_85e53_row1_col17\" class=\"data row1 col17\" >9031</td>\n",
       "      <td id=\"T_85e53_row1_col18\" class=\"data row1 col18\" ></td>\n",
       "      <td id=\"T_85e53_row1_col19\" class=\"data row1 col19\" ></td>\n",
       "      <td id=\"T_85e53_row1_col20\" class=\"data row1 col20\" ></td>\n",
       "      <td id=\"T_85e53_row1_col21\" class=\"data row1 col21\" ></td>\n",
       "      <td id=\"T_85e53_row1_col22\" class=\"data row1 col22\" ></td>\n",
       "      <td id=\"T_85e53_row1_col23\" class=\"data row1 col23\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_85e53_row1_col24\" class=\"data row1 col24\" >SAMN07839989</td>\n",
       "      <td id=\"T_85e53_row1_col25\" class=\"data row1 col25\" ></td>\n",
       "      <td id=\"T_85e53_row1_col26\" class=\"data row1 col26\" ></td>\n",
       "      <td id=\"T_85e53_row1_col27\" class=\"data row1 col27\" ></td>\n",
       "      <td id=\"T_85e53_row1_col28\" class=\"data row1 col28\" ></td>\n",
       "      <td id=\"T_85e53_row1_col29\" class=\"data row1 col29\" >nucleus basorostralis of the pallium (PMID: 15116397), annotated as pallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_85e53_row1_col30\" class=\"data row1 col30\" ></td>\n",
       "      <td id=\"T_85e53_row1_col31\" class=\"data row1 col31\" ></td>\n",
       "      <td id=\"T_85e53_row1_col32\" class=\"data row1 col32\" ></td>\n",
       "      <td id=\"T_85e53_row1_col33\" class=\"data row1 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_85e53_row1_col34\" class=\"data row1 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_85e53_row1_col35\" class=\"data row1 col35\" ></td>\n",
       "      <td id=\"T_85e53_row1_col36\" class=\"data row1 col36\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_85e53_row1_col37\" class=\"data row1 col37\" ></td>\n",
       "      <td id=\"T_85e53_row1_col38\" class=\"data row1 col38\" ></td>\n",
       "      <td id=\"T_85e53_row1_col39\" class=\"data row1 col39\" ></td>\n",
       "      <td id=\"T_85e53_row1_col40\" class=\"data row1 col40\" >E15</td>\n",
       "      <td id=\"T_85e53_row1_col41\" class=\"data row1 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_85e53_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_85e53_row2_col0\" class=\"data row2 col0\" >SRX3334823</td>\n",
       "      <td id=\"T_85e53_row2_col1\" class=\"data row2 col1\" >SRP121892</td>\n",
       "      <td id=\"T_85e53_row2_col2\" class=\"data row2 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_85e53_row2_col3\" class=\"data row2 col3\" >SRS2637229</td>\n",
       "      <td id=\"T_85e53_row2_col4\" class=\"data row2 col4\" >UBERON:0014759</td>\n",
       "      <td id=\"T_85e53_row2_col5\" class=\"data row2 col5\" >entopallium</td>\n",
       "      <td id=\"T_85e53_row2_col6\" class=\"data row2 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_85e53_row2_col7\" class=\"data row2 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_85e53_row2_col8\" class=\"data row2 col8\" ></td>\n",
       "      <td id=\"T_85e53_row2_col9\" class=\"data row2 col9\" >brain - entopallium</td>\n",
       "      <td id=\"T_85e53_row2_col10\" class=\"data row2 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_85e53_row2_col11\" class=\"data row2 col11\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row2_col12\" class=\"data row2 col12\" >not documented</td>\n",
       "      <td id=\"T_85e53_row2_col13\" class=\"data row2 col13\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row2_col14\" class=\"data row2 col14\" >NA</td>\n",
       "      <td id=\"T_85e53_row2_col15\" class=\"data row2 col15\" >White leghorn</td>\n",
       "      <td id=\"T_85e53_row2_col16\" class=\"data row2 col16\" ></td>\n",
       "      <td id=\"T_85e53_row2_col17\" class=\"data row2 col17\" >9031</td>\n",
       "      <td id=\"T_85e53_row2_col18\" class=\"data row2 col18\" ></td>\n",
       "      <td id=\"T_85e53_row2_col19\" class=\"data row2 col19\" ></td>\n",
       "      <td id=\"T_85e53_row2_col20\" class=\"data row2 col20\" ></td>\n",
       "      <td id=\"T_85e53_row2_col21\" class=\"data row2 col21\" ></td>\n",
       "      <td id=\"T_85e53_row2_col22\" class=\"data row2 col22\" ></td>\n",
       "      <td id=\"T_85e53_row2_col23\" class=\"data row2 col23\" >Gga_entopallium</td>\n",
       "      <td id=\"T_85e53_row2_col24\" class=\"data row2 col24\" >SAMN07839990</td>\n",
       "      <td id=\"T_85e53_row2_col25\" class=\"data row2 col25\" ></td>\n",
       "      <td id=\"T_85e53_row2_col26\" class=\"data row2 col26\" ></td>\n",
       "      <td id=\"T_85e53_row2_col27\" class=\"data row2 col27\" ></td>\n",
       "      <td id=\"T_85e53_row2_col28\" class=\"data row2 col28\" ></td>\n",
       "      <td id=\"T_85e53_row2_col29\" class=\"data row2 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_85e53_row2_col30\" class=\"data row2 col30\" ></td>\n",
       "      <td id=\"T_85e53_row2_col31\" class=\"data row2 col31\" ></td>\n",
       "      <td id=\"T_85e53_row2_col32\" class=\"data row2 col32\" ></td>\n",
       "      <td id=\"T_85e53_row2_col33\" class=\"data row2 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_85e53_row2_col34\" class=\"data row2 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_85e53_row2_col35\" class=\"data row2 col35\" ></td>\n",
       "      <td id=\"T_85e53_row2_col36\" class=\"data row2 col36\" >Gga_entopallium</td>\n",
       "      <td id=\"T_85e53_row2_col37\" class=\"data row2 col37\" ></td>\n",
       "      <td id=\"T_85e53_row2_col38\" class=\"data row2 col38\" ></td>\n",
       "      <td id=\"T_85e53_row2_col39\" class=\"data row2 col39\" ></td>\n",
       "      <td id=\"T_85e53_row2_col40\" class=\"data row2 col40\" >E16</td>\n",
       "      <td id=\"T_85e53_row2_col41\" class=\"data row2 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_85e53_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_85e53_row3_col0\" class=\"data row3 col0\" >SRX3334822</td>\n",
       "      <td id=\"T_85e53_row3_col1\" class=\"data row3 col1\" >SRP121892</td>\n",
       "      <td id=\"T_85e53_row3_col2\" class=\"data row3 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_85e53_row3_col3\" class=\"data row3 col3\" >SRS2637228</td>\n",
       "      <td id=\"T_85e53_row3_col4\" class=\"data row3 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_85e53_row3_col5\" class=\"data row3 col5\" >nidopallium</td>\n",
       "      <td id=\"T_85e53_row3_col6\" class=\"data row3 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_85e53_row3_col7\" class=\"data row3 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_85e53_row3_col8\" class=\"data row3 col8\" ></td>\n",
       "      <td id=\"T_85e53_row3_col9\" class=\"data row3 col9\" >brain - Field L</td>\n",
       "      <td id=\"T_85e53_row3_col10\" class=\"data row3 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_85e53_row3_col11\" class=\"data row3 col11\" >missing child term</td>\n",
       "      <td id=\"T_85e53_row3_col12\" class=\"data row3 col12\" >not documented</td>\n",
       "      <td id=\"T_85e53_row3_col13\" class=\"data row3 col13\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row3_col14\" class=\"data row3 col14\" >NA</td>\n",
       "      <td id=\"T_85e53_row3_col15\" class=\"data row3 col15\" >White leghorn</td>\n",
       "      <td id=\"T_85e53_row3_col16\" class=\"data row3 col16\" ></td>\n",
       "      <td id=\"T_85e53_row3_col17\" class=\"data row3 col17\" >9031</td>\n",
       "      <td id=\"T_85e53_row3_col18\" class=\"data row3 col18\" ></td>\n",
       "      <td id=\"T_85e53_row3_col19\" class=\"data row3 col19\" ></td>\n",
       "      <td id=\"T_85e53_row3_col20\" class=\"data row3 col20\" ></td>\n",
       "      <td id=\"T_85e53_row3_col21\" class=\"data row3 col21\" ></td>\n",
       "      <td id=\"T_85e53_row3_col22\" class=\"data row3 col22\" ></td>\n",
       "      <td id=\"T_85e53_row3_col23\" class=\"data row3 col23\" >Gga_FieldL</td>\n",
       "      <td id=\"T_85e53_row3_col24\" class=\"data row3 col24\" >SAMN07839991</td>\n",
       "      <td id=\"T_85e53_row3_col25\" class=\"data row3 col25\" ></td>\n",
       "      <td id=\"T_85e53_row3_col26\" class=\"data row3 col26\" ></td>\n",
       "      <td id=\"T_85e53_row3_col27\" class=\"data row3 col27\" ></td>\n",
       "      <td id=\"T_85e53_row3_col28\" class=\"data row3 col28\" ></td>\n",
       "      <td id=\"T_85e53_row3_col29\" class=\"data row3 col29\" >Field L is the nidopallial region containing the primary auditory thalamorecipient zone (PMID: 15116397), annotated as nidopallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_85e53_row3_col30\" class=\"data row3 col30\" ></td>\n",
       "      <td id=\"T_85e53_row3_col31\" class=\"data row3 col31\" ></td>\n",
       "      <td id=\"T_85e53_row3_col32\" class=\"data row3 col32\" ></td>\n",
       "      <td id=\"T_85e53_row3_col33\" class=\"data row3 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_85e53_row3_col34\" class=\"data row3 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_85e53_row3_col35\" class=\"data row3 col35\" ></td>\n",
       "      <td id=\"T_85e53_row3_col36\" class=\"data row3 col36\" >Gga_FieldL</td>\n",
       "      <td id=\"T_85e53_row3_col37\" class=\"data row3 col37\" ></td>\n",
       "      <td id=\"T_85e53_row3_col38\" class=\"data row3 col38\" ></td>\n",
       "      <td id=\"T_85e53_row3_col39\" class=\"data row3 col39\" ></td>\n",
       "      <td id=\"T_85e53_row3_col40\" class=\"data row3 col40\" >E17</td>\n",
       "      <td id=\"T_85e53_row3_col41\" class=\"data row3 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_85e53_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_85e53_row4_col0\" class=\"data row4 col0\" >SRX3334821</td>\n",
       "      <td id=\"T_85e53_row4_col1\" class=\"data row4 col1\" >SRP121892</td>\n",
       "      <td id=\"T_85e53_row4_col2\" class=\"data row4 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_85e53_row4_col3\" class=\"data row4 col3\" >SRS2637227</td>\n",
       "      <td id=\"T_85e53_row4_col4\" class=\"data row4 col4\" >UBERON:0014757</td>\n",
       "      <td id=\"T_85e53_row4_col5\" class=\"data row4 col5\" >hyperpallium apicale</td>\n",
       "      <td id=\"T_85e53_row4_col6\" class=\"data row4 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_85e53_row4_col7\" class=\"data row4 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_85e53_row4_col8\" class=\"data row4 col8\" ></td>\n",
       "      <td id=\"T_85e53_row4_col9\" class=\"data row4 col9\" >brain - HA (hyperpallium apicale)</td>\n",
       "      <td id=\"T_85e53_row4_col10\" class=\"data row4 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_85e53_row4_col11\" class=\"data row4 col11\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row4_col12\" class=\"data row4 col12\" >not documented</td>\n",
       "      <td id=\"T_85e53_row4_col13\" class=\"data row4 col13\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row4_col14\" class=\"data row4 col14\" >NA</td>\n",
       "      <td id=\"T_85e53_row4_col15\" class=\"data row4 col15\" >White leghorn</td>\n",
       "      <td id=\"T_85e53_row4_col16\" class=\"data row4 col16\" ></td>\n",
       "      <td id=\"T_85e53_row4_col17\" class=\"data row4 col17\" >9031</td>\n",
       "      <td id=\"T_85e53_row4_col18\" class=\"data row4 col18\" ></td>\n",
       "      <td id=\"T_85e53_row4_col19\" class=\"data row4 col19\" ></td>\n",
       "      <td id=\"T_85e53_row4_col20\" class=\"data row4 col20\" ></td>\n",
       "      <td id=\"T_85e53_row4_col21\" class=\"data row4 col21\" ></td>\n",
       "      <td id=\"T_85e53_row4_col22\" class=\"data row4 col22\" ></td>\n",
       "      <td id=\"T_85e53_row4_col23\" class=\"data row4 col23\" >Gga_HA</td>\n",
       "      <td id=\"T_85e53_row4_col24\" class=\"data row4 col24\" >SAMN07839992</td>\n",
       "      <td id=\"T_85e53_row4_col25\" class=\"data row4 col25\" ></td>\n",
       "      <td id=\"T_85e53_row4_col26\" class=\"data row4 col26\" ></td>\n",
       "      <td id=\"T_85e53_row4_col27\" class=\"data row4 col27\" ></td>\n",
       "      <td id=\"T_85e53_row4_col28\" class=\"data row4 col28\" ></td>\n",
       "      <td id=\"T_85e53_row4_col29\" class=\"data row4 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_85e53_row4_col30\" class=\"data row4 col30\" ></td>\n",
       "      <td id=\"T_85e53_row4_col31\" class=\"data row4 col31\" ></td>\n",
       "      <td id=\"T_85e53_row4_col32\" class=\"data row4 col32\" ></td>\n",
       "      <td id=\"T_85e53_row4_col33\" class=\"data row4 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_85e53_row4_col34\" class=\"data row4 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_85e53_row4_col35\" class=\"data row4 col35\" ></td>\n",
       "      <td id=\"T_85e53_row4_col36\" class=\"data row4 col36\" >Gga_HA</td>\n",
       "      <td id=\"T_85e53_row4_col37\" class=\"data row4 col37\" ></td>\n",
       "      <td id=\"T_85e53_row4_col38\" class=\"data row4 col38\" ></td>\n",
       "      <td id=\"T_85e53_row4_col39\" class=\"data row4 col39\" ></td>\n",
       "      <td id=\"T_85e53_row4_col40\" class=\"data row4 col40\" >E18</td>\n",
       "      <td id=\"T_85e53_row4_col41\" class=\"data row4 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_85e53_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_85e53_row5_col0\" class=\"data row5 col0\" >SRX3334820</td>\n",
       "      <td id=\"T_85e53_row5_col1\" class=\"data row5 col1\" >SRP121892</td>\n",
       "      <td id=\"T_85e53_row5_col2\" class=\"data row5 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_85e53_row5_col3\" class=\"data row5 col3\" >SRS2637226</td>\n",
       "      <td id=\"T_85e53_row5_col4\" class=\"data row5 col4\" >UBERON:0007349</td>\n",
       "      <td id=\"T_85e53_row5_col5\" class=\"data row5 col5\" >mesopallium</td>\n",
       "      <td id=\"T_85e53_row5_col6\" class=\"data row5 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_85e53_row5_col7\" class=\"data row5 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_85e53_row5_col8\" class=\"data row5 col8\" ></td>\n",
       "      <td id=\"T_85e53_row5_col9\" class=\"data row5 col9\" >brain - anterior mesopallium</td>\n",
       "      <td id=\"T_85e53_row5_col10\" class=\"data row5 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_85e53_row5_col11\" class=\"data row5 col11\" >missing child term</td>\n",
       "      <td id=\"T_85e53_row5_col12\" class=\"data row5 col12\" >not documented</td>\n",
       "      <td id=\"T_85e53_row5_col13\" class=\"data row5 col13\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row5_col14\" class=\"data row5 col14\" >NA</td>\n",
       "      <td id=\"T_85e53_row5_col15\" class=\"data row5 col15\" >White leghorn</td>\n",
       "      <td id=\"T_85e53_row5_col16\" class=\"data row5 col16\" ></td>\n",
       "      <td id=\"T_85e53_row5_col17\" class=\"data row5 col17\" >9031</td>\n",
       "      <td id=\"T_85e53_row5_col18\" class=\"data row5 col18\" ></td>\n",
       "      <td id=\"T_85e53_row5_col19\" class=\"data row5 col19\" ></td>\n",
       "      <td id=\"T_85e53_row5_col20\" class=\"data row5 col20\" ></td>\n",
       "      <td id=\"T_85e53_row5_col21\" class=\"data row5 col21\" ></td>\n",
       "      <td id=\"T_85e53_row5_col22\" class=\"data row5 col22\" ></td>\n",
       "      <td id=\"T_85e53_row5_col23\" class=\"data row5 col23\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_85e53_row5_col24\" class=\"data row5 col24\" >SAMN07839993</td>\n",
       "      <td id=\"T_85e53_row5_col25\" class=\"data row5 col25\" ></td>\n",
       "      <td id=\"T_85e53_row5_col26\" class=\"data row5 col26\" ></td>\n",
       "      <td id=\"T_85e53_row5_col27\" class=\"data row5 col27\" ></td>\n",
       "      <td id=\"T_85e53_row5_col28\" class=\"data row5 col28\" ></td>\n",
       "      <td id=\"T_85e53_row5_col29\" class=\"data row5 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_85e53_row5_col30\" class=\"data row5 col30\" ></td>\n",
       "      <td id=\"T_85e53_row5_col31\" class=\"data row5 col31\" ></td>\n",
       "      <td id=\"T_85e53_row5_col32\" class=\"data row5 col32\" ></td>\n",
       "      <td id=\"T_85e53_row5_col33\" class=\"data row5 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_85e53_row5_col34\" class=\"data row5 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_85e53_row5_col35\" class=\"data row5 col35\" ></td>\n",
       "      <td id=\"T_85e53_row5_col36\" class=\"data row5 col36\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_85e53_row5_col37\" class=\"data row5 col37\" ></td>\n",
       "      <td id=\"T_85e53_row5_col38\" class=\"data row5 col38\" ></td>\n",
       "      <td id=\"T_85e53_row5_col39\" class=\"data row5 col39\" ></td>\n",
       "      <td id=\"T_85e53_row5_col40\" class=\"data row5 col40\" >E19</td>\n",
       "      <td id=\"T_85e53_row5_col41\" class=\"data row5 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_85e53_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_85e53_row6_col0\" class=\"data row6 col0\" >SRX3334819</td>\n",
       "      <td id=\"T_85e53_row6_col1\" class=\"data row6 col1\" >SRP121892</td>\n",
       "      <td id=\"T_85e53_row6_col2\" class=\"data row6 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_85e53_row6_col3\" class=\"data row6 col3\" >SRS2637225</td>\n",
       "      <td id=\"T_85e53_row6_col4\" class=\"data row6 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_85e53_row6_col5\" class=\"data row6 col5\" >nidopallium</td>\n",
       "      <td id=\"T_85e53_row6_col6\" class=\"data row6 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_85e53_row6_col7\" class=\"data row6 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_85e53_row6_col8\" class=\"data row6 col8\" ></td>\n",
       "      <td id=\"T_85e53_row6_col9\" class=\"data row6 col9\" >brain - posterior nidopallium</td>\n",
       "      <td id=\"T_85e53_row6_col10\" class=\"data row6 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_85e53_row6_col11\" class=\"data row6 col11\" >missing child term</td>\n",
       "      <td id=\"T_85e53_row6_col12\" class=\"data row6 col12\" >not documented</td>\n",
       "      <td id=\"T_85e53_row6_col13\" class=\"data row6 col13\" >perfect match</td>\n",
       "      <td id=\"T_85e53_row6_col14\" class=\"data row6 col14\" >NA</td>\n",
       "      <td id=\"T_85e53_row6_col15\" class=\"data row6 col15\" >White leghorn</td>\n",
       "      <td id=\"T_85e53_row6_col16\" class=\"data row6 col16\" ></td>\n",
       "      <td id=\"T_85e53_row6_col17\" class=\"data row6 col17\" >9031</td>\n",
       "      <td id=\"T_85e53_row6_col18\" class=\"data row6 col18\" ></td>\n",
       "      <td id=\"T_85e53_row6_col19\" class=\"data row6 col19\" ></td>\n",
       "      <td id=\"T_85e53_row6_col20\" class=\"data row6 col20\" ></td>\n",
       "      <td id=\"T_85e53_row6_col21\" class=\"data row6 col21\" ></td>\n",
       "      <td id=\"T_85e53_row6_col22\" class=\"data row6 col22\" ></td>\n",
       "      <td id=\"T_85e53_row6_col23\" class=\"data row6 col23\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_85e53_row6_col24\" class=\"data row6 col24\" >SAMN07839994</td>\n",
       "      <td id=\"T_85e53_row6_col25\" class=\"data row6 col25\" ></td>\n",
       "      <td id=\"T_85e53_row6_col26\" class=\"data row6 col26\" ></td>\n",
       "      <td id=\"T_85e53_row6_col27\" class=\"data row6 col27\" ></td>\n",
       "      <td id=\"T_85e53_row6_col28\" class=\"data row6 col28\" ></td>\n",
       "      <td id=\"T_85e53_row6_col29\" class=\"data row6 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_85e53_row6_col30\" class=\"data row6 col30\" ></td>\n",
       "      <td id=\"T_85e53_row6_col31\" class=\"data row6 col31\" ></td>\n",
       "      <td id=\"T_85e53_row6_col32\" class=\"data row6 col32\" ></td>\n",
       "      <td id=\"T_85e53_row6_col33\" class=\"data row6 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_85e53_row6_col34\" class=\"data row6 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_85e53_row6_col35\" class=\"data row6 col35\" ></td>\n",
       "      <td id=\"T_85e53_row6_col36\" class=\"data row6 col36\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_85e53_row6_col37\" class=\"data row6 col37\" ></td>\n",
       "      <td id=\"T_85e53_row6_col38\" class=\"data row6 col38\" ></td>\n",
       "      <td id=\"T_85e53_row6_col39\" class=\"data row6 col39\" ></td>\n",
       "      <td id=\"T_85e53_row6_col40\" class=\"data row6 col40\" >E20</td>\n",
       "      <td id=\"T_85e53_row6_col41\" class=\"data row6 col41\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library.loc[library[\"sex\"] == \"not collected\", \"sex\"] = \"NA\"\n",
    "\n",
    "# strain White leghorn is correct as is\n",
    "#library.loc[:,'strain'] = ''\n",
    "\n",
    "#library.loc[:,'genotype'] = ''\n",
    "\n",
    "#library.loc[:,'speciesId'] = ''\n",
    "\n",
    "# view\n",
    "display_df(library)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### protocol\n",
    "see [bulk kits](https://gitlab.sib.swiss/Bgee/scRNA-Seq/-/blob/main/scripts/bulk_kits.csv) for some common protocols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_940f2\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_940f2_level0_col0\" class=\"col_heading level0 col0\" >#libraryId</th>\n",
       "      <th id=\"T_940f2_level0_col1\" class=\"col_heading level0 col1\" >experimentId</th>\n",
       "      <th id=\"T_940f2_level0_col2\" class=\"col_heading level0 col2\" >platform</th>\n",
       "      <th id=\"T_940f2_level0_col3\" class=\"col_heading level0 col3\" >SRSId</th>\n",
       "      <th id=\"T_940f2_level0_col4\" class=\"col_heading level0 col4\" >anatId</th>\n",
       "      <th id=\"T_940f2_level0_col5\" class=\"col_heading level0 col5\" >anatName</th>\n",
       "      <th id=\"T_940f2_level0_col6\" class=\"col_heading level0 col6\" >stageId</th>\n",
       "      <th id=\"T_940f2_level0_col7\" class=\"col_heading level0 col7\" >stageName</th>\n",
       "      <th id=\"T_940f2_level0_col8\" class=\"col_heading level0 col8\" >url_GSM</th>\n",
       "      <th id=\"T_940f2_level0_col9\" class=\"col_heading level0 col9\" >infoOrgan</th>\n",
       "      <th id=\"T_940f2_level0_col10\" class=\"col_heading level0 col10\" >infoStage</th>\n",
       "      <th id=\"T_940f2_level0_col11\" class=\"col_heading level0 col11\" >anatAnnotationStatus</th>\n",
       "      <th id=\"T_940f2_level0_col12\" class=\"col_heading level0 col12\" >anatBiologicalStatus</th>\n",
       "      <th id=\"T_940f2_level0_col13\" class=\"col_heading level0 col13\" >stageAnnotationStatus</th>\n",
       "      <th id=\"T_940f2_level0_col14\" class=\"col_heading level0 col14\" >sex</th>\n",
       "      <th id=\"T_940f2_level0_col15\" class=\"col_heading level0 col15\" >strain</th>\n",
       "      <th id=\"T_940f2_level0_col16\" class=\"col_heading level0 col16\" >genotype</th>\n",
       "      <th id=\"T_940f2_level0_col17\" class=\"col_heading level0 col17\" >speciesId</th>\n",
       "      <th id=\"T_940f2_level0_col18\" class=\"col_heading level0 col18\" >protocol</th>\n",
       "      <th id=\"T_940f2_level0_col19\" class=\"col_heading level0 col19\" >protocolType</th>\n",
       "      <th id=\"T_940f2_level0_col20\" class=\"col_heading level0 col20\" >RNASelection</th>\n",
       "      <th id=\"T_940f2_level0_col21\" class=\"col_heading level0 col21\" >globin_reduction</th>\n",
       "      <th id=\"T_940f2_level0_col22\" class=\"col_heading level0 col22\" >replicate</th>\n",
       "      <th id=\"T_940f2_level0_col23\" class=\"col_heading level0 col23\" >lib_name</th>\n",
       "      <th id=\"T_940f2_level0_col24\" class=\"col_heading level0 col24\" >sampleName</th>\n",
       "      <th id=\"T_940f2_level0_col25\" class=\"col_heading level0 col25\" >sampleAge_value</th>\n",
       "      <th id=\"T_940f2_level0_col26\" class=\"col_heading level0 col26\" >sampleAge_unit</th>\n",
       "      <th id=\"T_940f2_level0_col27\" class=\"col_heading level0 col27\" >PATOid</th>\n",
       "      <th id=\"T_940f2_level0_col28\" class=\"col_heading level0 col28\" >PATOname</th>\n",
       "      <th id=\"T_940f2_level0_col29\" class=\"col_heading level0 col29\" >comment</th>\n",
       "      <th id=\"T_940f2_level0_col30\" class=\"col_heading level0 col30\" >condition</th>\n",
       "      <th id=\"T_940f2_level0_col31\" class=\"col_heading level0 col31\" >physiologicalStatus</th>\n",
       "      <th id=\"T_940f2_level0_col32\" class=\"col_heading level0 col32\" >annotatorId</th>\n",
       "      <th id=\"T_940f2_level0_col33\" class=\"col_heading level0 col33\" >lastModificationDate</th>\n",
       "      <th id=\"T_940f2_level0_col34\" class=\"col_heading level0 col34\" >library_contruction_protocol</th>\n",
       "      <th id=\"T_940f2_level0_col35\" class=\"col_heading level0 col35\" >source_qc</th>\n",
       "      <th id=\"T_940f2_level0_col36\" class=\"col_heading level0 col36\" >lib_name_2</th>\n",
       "      <th id=\"T_940f2_level0_col37\" class=\"col_heading level0 col37\" >lib_name_3</th>\n",
       "      <th id=\"T_940f2_level0_col38\" class=\"col_heading level0 col38\" >source_name</th>\n",
       "      <th id=\"T_940f2_level0_col39\" class=\"col_heading level0 col39\" >individual</th>\n",
       "      <th id=\"T_940f2_level0_col40\" class=\"col_heading level0 col40\" >infoStage_2</th>\n",
       "      <th id=\"T_940f2_level0_col41\" class=\"col_heading level0 col41\" >infoStage_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_940f2_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_940f2_row0_col0\" class=\"data row0 col0\" >SRX3334825</td>\n",
       "      <td id=\"T_940f2_row0_col1\" class=\"data row0 col1\" >SRP121892</td>\n",
       "      <td id=\"T_940f2_row0_col2\" class=\"data row0 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_940f2_row0_col3\" class=\"data row0 col3\" >SRS2637233</td>\n",
       "      <td id=\"T_940f2_row0_col4\" class=\"data row0 col4\" >UBERON:0007350</td>\n",
       "      <td id=\"T_940f2_row0_col5\" class=\"data row0 col5\" >arcopallium</td>\n",
       "      <td id=\"T_940f2_row0_col6\" class=\"data row0 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_940f2_row0_col7\" class=\"data row0 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_940f2_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_940f2_row0_col9\" class=\"data row0 col9\" >brain - arcopallium</td>\n",
       "      <td id=\"T_940f2_row0_col10\" class=\"data row0 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_940f2_row0_col11\" class=\"data row0 col11\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row0_col12\" class=\"data row0 col12\" >not documented</td>\n",
       "      <td id=\"T_940f2_row0_col13\" class=\"data row0 col13\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row0_col14\" class=\"data row0 col14\" >NA</td>\n",
       "      <td id=\"T_940f2_row0_col15\" class=\"data row0 col15\" >White leghorn</td>\n",
       "      <td id=\"T_940f2_row0_col16\" class=\"data row0 col16\" ></td>\n",
       "      <td id=\"T_940f2_row0_col17\" class=\"data row0 col17\" >9031</td>\n",
       "      <td id=\"T_940f2_row0_col18\" class=\"data row0 col18\" ></td>\n",
       "      <td id=\"T_940f2_row0_col19\" class=\"data row0 col19\" ></td>\n",
       "      <td id=\"T_940f2_row0_col20\" class=\"data row0 col20\" >polyA</td>\n",
       "      <td id=\"T_940f2_row0_col21\" class=\"data row0 col21\" ></td>\n",
       "      <td id=\"T_940f2_row0_col22\" class=\"data row0 col22\" ></td>\n",
       "      <td id=\"T_940f2_row0_col23\" class=\"data row0 col23\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_940f2_row0_col24\" class=\"data row0 col24\" >SAMN07839988</td>\n",
       "      <td id=\"T_940f2_row0_col25\" class=\"data row0 col25\" ></td>\n",
       "      <td id=\"T_940f2_row0_col26\" class=\"data row0 col26\" ></td>\n",
       "      <td id=\"T_940f2_row0_col27\" class=\"data row0 col27\" ></td>\n",
       "      <td id=\"T_940f2_row0_col28\" class=\"data row0 col28\" ></td>\n",
       "      <td id=\"T_940f2_row0_col29\" class=\"data row0 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_940f2_row0_col30\" class=\"data row0 col30\" ></td>\n",
       "      <td id=\"T_940f2_row0_col31\" class=\"data row0 col31\" ></td>\n",
       "      <td id=\"T_940f2_row0_col32\" class=\"data row0 col32\" ></td>\n",
       "      <td id=\"T_940f2_row0_col33\" class=\"data row0 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_940f2_row0_col34\" class=\"data row0 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_940f2_row0_col35\" class=\"data row0 col35\" ></td>\n",
       "      <td id=\"T_940f2_row0_col36\" class=\"data row0 col36\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_940f2_row0_col37\" class=\"data row0 col37\" ></td>\n",
       "      <td id=\"T_940f2_row0_col38\" class=\"data row0 col38\" ></td>\n",
       "      <td id=\"T_940f2_row0_col39\" class=\"data row0 col39\" ></td>\n",
       "      <td id=\"T_940f2_row0_col40\" class=\"data row0 col40\" >E14</td>\n",
       "      <td id=\"T_940f2_row0_col41\" class=\"data row0 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_940f2_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_940f2_row1_col0\" class=\"data row1 col0\" >SRX3334824</td>\n",
       "      <td id=\"T_940f2_row1_col1\" class=\"data row1 col1\" >SRP121892</td>\n",
       "      <td id=\"T_940f2_row1_col2\" class=\"data row1 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_940f2_row1_col3\" class=\"data row1 col3\" >SRS2637232</td>\n",
       "      <td id=\"T_940f2_row1_col4\" class=\"data row1 col4\" >UBERON:0000203</td>\n",
       "      <td id=\"T_940f2_row1_col5\" class=\"data row1 col5\" >pallium</td>\n",
       "      <td id=\"T_940f2_row1_col6\" class=\"data row1 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_940f2_row1_col7\" class=\"data row1 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_940f2_row1_col8\" class=\"data row1 col8\" ></td>\n",
       "      <td id=\"T_940f2_row1_col9\" class=\"data row1 col9\" >brain - basorostralis</td>\n",
       "      <td id=\"T_940f2_row1_col10\" class=\"data row1 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_940f2_row1_col11\" class=\"data row1 col11\" >missing child term</td>\n",
       "      <td id=\"T_940f2_row1_col12\" class=\"data row1 col12\" >not documented</td>\n",
       "      <td id=\"T_940f2_row1_col13\" class=\"data row1 col13\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row1_col14\" class=\"data row1 col14\" >NA</td>\n",
       "      <td id=\"T_940f2_row1_col15\" class=\"data row1 col15\" >White leghorn</td>\n",
       "      <td id=\"T_940f2_row1_col16\" class=\"data row1 col16\" ></td>\n",
       "      <td id=\"T_940f2_row1_col17\" class=\"data row1 col17\" >9031</td>\n",
       "      <td id=\"T_940f2_row1_col18\" class=\"data row1 col18\" ></td>\n",
       "      <td id=\"T_940f2_row1_col19\" class=\"data row1 col19\" ></td>\n",
       "      <td id=\"T_940f2_row1_col20\" class=\"data row1 col20\" >polyA</td>\n",
       "      <td id=\"T_940f2_row1_col21\" class=\"data row1 col21\" ></td>\n",
       "      <td id=\"T_940f2_row1_col22\" class=\"data row1 col22\" ></td>\n",
       "      <td id=\"T_940f2_row1_col23\" class=\"data row1 col23\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_940f2_row1_col24\" class=\"data row1 col24\" >SAMN07839989</td>\n",
       "      <td id=\"T_940f2_row1_col25\" class=\"data row1 col25\" ></td>\n",
       "      <td id=\"T_940f2_row1_col26\" class=\"data row1 col26\" ></td>\n",
       "      <td id=\"T_940f2_row1_col27\" class=\"data row1 col27\" ></td>\n",
       "      <td id=\"T_940f2_row1_col28\" class=\"data row1 col28\" ></td>\n",
       "      <td id=\"T_940f2_row1_col29\" class=\"data row1 col29\" >nucleus basorostralis of the pallium (PMID: 15116397), annotated as pallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_940f2_row1_col30\" class=\"data row1 col30\" ></td>\n",
       "      <td id=\"T_940f2_row1_col31\" class=\"data row1 col31\" ></td>\n",
       "      <td id=\"T_940f2_row1_col32\" class=\"data row1 col32\" ></td>\n",
       "      <td id=\"T_940f2_row1_col33\" class=\"data row1 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_940f2_row1_col34\" class=\"data row1 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_940f2_row1_col35\" class=\"data row1 col35\" ></td>\n",
       "      <td id=\"T_940f2_row1_col36\" class=\"data row1 col36\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_940f2_row1_col37\" class=\"data row1 col37\" ></td>\n",
       "      <td id=\"T_940f2_row1_col38\" class=\"data row1 col38\" ></td>\n",
       "      <td id=\"T_940f2_row1_col39\" class=\"data row1 col39\" ></td>\n",
       "      <td id=\"T_940f2_row1_col40\" class=\"data row1 col40\" >E15</td>\n",
       "      <td id=\"T_940f2_row1_col41\" class=\"data row1 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_940f2_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_940f2_row2_col0\" class=\"data row2 col0\" >SRX3334823</td>\n",
       "      <td id=\"T_940f2_row2_col1\" class=\"data row2 col1\" >SRP121892</td>\n",
       "      <td id=\"T_940f2_row2_col2\" class=\"data row2 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_940f2_row2_col3\" class=\"data row2 col3\" >SRS2637229</td>\n",
       "      <td id=\"T_940f2_row2_col4\" class=\"data row2 col4\" >UBERON:0014759</td>\n",
       "      <td id=\"T_940f2_row2_col5\" class=\"data row2 col5\" >entopallium</td>\n",
       "      <td id=\"T_940f2_row2_col6\" class=\"data row2 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_940f2_row2_col7\" class=\"data row2 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_940f2_row2_col8\" class=\"data row2 col8\" ></td>\n",
       "      <td id=\"T_940f2_row2_col9\" class=\"data row2 col9\" >brain - entopallium</td>\n",
       "      <td id=\"T_940f2_row2_col10\" class=\"data row2 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_940f2_row2_col11\" class=\"data row2 col11\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row2_col12\" class=\"data row2 col12\" >not documented</td>\n",
       "      <td id=\"T_940f2_row2_col13\" class=\"data row2 col13\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row2_col14\" class=\"data row2 col14\" >NA</td>\n",
       "      <td id=\"T_940f2_row2_col15\" class=\"data row2 col15\" >White leghorn</td>\n",
       "      <td id=\"T_940f2_row2_col16\" class=\"data row2 col16\" ></td>\n",
       "      <td id=\"T_940f2_row2_col17\" class=\"data row2 col17\" >9031</td>\n",
       "      <td id=\"T_940f2_row2_col18\" class=\"data row2 col18\" ></td>\n",
       "      <td id=\"T_940f2_row2_col19\" class=\"data row2 col19\" ></td>\n",
       "      <td id=\"T_940f2_row2_col20\" class=\"data row2 col20\" >polyA</td>\n",
       "      <td id=\"T_940f2_row2_col21\" class=\"data row2 col21\" ></td>\n",
       "      <td id=\"T_940f2_row2_col22\" class=\"data row2 col22\" ></td>\n",
       "      <td id=\"T_940f2_row2_col23\" class=\"data row2 col23\" >Gga_entopallium</td>\n",
       "      <td id=\"T_940f2_row2_col24\" class=\"data row2 col24\" >SAMN07839990</td>\n",
       "      <td id=\"T_940f2_row2_col25\" class=\"data row2 col25\" ></td>\n",
       "      <td id=\"T_940f2_row2_col26\" class=\"data row2 col26\" ></td>\n",
       "      <td id=\"T_940f2_row2_col27\" class=\"data row2 col27\" ></td>\n",
       "      <td id=\"T_940f2_row2_col28\" class=\"data row2 col28\" ></td>\n",
       "      <td id=\"T_940f2_row2_col29\" class=\"data row2 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_940f2_row2_col30\" class=\"data row2 col30\" ></td>\n",
       "      <td id=\"T_940f2_row2_col31\" class=\"data row2 col31\" ></td>\n",
       "      <td id=\"T_940f2_row2_col32\" class=\"data row2 col32\" ></td>\n",
       "      <td id=\"T_940f2_row2_col33\" class=\"data row2 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_940f2_row2_col34\" class=\"data row2 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_940f2_row2_col35\" class=\"data row2 col35\" ></td>\n",
       "      <td id=\"T_940f2_row2_col36\" class=\"data row2 col36\" >Gga_entopallium</td>\n",
       "      <td id=\"T_940f2_row2_col37\" class=\"data row2 col37\" ></td>\n",
       "      <td id=\"T_940f2_row2_col38\" class=\"data row2 col38\" ></td>\n",
       "      <td id=\"T_940f2_row2_col39\" class=\"data row2 col39\" ></td>\n",
       "      <td id=\"T_940f2_row2_col40\" class=\"data row2 col40\" >E16</td>\n",
       "      <td id=\"T_940f2_row2_col41\" class=\"data row2 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_940f2_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_940f2_row3_col0\" class=\"data row3 col0\" >SRX3334822</td>\n",
       "      <td id=\"T_940f2_row3_col1\" class=\"data row3 col1\" >SRP121892</td>\n",
       "      <td id=\"T_940f2_row3_col2\" class=\"data row3 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_940f2_row3_col3\" class=\"data row3 col3\" >SRS2637228</td>\n",
       "      <td id=\"T_940f2_row3_col4\" class=\"data row3 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_940f2_row3_col5\" class=\"data row3 col5\" >nidopallium</td>\n",
       "      <td id=\"T_940f2_row3_col6\" class=\"data row3 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_940f2_row3_col7\" class=\"data row3 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_940f2_row3_col8\" class=\"data row3 col8\" ></td>\n",
       "      <td id=\"T_940f2_row3_col9\" class=\"data row3 col9\" >brain - Field L</td>\n",
       "      <td id=\"T_940f2_row3_col10\" class=\"data row3 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_940f2_row3_col11\" class=\"data row3 col11\" >missing child term</td>\n",
       "      <td id=\"T_940f2_row3_col12\" class=\"data row3 col12\" >not documented</td>\n",
       "      <td id=\"T_940f2_row3_col13\" class=\"data row3 col13\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row3_col14\" class=\"data row3 col14\" >NA</td>\n",
       "      <td id=\"T_940f2_row3_col15\" class=\"data row3 col15\" >White leghorn</td>\n",
       "      <td id=\"T_940f2_row3_col16\" class=\"data row3 col16\" ></td>\n",
       "      <td id=\"T_940f2_row3_col17\" class=\"data row3 col17\" >9031</td>\n",
       "      <td id=\"T_940f2_row3_col18\" class=\"data row3 col18\" ></td>\n",
       "      <td id=\"T_940f2_row3_col19\" class=\"data row3 col19\" ></td>\n",
       "      <td id=\"T_940f2_row3_col20\" class=\"data row3 col20\" >polyA</td>\n",
       "      <td id=\"T_940f2_row3_col21\" class=\"data row3 col21\" ></td>\n",
       "      <td id=\"T_940f2_row3_col22\" class=\"data row3 col22\" ></td>\n",
       "      <td id=\"T_940f2_row3_col23\" class=\"data row3 col23\" >Gga_FieldL</td>\n",
       "      <td id=\"T_940f2_row3_col24\" class=\"data row3 col24\" >SAMN07839991</td>\n",
       "      <td id=\"T_940f2_row3_col25\" class=\"data row3 col25\" ></td>\n",
       "      <td id=\"T_940f2_row3_col26\" class=\"data row3 col26\" ></td>\n",
       "      <td id=\"T_940f2_row3_col27\" class=\"data row3 col27\" ></td>\n",
       "      <td id=\"T_940f2_row3_col28\" class=\"data row3 col28\" ></td>\n",
       "      <td id=\"T_940f2_row3_col29\" class=\"data row3 col29\" >Field L is the nidopallial region containing the primary auditory thalamorecipient zone (PMID: 15116397), annotated as nidopallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_940f2_row3_col30\" class=\"data row3 col30\" ></td>\n",
       "      <td id=\"T_940f2_row3_col31\" class=\"data row3 col31\" ></td>\n",
       "      <td id=\"T_940f2_row3_col32\" class=\"data row3 col32\" ></td>\n",
       "      <td id=\"T_940f2_row3_col33\" class=\"data row3 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_940f2_row3_col34\" class=\"data row3 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_940f2_row3_col35\" class=\"data row3 col35\" ></td>\n",
       "      <td id=\"T_940f2_row3_col36\" class=\"data row3 col36\" >Gga_FieldL</td>\n",
       "      <td id=\"T_940f2_row3_col37\" class=\"data row3 col37\" ></td>\n",
       "      <td id=\"T_940f2_row3_col38\" class=\"data row3 col38\" ></td>\n",
       "      <td id=\"T_940f2_row3_col39\" class=\"data row3 col39\" ></td>\n",
       "      <td id=\"T_940f2_row3_col40\" class=\"data row3 col40\" >E17</td>\n",
       "      <td id=\"T_940f2_row3_col41\" class=\"data row3 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_940f2_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_940f2_row4_col0\" class=\"data row4 col0\" >SRX3334821</td>\n",
       "      <td id=\"T_940f2_row4_col1\" class=\"data row4 col1\" >SRP121892</td>\n",
       "      <td id=\"T_940f2_row4_col2\" class=\"data row4 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_940f2_row4_col3\" class=\"data row4 col3\" >SRS2637227</td>\n",
       "      <td id=\"T_940f2_row4_col4\" class=\"data row4 col4\" >UBERON:0014757</td>\n",
       "      <td id=\"T_940f2_row4_col5\" class=\"data row4 col5\" >hyperpallium apicale</td>\n",
       "      <td id=\"T_940f2_row4_col6\" class=\"data row4 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_940f2_row4_col7\" class=\"data row4 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_940f2_row4_col8\" class=\"data row4 col8\" ></td>\n",
       "      <td id=\"T_940f2_row4_col9\" class=\"data row4 col9\" >brain - HA (hyperpallium apicale)</td>\n",
       "      <td id=\"T_940f2_row4_col10\" class=\"data row4 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_940f2_row4_col11\" class=\"data row4 col11\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row4_col12\" class=\"data row4 col12\" >not documented</td>\n",
       "      <td id=\"T_940f2_row4_col13\" class=\"data row4 col13\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row4_col14\" class=\"data row4 col14\" >NA</td>\n",
       "      <td id=\"T_940f2_row4_col15\" class=\"data row4 col15\" >White leghorn</td>\n",
       "      <td id=\"T_940f2_row4_col16\" class=\"data row4 col16\" ></td>\n",
       "      <td id=\"T_940f2_row4_col17\" class=\"data row4 col17\" >9031</td>\n",
       "      <td id=\"T_940f2_row4_col18\" class=\"data row4 col18\" ></td>\n",
       "      <td id=\"T_940f2_row4_col19\" class=\"data row4 col19\" ></td>\n",
       "      <td id=\"T_940f2_row4_col20\" class=\"data row4 col20\" >polyA</td>\n",
       "      <td id=\"T_940f2_row4_col21\" class=\"data row4 col21\" ></td>\n",
       "      <td id=\"T_940f2_row4_col22\" class=\"data row4 col22\" ></td>\n",
       "      <td id=\"T_940f2_row4_col23\" class=\"data row4 col23\" >Gga_HA</td>\n",
       "      <td id=\"T_940f2_row4_col24\" class=\"data row4 col24\" >SAMN07839992</td>\n",
       "      <td id=\"T_940f2_row4_col25\" class=\"data row4 col25\" ></td>\n",
       "      <td id=\"T_940f2_row4_col26\" class=\"data row4 col26\" ></td>\n",
       "      <td id=\"T_940f2_row4_col27\" class=\"data row4 col27\" ></td>\n",
       "      <td id=\"T_940f2_row4_col28\" class=\"data row4 col28\" ></td>\n",
       "      <td id=\"T_940f2_row4_col29\" class=\"data row4 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_940f2_row4_col30\" class=\"data row4 col30\" ></td>\n",
       "      <td id=\"T_940f2_row4_col31\" class=\"data row4 col31\" ></td>\n",
       "      <td id=\"T_940f2_row4_col32\" class=\"data row4 col32\" ></td>\n",
       "      <td id=\"T_940f2_row4_col33\" class=\"data row4 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_940f2_row4_col34\" class=\"data row4 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_940f2_row4_col35\" class=\"data row4 col35\" ></td>\n",
       "      <td id=\"T_940f2_row4_col36\" class=\"data row4 col36\" >Gga_HA</td>\n",
       "      <td id=\"T_940f2_row4_col37\" class=\"data row4 col37\" ></td>\n",
       "      <td id=\"T_940f2_row4_col38\" class=\"data row4 col38\" ></td>\n",
       "      <td id=\"T_940f2_row4_col39\" class=\"data row4 col39\" ></td>\n",
       "      <td id=\"T_940f2_row4_col40\" class=\"data row4 col40\" >E18</td>\n",
       "      <td id=\"T_940f2_row4_col41\" class=\"data row4 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_940f2_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_940f2_row5_col0\" class=\"data row5 col0\" >SRX3334820</td>\n",
       "      <td id=\"T_940f2_row5_col1\" class=\"data row5 col1\" >SRP121892</td>\n",
       "      <td id=\"T_940f2_row5_col2\" class=\"data row5 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_940f2_row5_col3\" class=\"data row5 col3\" >SRS2637226</td>\n",
       "      <td id=\"T_940f2_row5_col4\" class=\"data row5 col4\" >UBERON:0007349</td>\n",
       "      <td id=\"T_940f2_row5_col5\" class=\"data row5 col5\" >mesopallium</td>\n",
       "      <td id=\"T_940f2_row5_col6\" class=\"data row5 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_940f2_row5_col7\" class=\"data row5 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_940f2_row5_col8\" class=\"data row5 col8\" ></td>\n",
       "      <td id=\"T_940f2_row5_col9\" class=\"data row5 col9\" >brain - anterior mesopallium</td>\n",
       "      <td id=\"T_940f2_row5_col10\" class=\"data row5 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_940f2_row5_col11\" class=\"data row5 col11\" >missing child term</td>\n",
       "      <td id=\"T_940f2_row5_col12\" class=\"data row5 col12\" >not documented</td>\n",
       "      <td id=\"T_940f2_row5_col13\" class=\"data row5 col13\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row5_col14\" class=\"data row5 col14\" >NA</td>\n",
       "      <td id=\"T_940f2_row5_col15\" class=\"data row5 col15\" >White leghorn</td>\n",
       "      <td id=\"T_940f2_row5_col16\" class=\"data row5 col16\" ></td>\n",
       "      <td id=\"T_940f2_row5_col17\" class=\"data row5 col17\" >9031</td>\n",
       "      <td id=\"T_940f2_row5_col18\" class=\"data row5 col18\" ></td>\n",
       "      <td id=\"T_940f2_row5_col19\" class=\"data row5 col19\" ></td>\n",
       "      <td id=\"T_940f2_row5_col20\" class=\"data row5 col20\" >polyA</td>\n",
       "      <td id=\"T_940f2_row5_col21\" class=\"data row5 col21\" ></td>\n",
       "      <td id=\"T_940f2_row5_col22\" class=\"data row5 col22\" ></td>\n",
       "      <td id=\"T_940f2_row5_col23\" class=\"data row5 col23\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_940f2_row5_col24\" class=\"data row5 col24\" >SAMN07839993</td>\n",
       "      <td id=\"T_940f2_row5_col25\" class=\"data row5 col25\" ></td>\n",
       "      <td id=\"T_940f2_row5_col26\" class=\"data row5 col26\" ></td>\n",
       "      <td id=\"T_940f2_row5_col27\" class=\"data row5 col27\" ></td>\n",
       "      <td id=\"T_940f2_row5_col28\" class=\"data row5 col28\" ></td>\n",
       "      <td id=\"T_940f2_row5_col29\" class=\"data row5 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_940f2_row5_col30\" class=\"data row5 col30\" ></td>\n",
       "      <td id=\"T_940f2_row5_col31\" class=\"data row5 col31\" ></td>\n",
       "      <td id=\"T_940f2_row5_col32\" class=\"data row5 col32\" ></td>\n",
       "      <td id=\"T_940f2_row5_col33\" class=\"data row5 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_940f2_row5_col34\" class=\"data row5 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_940f2_row5_col35\" class=\"data row5 col35\" ></td>\n",
       "      <td id=\"T_940f2_row5_col36\" class=\"data row5 col36\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_940f2_row5_col37\" class=\"data row5 col37\" ></td>\n",
       "      <td id=\"T_940f2_row5_col38\" class=\"data row5 col38\" ></td>\n",
       "      <td id=\"T_940f2_row5_col39\" class=\"data row5 col39\" ></td>\n",
       "      <td id=\"T_940f2_row5_col40\" class=\"data row5 col40\" >E19</td>\n",
       "      <td id=\"T_940f2_row5_col41\" class=\"data row5 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_940f2_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_940f2_row6_col0\" class=\"data row6 col0\" >SRX3334819</td>\n",
       "      <td id=\"T_940f2_row6_col1\" class=\"data row6 col1\" >SRP121892</td>\n",
       "      <td id=\"T_940f2_row6_col2\" class=\"data row6 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_940f2_row6_col3\" class=\"data row6 col3\" >SRS2637225</td>\n",
       "      <td id=\"T_940f2_row6_col4\" class=\"data row6 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_940f2_row6_col5\" class=\"data row6 col5\" >nidopallium</td>\n",
       "      <td id=\"T_940f2_row6_col6\" class=\"data row6 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_940f2_row6_col7\" class=\"data row6 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_940f2_row6_col8\" class=\"data row6 col8\" ></td>\n",
       "      <td id=\"T_940f2_row6_col9\" class=\"data row6 col9\" >brain - posterior nidopallium</td>\n",
       "      <td id=\"T_940f2_row6_col10\" class=\"data row6 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_940f2_row6_col11\" class=\"data row6 col11\" >missing child term</td>\n",
       "      <td id=\"T_940f2_row6_col12\" class=\"data row6 col12\" >not documented</td>\n",
       "      <td id=\"T_940f2_row6_col13\" class=\"data row6 col13\" >perfect match</td>\n",
       "      <td id=\"T_940f2_row6_col14\" class=\"data row6 col14\" >NA</td>\n",
       "      <td id=\"T_940f2_row6_col15\" class=\"data row6 col15\" >White leghorn</td>\n",
       "      <td id=\"T_940f2_row6_col16\" class=\"data row6 col16\" ></td>\n",
       "      <td id=\"T_940f2_row6_col17\" class=\"data row6 col17\" >9031</td>\n",
       "      <td id=\"T_940f2_row6_col18\" class=\"data row6 col18\" ></td>\n",
       "      <td id=\"T_940f2_row6_col19\" class=\"data row6 col19\" ></td>\n",
       "      <td id=\"T_940f2_row6_col20\" class=\"data row6 col20\" >polyA</td>\n",
       "      <td id=\"T_940f2_row6_col21\" class=\"data row6 col21\" ></td>\n",
       "      <td id=\"T_940f2_row6_col22\" class=\"data row6 col22\" ></td>\n",
       "      <td id=\"T_940f2_row6_col23\" class=\"data row6 col23\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_940f2_row6_col24\" class=\"data row6 col24\" >SAMN07839994</td>\n",
       "      <td id=\"T_940f2_row6_col25\" class=\"data row6 col25\" ></td>\n",
       "      <td id=\"T_940f2_row6_col26\" class=\"data row6 col26\" ></td>\n",
       "      <td id=\"T_940f2_row6_col27\" class=\"data row6 col27\" ></td>\n",
       "      <td id=\"T_940f2_row6_col28\" class=\"data row6 col28\" ></td>\n",
       "      <td id=\"T_940f2_row6_col29\" class=\"data row6 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_940f2_row6_col30\" class=\"data row6 col30\" ></td>\n",
       "      <td id=\"T_940f2_row6_col31\" class=\"data row6 col31\" ></td>\n",
       "      <td id=\"T_940f2_row6_col32\" class=\"data row6 col32\" ></td>\n",
       "      <td id=\"T_940f2_row6_col33\" class=\"data row6 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_940f2_row6_col34\" class=\"data row6 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_940f2_row6_col35\" class=\"data row6 col35\" ></td>\n",
       "      <td id=\"T_940f2_row6_col36\" class=\"data row6 col36\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_940f2_row6_col37\" class=\"data row6 col37\" ></td>\n",
       "      <td id=\"T_940f2_row6_col38\" class=\"data row6 col38\" ></td>\n",
       "      <td id=\"T_940f2_row6_col39\" class=\"data row6 col39\" ></td>\n",
       "      <td id=\"T_940f2_row6_col40\" class=\"data row6 col40\" >E20</td>\n",
       "      <td id=\"T_940f2_row6_col41\" class=\"data row6 col41\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# making these variables because we use them again in the experiment file\n",
    "#my_protocol = ''\n",
    "# full_length or 3'\n",
    "#my_protocolType = ''\n",
    "\n",
    "#library.loc[:,'protocol'] = my_protocol\n",
    "#library.loc[:,'protocolType'] = my_protocolType\n",
    "# polyA, ribo-minus, miRNA, lncRNA, circRNA\n",
    "library.loc[:,'RNASelection'] = 'polyA'\n",
    "\n",
    "# view\n",
    "display_df(library)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### globin, replicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no duplicate values in SRSId\n"
     ]
    }
   ],
   "source": [
    "# check for duplicate SRSId values\n",
    "dup_check(library, \"SRSId\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#library.loc[:,'globin_reduction'] = 'Y'\n",
    "\n",
    "# replicates\n",
    "#library.loc[library[\"#libraryId\"] == \"old\", \"replicate\"] = \"1\"\n",
    "#library.loc[library[\"#libraryId\"].isin([\"one\", \"two\"]), \"replicate\"] = \"1\"\n",
    "\n",
    "# view\n",
    "display_df(library)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### sample age, pato, physiological status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_73286\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_73286_level0_col0\" class=\"col_heading level0 col0\" >#libraryId</th>\n",
       "      <th id=\"T_73286_level0_col1\" class=\"col_heading level0 col1\" >experimentId</th>\n",
       "      <th id=\"T_73286_level0_col2\" class=\"col_heading level0 col2\" >platform</th>\n",
       "      <th id=\"T_73286_level0_col3\" class=\"col_heading level0 col3\" >SRSId</th>\n",
       "      <th id=\"T_73286_level0_col4\" class=\"col_heading level0 col4\" >anatId</th>\n",
       "      <th id=\"T_73286_level0_col5\" class=\"col_heading level0 col5\" >anatName</th>\n",
       "      <th id=\"T_73286_level0_col6\" class=\"col_heading level0 col6\" >stageId</th>\n",
       "      <th id=\"T_73286_level0_col7\" class=\"col_heading level0 col7\" >stageName</th>\n",
       "      <th id=\"T_73286_level0_col8\" class=\"col_heading level0 col8\" >url_GSM</th>\n",
       "      <th id=\"T_73286_level0_col9\" class=\"col_heading level0 col9\" >infoOrgan</th>\n",
       "      <th id=\"T_73286_level0_col10\" class=\"col_heading level0 col10\" >infoStage</th>\n",
       "      <th id=\"T_73286_level0_col11\" class=\"col_heading level0 col11\" >anatAnnotationStatus</th>\n",
       "      <th id=\"T_73286_level0_col12\" class=\"col_heading level0 col12\" >anatBiologicalStatus</th>\n",
       "      <th id=\"T_73286_level0_col13\" class=\"col_heading level0 col13\" >stageAnnotationStatus</th>\n",
       "      <th id=\"T_73286_level0_col14\" class=\"col_heading level0 col14\" >sex</th>\n",
       "      <th id=\"T_73286_level0_col15\" class=\"col_heading level0 col15\" >strain</th>\n",
       "      <th id=\"T_73286_level0_col16\" class=\"col_heading level0 col16\" >genotype</th>\n",
       "      <th id=\"T_73286_level0_col17\" class=\"col_heading level0 col17\" >speciesId</th>\n",
       "      <th id=\"T_73286_level0_col18\" class=\"col_heading level0 col18\" >protocol</th>\n",
       "      <th id=\"T_73286_level0_col19\" class=\"col_heading level0 col19\" >protocolType</th>\n",
       "      <th id=\"T_73286_level0_col20\" class=\"col_heading level0 col20\" >RNASelection</th>\n",
       "      <th id=\"T_73286_level0_col21\" class=\"col_heading level0 col21\" >globin_reduction</th>\n",
       "      <th id=\"T_73286_level0_col22\" class=\"col_heading level0 col22\" >replicate</th>\n",
       "      <th id=\"T_73286_level0_col23\" class=\"col_heading level0 col23\" >lib_name</th>\n",
       "      <th id=\"T_73286_level0_col24\" class=\"col_heading level0 col24\" >sampleName</th>\n",
       "      <th id=\"T_73286_level0_col25\" class=\"col_heading level0 col25\" >sampleAge_value</th>\n",
       "      <th id=\"T_73286_level0_col26\" class=\"col_heading level0 col26\" >sampleAge_unit</th>\n",
       "      <th id=\"T_73286_level0_col27\" class=\"col_heading level0 col27\" >PATOid</th>\n",
       "      <th id=\"T_73286_level0_col28\" class=\"col_heading level0 col28\" >PATOname</th>\n",
       "      <th id=\"T_73286_level0_col29\" class=\"col_heading level0 col29\" >comment</th>\n",
       "      <th id=\"T_73286_level0_col30\" class=\"col_heading level0 col30\" >condition</th>\n",
       "      <th id=\"T_73286_level0_col31\" class=\"col_heading level0 col31\" >physiologicalStatus</th>\n",
       "      <th id=\"T_73286_level0_col32\" class=\"col_heading level0 col32\" >annotatorId</th>\n",
       "      <th id=\"T_73286_level0_col33\" class=\"col_heading level0 col33\" >lastModificationDate</th>\n",
       "      <th id=\"T_73286_level0_col34\" class=\"col_heading level0 col34\" >library_contruction_protocol</th>\n",
       "      <th id=\"T_73286_level0_col35\" class=\"col_heading level0 col35\" >source_qc</th>\n",
       "      <th id=\"T_73286_level0_col36\" class=\"col_heading level0 col36\" >lib_name_2</th>\n",
       "      <th id=\"T_73286_level0_col37\" class=\"col_heading level0 col37\" >lib_name_3</th>\n",
       "      <th id=\"T_73286_level0_col38\" class=\"col_heading level0 col38\" >source_name</th>\n",
       "      <th id=\"T_73286_level0_col39\" class=\"col_heading level0 col39\" >individual</th>\n",
       "      <th id=\"T_73286_level0_col40\" class=\"col_heading level0 col40\" >infoStage_2</th>\n",
       "      <th id=\"T_73286_level0_col41\" class=\"col_heading level0 col41\" >infoStage_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_73286_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_73286_row0_col0\" class=\"data row0 col0\" >SRX3334825</td>\n",
       "      <td id=\"T_73286_row0_col1\" class=\"data row0 col1\" >SRP121892</td>\n",
       "      <td id=\"T_73286_row0_col2\" class=\"data row0 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_73286_row0_col3\" class=\"data row0 col3\" >SRS2637233</td>\n",
       "      <td id=\"T_73286_row0_col4\" class=\"data row0 col4\" >UBERON:0007350</td>\n",
       "      <td id=\"T_73286_row0_col5\" class=\"data row0 col5\" >arcopallium</td>\n",
       "      <td id=\"T_73286_row0_col6\" class=\"data row0 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_73286_row0_col7\" class=\"data row0 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_73286_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_73286_row0_col9\" class=\"data row0 col9\" >brain - arcopallium</td>\n",
       "      <td id=\"T_73286_row0_col10\" class=\"data row0 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_73286_row0_col11\" class=\"data row0 col11\" >perfect match</td>\n",
       "      <td id=\"T_73286_row0_col12\" class=\"data row0 col12\" >not documented</td>\n",
       "      <td id=\"T_73286_row0_col13\" class=\"data row0 col13\" >perfect match</td>\n",
       "      <td id=\"T_73286_row0_col14\" class=\"data row0 col14\" >NA</td>\n",
       "      <td id=\"T_73286_row0_col15\" class=\"data row0 col15\" >White leghorn</td>\n",
       "      <td id=\"T_73286_row0_col16\" class=\"data row0 col16\" ></td>\n",
       "      <td id=\"T_73286_row0_col17\" class=\"data row0 col17\" >9031</td>\n",
       "      <td id=\"T_73286_row0_col18\" class=\"data row0 col18\" ></td>\n",
       "      <td id=\"T_73286_row0_col19\" class=\"data row0 col19\" ></td>\n",
       "      <td id=\"T_73286_row0_col20\" class=\"data row0 col20\" >polyA</td>\n",
       "      <td id=\"T_73286_row0_col21\" class=\"data row0 col21\" ></td>\n",
       "      <td id=\"T_73286_row0_col22\" class=\"data row0 col22\" ></td>\n",
       "      <td id=\"T_73286_row0_col23\" class=\"data row0 col23\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_73286_row0_col24\" class=\"data row0 col24\" >SAMN07839988</td>\n",
       "      <td id=\"T_73286_row0_col25\" class=\"data row0 col25\" >14</td>\n",
       "      <td id=\"T_73286_row0_col26\" class=\"data row0 col26\" >embryonic day</td>\n",
       "      <td id=\"T_73286_row0_col27\" class=\"data row0 col27\" ></td>\n",
       "      <td id=\"T_73286_row0_col28\" class=\"data row0 col28\" ></td>\n",
       "      <td id=\"T_73286_row0_col29\" class=\"data row0 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_73286_row0_col30\" class=\"data row0 col30\" ></td>\n",
       "      <td id=\"T_73286_row0_col31\" class=\"data row0 col31\" ></td>\n",
       "      <td id=\"T_73286_row0_col32\" class=\"data row0 col32\" ></td>\n",
       "      <td id=\"T_73286_row0_col33\" class=\"data row0 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_73286_row0_col34\" class=\"data row0 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_73286_row0_col35\" class=\"data row0 col35\" ></td>\n",
       "      <td id=\"T_73286_row0_col36\" class=\"data row0 col36\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_73286_row0_col37\" class=\"data row0 col37\" ></td>\n",
       "      <td id=\"T_73286_row0_col38\" class=\"data row0 col38\" ></td>\n",
       "      <td id=\"T_73286_row0_col39\" class=\"data row0 col39\" ></td>\n",
       "      <td id=\"T_73286_row0_col40\" class=\"data row0 col40\" >E14</td>\n",
       "      <td id=\"T_73286_row0_col41\" class=\"data row0 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_73286_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_73286_row1_col0\" class=\"data row1 col0\" >SRX3334824</td>\n",
       "      <td id=\"T_73286_row1_col1\" class=\"data row1 col1\" >SRP121892</td>\n",
       "      <td id=\"T_73286_row1_col2\" class=\"data row1 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_73286_row1_col3\" class=\"data row1 col3\" >SRS2637232</td>\n",
       "      <td id=\"T_73286_row1_col4\" class=\"data row1 col4\" >UBERON:0000203</td>\n",
       "      <td id=\"T_73286_row1_col5\" class=\"data row1 col5\" >pallium</td>\n",
       "      <td id=\"T_73286_row1_col6\" class=\"data row1 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_73286_row1_col7\" class=\"data row1 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_73286_row1_col8\" class=\"data row1 col8\" ></td>\n",
       "      <td id=\"T_73286_row1_col9\" class=\"data row1 col9\" >brain - basorostralis</td>\n",
       "      <td id=\"T_73286_row1_col10\" class=\"data row1 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_73286_row1_col11\" class=\"data row1 col11\" >missing child term</td>\n",
       "      <td id=\"T_73286_row1_col12\" class=\"data row1 col12\" >not documented</td>\n",
       "      <td id=\"T_73286_row1_col13\" class=\"data row1 col13\" >perfect match</td>\n",
       "      <td id=\"T_73286_row1_col14\" class=\"data row1 col14\" >NA</td>\n",
       "      <td id=\"T_73286_row1_col15\" class=\"data row1 col15\" >White leghorn</td>\n",
       "      <td id=\"T_73286_row1_col16\" class=\"data row1 col16\" ></td>\n",
       "      <td id=\"T_73286_row1_col17\" class=\"data row1 col17\" >9031</td>\n",
       "      <td id=\"T_73286_row1_col18\" class=\"data row1 col18\" ></td>\n",
       "      <td id=\"T_73286_row1_col19\" class=\"data row1 col19\" ></td>\n",
       "      <td id=\"T_73286_row1_col20\" class=\"data row1 col20\" >polyA</td>\n",
       "      <td id=\"T_73286_row1_col21\" class=\"data row1 col21\" ></td>\n",
       "      <td id=\"T_73286_row1_col22\" class=\"data row1 col22\" ></td>\n",
       "      <td id=\"T_73286_row1_col23\" class=\"data row1 col23\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_73286_row1_col24\" class=\"data row1 col24\" >SAMN07839989</td>\n",
       "      <td id=\"T_73286_row1_col25\" class=\"data row1 col25\" >14</td>\n",
       "      <td id=\"T_73286_row1_col26\" class=\"data row1 col26\" >embryonic day</td>\n",
       "      <td id=\"T_73286_row1_col27\" class=\"data row1 col27\" ></td>\n",
       "      <td id=\"T_73286_row1_col28\" class=\"data row1 col28\" ></td>\n",
       "      <td id=\"T_73286_row1_col29\" class=\"data row1 col29\" >nucleus basorostralis of the pallium (PMID: 15116397), annotated as pallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_73286_row1_col30\" class=\"data row1 col30\" ></td>\n",
       "      <td id=\"T_73286_row1_col31\" class=\"data row1 col31\" ></td>\n",
       "      <td id=\"T_73286_row1_col32\" class=\"data row1 col32\" ></td>\n",
       "      <td id=\"T_73286_row1_col33\" class=\"data row1 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_73286_row1_col34\" class=\"data row1 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_73286_row1_col35\" class=\"data row1 col35\" ></td>\n",
       "      <td id=\"T_73286_row1_col36\" class=\"data row1 col36\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_73286_row1_col37\" class=\"data row1 col37\" ></td>\n",
       "      <td id=\"T_73286_row1_col38\" class=\"data row1 col38\" ></td>\n",
       "      <td id=\"T_73286_row1_col39\" class=\"data row1 col39\" ></td>\n",
       "      <td id=\"T_73286_row1_col40\" class=\"data row1 col40\" >E15</td>\n",
       "      <td id=\"T_73286_row1_col41\" class=\"data row1 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_73286_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_73286_row2_col0\" class=\"data row2 col0\" >SRX3334823</td>\n",
       "      <td id=\"T_73286_row2_col1\" class=\"data row2 col1\" >SRP121892</td>\n",
       "      <td id=\"T_73286_row2_col2\" class=\"data row2 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_73286_row2_col3\" class=\"data row2 col3\" >SRS2637229</td>\n",
       "      <td id=\"T_73286_row2_col4\" class=\"data row2 col4\" >UBERON:0014759</td>\n",
       "      <td id=\"T_73286_row2_col5\" class=\"data row2 col5\" >entopallium</td>\n",
       "      <td id=\"T_73286_row2_col6\" class=\"data row2 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_73286_row2_col7\" class=\"data row2 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_73286_row2_col8\" class=\"data row2 col8\" ></td>\n",
       "      <td id=\"T_73286_row2_col9\" class=\"data row2 col9\" >brain - entopallium</td>\n",
       "      <td id=\"T_73286_row2_col10\" class=\"data row2 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_73286_row2_col11\" class=\"data row2 col11\" >perfect match</td>\n",
       "      <td id=\"T_73286_row2_col12\" class=\"data row2 col12\" >not documented</td>\n",
       "      <td id=\"T_73286_row2_col13\" class=\"data row2 col13\" >perfect match</td>\n",
       "      <td id=\"T_73286_row2_col14\" class=\"data row2 col14\" >NA</td>\n",
       "      <td id=\"T_73286_row2_col15\" class=\"data row2 col15\" >White leghorn</td>\n",
       "      <td id=\"T_73286_row2_col16\" class=\"data row2 col16\" ></td>\n",
       "      <td id=\"T_73286_row2_col17\" class=\"data row2 col17\" >9031</td>\n",
       "      <td id=\"T_73286_row2_col18\" class=\"data row2 col18\" ></td>\n",
       "      <td id=\"T_73286_row2_col19\" class=\"data row2 col19\" ></td>\n",
       "      <td id=\"T_73286_row2_col20\" class=\"data row2 col20\" >polyA</td>\n",
       "      <td id=\"T_73286_row2_col21\" class=\"data row2 col21\" ></td>\n",
       "      <td id=\"T_73286_row2_col22\" class=\"data row2 col22\" ></td>\n",
       "      <td id=\"T_73286_row2_col23\" class=\"data row2 col23\" >Gga_entopallium</td>\n",
       "      <td id=\"T_73286_row2_col24\" class=\"data row2 col24\" >SAMN07839990</td>\n",
       "      <td id=\"T_73286_row2_col25\" class=\"data row2 col25\" >14</td>\n",
       "      <td id=\"T_73286_row2_col26\" class=\"data row2 col26\" >embryonic day</td>\n",
       "      <td id=\"T_73286_row2_col27\" class=\"data row2 col27\" ></td>\n",
       "      <td id=\"T_73286_row2_col28\" class=\"data row2 col28\" ></td>\n",
       "      <td id=\"T_73286_row2_col29\" class=\"data row2 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_73286_row2_col30\" class=\"data row2 col30\" ></td>\n",
       "      <td id=\"T_73286_row2_col31\" class=\"data row2 col31\" ></td>\n",
       "      <td id=\"T_73286_row2_col32\" class=\"data row2 col32\" ></td>\n",
       "      <td id=\"T_73286_row2_col33\" class=\"data row2 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_73286_row2_col34\" class=\"data row2 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_73286_row2_col35\" class=\"data row2 col35\" ></td>\n",
       "      <td id=\"T_73286_row2_col36\" class=\"data row2 col36\" >Gga_entopallium</td>\n",
       "      <td id=\"T_73286_row2_col37\" class=\"data row2 col37\" ></td>\n",
       "      <td id=\"T_73286_row2_col38\" class=\"data row2 col38\" ></td>\n",
       "      <td id=\"T_73286_row2_col39\" class=\"data row2 col39\" ></td>\n",
       "      <td id=\"T_73286_row2_col40\" class=\"data row2 col40\" >E16</td>\n",
       "      <td id=\"T_73286_row2_col41\" class=\"data row2 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_73286_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_73286_row3_col0\" class=\"data row3 col0\" >SRX3334822</td>\n",
       "      <td id=\"T_73286_row3_col1\" class=\"data row3 col1\" >SRP121892</td>\n",
       "      <td id=\"T_73286_row3_col2\" class=\"data row3 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_73286_row3_col3\" class=\"data row3 col3\" >SRS2637228</td>\n",
       "      <td id=\"T_73286_row3_col4\" class=\"data row3 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_73286_row3_col5\" class=\"data row3 col5\" >nidopallium</td>\n",
       "      <td id=\"T_73286_row3_col6\" class=\"data row3 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_73286_row3_col7\" class=\"data row3 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_73286_row3_col8\" class=\"data row3 col8\" ></td>\n",
       "      <td id=\"T_73286_row3_col9\" class=\"data row3 col9\" >brain - Field L</td>\n",
       "      <td id=\"T_73286_row3_col10\" class=\"data row3 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_73286_row3_col11\" class=\"data row3 col11\" >missing child term</td>\n",
       "      <td id=\"T_73286_row3_col12\" class=\"data row3 col12\" >not documented</td>\n",
       "      <td id=\"T_73286_row3_col13\" class=\"data row3 col13\" >perfect match</td>\n",
       "      <td id=\"T_73286_row3_col14\" class=\"data row3 col14\" >NA</td>\n",
       "      <td id=\"T_73286_row3_col15\" class=\"data row3 col15\" >White leghorn</td>\n",
       "      <td id=\"T_73286_row3_col16\" class=\"data row3 col16\" ></td>\n",
       "      <td id=\"T_73286_row3_col17\" class=\"data row3 col17\" >9031</td>\n",
       "      <td id=\"T_73286_row3_col18\" class=\"data row3 col18\" ></td>\n",
       "      <td id=\"T_73286_row3_col19\" class=\"data row3 col19\" ></td>\n",
       "      <td id=\"T_73286_row3_col20\" class=\"data row3 col20\" >polyA</td>\n",
       "      <td id=\"T_73286_row3_col21\" class=\"data row3 col21\" ></td>\n",
       "      <td id=\"T_73286_row3_col22\" class=\"data row3 col22\" ></td>\n",
       "      <td id=\"T_73286_row3_col23\" class=\"data row3 col23\" >Gga_FieldL</td>\n",
       "      <td id=\"T_73286_row3_col24\" class=\"data row3 col24\" >SAMN07839991</td>\n",
       "      <td id=\"T_73286_row3_col25\" class=\"data row3 col25\" >14</td>\n",
       "      <td id=\"T_73286_row3_col26\" class=\"data row3 col26\" >embryonic day</td>\n",
       "      <td id=\"T_73286_row3_col27\" class=\"data row3 col27\" ></td>\n",
       "      <td id=\"T_73286_row3_col28\" class=\"data row3 col28\" ></td>\n",
       "      <td id=\"T_73286_row3_col29\" class=\"data row3 col29\" >Field L is the nidopallial region containing the primary auditory thalamorecipient zone (PMID: 15116397), annotated as nidopallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_73286_row3_col30\" class=\"data row3 col30\" ></td>\n",
       "      <td id=\"T_73286_row3_col31\" class=\"data row3 col31\" ></td>\n",
       "      <td id=\"T_73286_row3_col32\" class=\"data row3 col32\" ></td>\n",
       "      <td id=\"T_73286_row3_col33\" class=\"data row3 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_73286_row3_col34\" class=\"data row3 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_73286_row3_col35\" class=\"data row3 col35\" ></td>\n",
       "      <td id=\"T_73286_row3_col36\" class=\"data row3 col36\" >Gga_FieldL</td>\n",
       "      <td id=\"T_73286_row3_col37\" class=\"data row3 col37\" ></td>\n",
       "      <td id=\"T_73286_row3_col38\" class=\"data row3 col38\" ></td>\n",
       "      <td id=\"T_73286_row3_col39\" class=\"data row3 col39\" ></td>\n",
       "      <td id=\"T_73286_row3_col40\" class=\"data row3 col40\" >E17</td>\n",
       "      <td id=\"T_73286_row3_col41\" class=\"data row3 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_73286_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_73286_row4_col0\" class=\"data row4 col0\" >SRX3334821</td>\n",
       "      <td id=\"T_73286_row4_col1\" class=\"data row4 col1\" >SRP121892</td>\n",
       "      <td id=\"T_73286_row4_col2\" class=\"data row4 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_73286_row4_col3\" class=\"data row4 col3\" >SRS2637227</td>\n",
       "      <td id=\"T_73286_row4_col4\" class=\"data row4 col4\" >UBERON:0014757</td>\n",
       "      <td id=\"T_73286_row4_col5\" class=\"data row4 col5\" >hyperpallium apicale</td>\n",
       "      <td id=\"T_73286_row4_col6\" class=\"data row4 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_73286_row4_col7\" class=\"data row4 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_73286_row4_col8\" class=\"data row4 col8\" ></td>\n",
       "      <td id=\"T_73286_row4_col9\" class=\"data row4 col9\" >brain - HA (hyperpallium apicale)</td>\n",
       "      <td id=\"T_73286_row4_col10\" class=\"data row4 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_73286_row4_col11\" class=\"data row4 col11\" >perfect match</td>\n",
       "      <td id=\"T_73286_row4_col12\" class=\"data row4 col12\" >not documented</td>\n",
       "      <td id=\"T_73286_row4_col13\" class=\"data row4 col13\" >perfect match</td>\n",
       "      <td id=\"T_73286_row4_col14\" class=\"data row4 col14\" >NA</td>\n",
       "      <td id=\"T_73286_row4_col15\" class=\"data row4 col15\" >White leghorn</td>\n",
       "      <td id=\"T_73286_row4_col16\" class=\"data row4 col16\" ></td>\n",
       "      <td id=\"T_73286_row4_col17\" class=\"data row4 col17\" >9031</td>\n",
       "      <td id=\"T_73286_row4_col18\" class=\"data row4 col18\" ></td>\n",
       "      <td id=\"T_73286_row4_col19\" class=\"data row4 col19\" ></td>\n",
       "      <td id=\"T_73286_row4_col20\" class=\"data row4 col20\" >polyA</td>\n",
       "      <td id=\"T_73286_row4_col21\" class=\"data row4 col21\" ></td>\n",
       "      <td id=\"T_73286_row4_col22\" class=\"data row4 col22\" ></td>\n",
       "      <td id=\"T_73286_row4_col23\" class=\"data row4 col23\" >Gga_HA</td>\n",
       "      <td id=\"T_73286_row4_col24\" class=\"data row4 col24\" >SAMN07839992</td>\n",
       "      <td id=\"T_73286_row4_col25\" class=\"data row4 col25\" >14</td>\n",
       "      <td id=\"T_73286_row4_col26\" class=\"data row4 col26\" >embryonic day</td>\n",
       "      <td id=\"T_73286_row4_col27\" class=\"data row4 col27\" ></td>\n",
       "      <td id=\"T_73286_row4_col28\" class=\"data row4 col28\" ></td>\n",
       "      <td id=\"T_73286_row4_col29\" class=\"data row4 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_73286_row4_col30\" class=\"data row4 col30\" ></td>\n",
       "      <td id=\"T_73286_row4_col31\" class=\"data row4 col31\" ></td>\n",
       "      <td id=\"T_73286_row4_col32\" class=\"data row4 col32\" ></td>\n",
       "      <td id=\"T_73286_row4_col33\" class=\"data row4 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_73286_row4_col34\" class=\"data row4 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_73286_row4_col35\" class=\"data row4 col35\" ></td>\n",
       "      <td id=\"T_73286_row4_col36\" class=\"data row4 col36\" >Gga_HA</td>\n",
       "      <td id=\"T_73286_row4_col37\" class=\"data row4 col37\" ></td>\n",
       "      <td id=\"T_73286_row4_col38\" class=\"data row4 col38\" ></td>\n",
       "      <td id=\"T_73286_row4_col39\" class=\"data row4 col39\" ></td>\n",
       "      <td id=\"T_73286_row4_col40\" class=\"data row4 col40\" >E18</td>\n",
       "      <td id=\"T_73286_row4_col41\" class=\"data row4 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_73286_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_73286_row5_col0\" class=\"data row5 col0\" >SRX3334820</td>\n",
       "      <td id=\"T_73286_row5_col1\" class=\"data row5 col1\" >SRP121892</td>\n",
       "      <td id=\"T_73286_row5_col2\" class=\"data row5 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_73286_row5_col3\" class=\"data row5 col3\" >SRS2637226</td>\n",
       "      <td id=\"T_73286_row5_col4\" class=\"data row5 col4\" >UBERON:0007349</td>\n",
       "      <td id=\"T_73286_row5_col5\" class=\"data row5 col5\" >mesopallium</td>\n",
       "      <td id=\"T_73286_row5_col6\" class=\"data row5 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_73286_row5_col7\" class=\"data row5 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_73286_row5_col8\" class=\"data row5 col8\" ></td>\n",
       "      <td id=\"T_73286_row5_col9\" class=\"data row5 col9\" >brain - anterior mesopallium</td>\n",
       "      <td id=\"T_73286_row5_col10\" class=\"data row5 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_73286_row5_col11\" class=\"data row5 col11\" >missing child term</td>\n",
       "      <td id=\"T_73286_row5_col12\" class=\"data row5 col12\" >not documented</td>\n",
       "      <td id=\"T_73286_row5_col13\" class=\"data row5 col13\" >perfect match</td>\n",
       "      <td id=\"T_73286_row5_col14\" class=\"data row5 col14\" >NA</td>\n",
       "      <td id=\"T_73286_row5_col15\" class=\"data row5 col15\" >White leghorn</td>\n",
       "      <td id=\"T_73286_row5_col16\" class=\"data row5 col16\" ></td>\n",
       "      <td id=\"T_73286_row5_col17\" class=\"data row5 col17\" >9031</td>\n",
       "      <td id=\"T_73286_row5_col18\" class=\"data row5 col18\" ></td>\n",
       "      <td id=\"T_73286_row5_col19\" class=\"data row5 col19\" ></td>\n",
       "      <td id=\"T_73286_row5_col20\" class=\"data row5 col20\" >polyA</td>\n",
       "      <td id=\"T_73286_row5_col21\" class=\"data row5 col21\" ></td>\n",
       "      <td id=\"T_73286_row5_col22\" class=\"data row5 col22\" ></td>\n",
       "      <td id=\"T_73286_row5_col23\" class=\"data row5 col23\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_73286_row5_col24\" class=\"data row5 col24\" >SAMN07839993</td>\n",
       "      <td id=\"T_73286_row5_col25\" class=\"data row5 col25\" >14</td>\n",
       "      <td id=\"T_73286_row5_col26\" class=\"data row5 col26\" >embryonic day</td>\n",
       "      <td id=\"T_73286_row5_col27\" class=\"data row5 col27\" ></td>\n",
       "      <td id=\"T_73286_row5_col28\" class=\"data row5 col28\" ></td>\n",
       "      <td id=\"T_73286_row5_col29\" class=\"data row5 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_73286_row5_col30\" class=\"data row5 col30\" ></td>\n",
       "      <td id=\"T_73286_row5_col31\" class=\"data row5 col31\" ></td>\n",
       "      <td id=\"T_73286_row5_col32\" class=\"data row5 col32\" ></td>\n",
       "      <td id=\"T_73286_row5_col33\" class=\"data row5 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_73286_row5_col34\" class=\"data row5 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_73286_row5_col35\" class=\"data row5 col35\" ></td>\n",
       "      <td id=\"T_73286_row5_col36\" class=\"data row5 col36\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_73286_row5_col37\" class=\"data row5 col37\" ></td>\n",
       "      <td id=\"T_73286_row5_col38\" class=\"data row5 col38\" ></td>\n",
       "      <td id=\"T_73286_row5_col39\" class=\"data row5 col39\" ></td>\n",
       "      <td id=\"T_73286_row5_col40\" class=\"data row5 col40\" >E19</td>\n",
       "      <td id=\"T_73286_row5_col41\" class=\"data row5 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_73286_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_73286_row6_col0\" class=\"data row6 col0\" >SRX3334819</td>\n",
       "      <td id=\"T_73286_row6_col1\" class=\"data row6 col1\" >SRP121892</td>\n",
       "      <td id=\"T_73286_row6_col2\" class=\"data row6 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_73286_row6_col3\" class=\"data row6 col3\" >SRS2637225</td>\n",
       "      <td id=\"T_73286_row6_col4\" class=\"data row6 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_73286_row6_col5\" class=\"data row6 col5\" >nidopallium</td>\n",
       "      <td id=\"T_73286_row6_col6\" class=\"data row6 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_73286_row6_col7\" class=\"data row6 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_73286_row6_col8\" class=\"data row6 col8\" ></td>\n",
       "      <td id=\"T_73286_row6_col9\" class=\"data row6 col9\" >brain - posterior nidopallium</td>\n",
       "      <td id=\"T_73286_row6_col10\" class=\"data row6 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_73286_row6_col11\" class=\"data row6 col11\" >missing child term</td>\n",
       "      <td id=\"T_73286_row6_col12\" class=\"data row6 col12\" >not documented</td>\n",
       "      <td id=\"T_73286_row6_col13\" class=\"data row6 col13\" >perfect match</td>\n",
       "      <td id=\"T_73286_row6_col14\" class=\"data row6 col14\" >NA</td>\n",
       "      <td id=\"T_73286_row6_col15\" class=\"data row6 col15\" >White leghorn</td>\n",
       "      <td id=\"T_73286_row6_col16\" class=\"data row6 col16\" ></td>\n",
       "      <td id=\"T_73286_row6_col17\" class=\"data row6 col17\" >9031</td>\n",
       "      <td id=\"T_73286_row6_col18\" class=\"data row6 col18\" ></td>\n",
       "      <td id=\"T_73286_row6_col19\" class=\"data row6 col19\" ></td>\n",
       "      <td id=\"T_73286_row6_col20\" class=\"data row6 col20\" >polyA</td>\n",
       "      <td id=\"T_73286_row6_col21\" class=\"data row6 col21\" ></td>\n",
       "      <td id=\"T_73286_row6_col22\" class=\"data row6 col22\" ></td>\n",
       "      <td id=\"T_73286_row6_col23\" class=\"data row6 col23\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_73286_row6_col24\" class=\"data row6 col24\" >SAMN07839994</td>\n",
       "      <td id=\"T_73286_row6_col25\" class=\"data row6 col25\" >14</td>\n",
       "      <td id=\"T_73286_row6_col26\" class=\"data row6 col26\" >embryonic day</td>\n",
       "      <td id=\"T_73286_row6_col27\" class=\"data row6 col27\" ></td>\n",
       "      <td id=\"T_73286_row6_col28\" class=\"data row6 col28\" ></td>\n",
       "      <td id=\"T_73286_row6_col29\" class=\"data row6 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_73286_row6_col30\" class=\"data row6 col30\" ></td>\n",
       "      <td id=\"T_73286_row6_col31\" class=\"data row6 col31\" ></td>\n",
       "      <td id=\"T_73286_row6_col32\" class=\"data row6 col32\" ></td>\n",
       "      <td id=\"T_73286_row6_col33\" class=\"data row6 col33\" >10/10/2024</td>\n",
       "      <td id=\"T_73286_row6_col34\" class=\"data row6 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_73286_row6_col35\" class=\"data row6 col35\" ></td>\n",
       "      <td id=\"T_73286_row6_col36\" class=\"data row6 col36\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_73286_row6_col37\" class=\"data row6 col37\" ></td>\n",
       "      <td id=\"T_73286_row6_col38\" class=\"data row6 col38\" ></td>\n",
       "      <td id=\"T_73286_row6_col39\" class=\"data row6 col39\" ></td>\n",
       "      <td id=\"T_73286_row6_col40\" class=\"data row6 col40\" >E20</td>\n",
       "      <td id=\"T_73286_row6_col41\" class=\"data row6 col41\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library.loc[:,'sampleAge_value'] = '14'\n",
    "library.loc[:,'sampleAge_unit'] = 'embryonic day'\n",
    "\n",
    "# ex. castrated male\n",
    "#library.loc[:,'PATOid'] = ''\n",
    "#library.loc[:,'PATOname'] = ''\n",
    "\n",
    "# ex. castrated, pregnant, pre-smoltification, post-smoltification, laying eggs\n",
    "#library.loc[:,'physiologicalStatus'] = ''\n",
    "\n",
    "# view\n",
    "display_df(library)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### condition"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ex. control, diet, light, reproductive capacity, time post mortem, time post feeding, \n",
    "# exercise details, menstruation, personality, litter size \n",
    "#library.loc[library[\"condition\"] == \"old\", \"condition\"] = \"new\"\n",
    "\n",
    "# view\n",
    "display_df(library)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### annotator id, last modification date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_bb77a\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_bb77a_level0_col0\" class=\"col_heading level0 col0\" >#libraryId</th>\n",
       "      <th id=\"T_bb77a_level0_col1\" class=\"col_heading level0 col1\" >experimentId</th>\n",
       "      <th id=\"T_bb77a_level0_col2\" class=\"col_heading level0 col2\" >platform</th>\n",
       "      <th id=\"T_bb77a_level0_col3\" class=\"col_heading level0 col3\" >SRSId</th>\n",
       "      <th id=\"T_bb77a_level0_col4\" class=\"col_heading level0 col4\" >anatId</th>\n",
       "      <th id=\"T_bb77a_level0_col5\" class=\"col_heading level0 col5\" >anatName</th>\n",
       "      <th id=\"T_bb77a_level0_col6\" class=\"col_heading level0 col6\" >stageId</th>\n",
       "      <th id=\"T_bb77a_level0_col7\" class=\"col_heading level0 col7\" >stageName</th>\n",
       "      <th id=\"T_bb77a_level0_col8\" class=\"col_heading level0 col8\" >url_GSM</th>\n",
       "      <th id=\"T_bb77a_level0_col9\" class=\"col_heading level0 col9\" >infoOrgan</th>\n",
       "      <th id=\"T_bb77a_level0_col10\" class=\"col_heading level0 col10\" >infoStage</th>\n",
       "      <th id=\"T_bb77a_level0_col11\" class=\"col_heading level0 col11\" >anatAnnotationStatus</th>\n",
       "      <th id=\"T_bb77a_level0_col12\" class=\"col_heading level0 col12\" >anatBiologicalStatus</th>\n",
       "      <th id=\"T_bb77a_level0_col13\" class=\"col_heading level0 col13\" >stageAnnotationStatus</th>\n",
       "      <th id=\"T_bb77a_level0_col14\" class=\"col_heading level0 col14\" >sex</th>\n",
       "      <th id=\"T_bb77a_level0_col15\" class=\"col_heading level0 col15\" >strain</th>\n",
       "      <th id=\"T_bb77a_level0_col16\" class=\"col_heading level0 col16\" >genotype</th>\n",
       "      <th id=\"T_bb77a_level0_col17\" class=\"col_heading level0 col17\" >speciesId</th>\n",
       "      <th id=\"T_bb77a_level0_col18\" class=\"col_heading level0 col18\" >protocol</th>\n",
       "      <th id=\"T_bb77a_level0_col19\" class=\"col_heading level0 col19\" >protocolType</th>\n",
       "      <th id=\"T_bb77a_level0_col20\" class=\"col_heading level0 col20\" >RNASelection</th>\n",
       "      <th id=\"T_bb77a_level0_col21\" class=\"col_heading level0 col21\" >globin_reduction</th>\n",
       "      <th id=\"T_bb77a_level0_col22\" class=\"col_heading level0 col22\" >replicate</th>\n",
       "      <th id=\"T_bb77a_level0_col23\" class=\"col_heading level0 col23\" >lib_name</th>\n",
       "      <th id=\"T_bb77a_level0_col24\" class=\"col_heading level0 col24\" >sampleName</th>\n",
       "      <th id=\"T_bb77a_level0_col25\" class=\"col_heading level0 col25\" >sampleAge_value</th>\n",
       "      <th id=\"T_bb77a_level0_col26\" class=\"col_heading level0 col26\" >sampleAge_unit</th>\n",
       "      <th id=\"T_bb77a_level0_col27\" class=\"col_heading level0 col27\" >PATOid</th>\n",
       "      <th id=\"T_bb77a_level0_col28\" class=\"col_heading level0 col28\" >PATOname</th>\n",
       "      <th id=\"T_bb77a_level0_col29\" class=\"col_heading level0 col29\" >comment</th>\n",
       "      <th id=\"T_bb77a_level0_col30\" class=\"col_heading level0 col30\" >condition</th>\n",
       "      <th id=\"T_bb77a_level0_col31\" class=\"col_heading level0 col31\" >physiologicalStatus</th>\n",
       "      <th id=\"T_bb77a_level0_col32\" class=\"col_heading level0 col32\" >annotatorId</th>\n",
       "      <th id=\"T_bb77a_level0_col33\" class=\"col_heading level0 col33\" >lastModificationDate</th>\n",
       "      <th id=\"T_bb77a_level0_col34\" class=\"col_heading level0 col34\" >library_contruction_protocol</th>\n",
       "      <th id=\"T_bb77a_level0_col35\" class=\"col_heading level0 col35\" >source_qc</th>\n",
       "      <th id=\"T_bb77a_level0_col36\" class=\"col_heading level0 col36\" >lib_name_2</th>\n",
       "      <th id=\"T_bb77a_level0_col37\" class=\"col_heading level0 col37\" >lib_name_3</th>\n",
       "      <th id=\"T_bb77a_level0_col38\" class=\"col_heading level0 col38\" >source_name</th>\n",
       "      <th id=\"T_bb77a_level0_col39\" class=\"col_heading level0 col39\" >individual</th>\n",
       "      <th id=\"T_bb77a_level0_col40\" class=\"col_heading level0 col40\" >infoStage_2</th>\n",
       "      <th id=\"T_bb77a_level0_col41\" class=\"col_heading level0 col41\" >infoStage_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_bb77a_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_bb77a_row0_col0\" class=\"data row0 col0\" >SRX3334825</td>\n",
       "      <td id=\"T_bb77a_row0_col1\" class=\"data row0 col1\" >SRP121892</td>\n",
       "      <td id=\"T_bb77a_row0_col2\" class=\"data row0 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_bb77a_row0_col3\" class=\"data row0 col3\" >SRS2637233</td>\n",
       "      <td id=\"T_bb77a_row0_col4\" class=\"data row0 col4\" >UBERON:0007350</td>\n",
       "      <td id=\"T_bb77a_row0_col5\" class=\"data row0 col5\" >arcopallium</td>\n",
       "      <td id=\"T_bb77a_row0_col6\" class=\"data row0 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_bb77a_row0_col7\" class=\"data row0 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_bb77a_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col9\" class=\"data row0 col9\" >brain - arcopallium</td>\n",
       "      <td id=\"T_bb77a_row0_col10\" class=\"data row0 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_bb77a_row0_col11\" class=\"data row0 col11\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row0_col12\" class=\"data row0 col12\" >not documented</td>\n",
       "      <td id=\"T_bb77a_row0_col13\" class=\"data row0 col13\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row0_col14\" class=\"data row0 col14\" >NA</td>\n",
       "      <td id=\"T_bb77a_row0_col15\" class=\"data row0 col15\" >White leghorn</td>\n",
       "      <td id=\"T_bb77a_row0_col16\" class=\"data row0 col16\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col17\" class=\"data row0 col17\" >9031</td>\n",
       "      <td id=\"T_bb77a_row0_col18\" class=\"data row0 col18\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col19\" class=\"data row0 col19\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col20\" class=\"data row0 col20\" >polyA</td>\n",
       "      <td id=\"T_bb77a_row0_col21\" class=\"data row0 col21\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col22\" class=\"data row0 col22\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col23\" class=\"data row0 col23\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_bb77a_row0_col24\" class=\"data row0 col24\" >SAMN07839988</td>\n",
       "      <td id=\"T_bb77a_row0_col25\" class=\"data row0 col25\" >14</td>\n",
       "      <td id=\"T_bb77a_row0_col26\" class=\"data row0 col26\" >embryonic day</td>\n",
       "      <td id=\"T_bb77a_row0_col27\" class=\"data row0 col27\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col28\" class=\"data row0 col28\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col29\" class=\"data row0 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_bb77a_row0_col30\" class=\"data row0 col30\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col31\" class=\"data row0 col31\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col32\" class=\"data row0 col32\" >SAC</td>\n",
       "      <td id=\"T_bb77a_row0_col33\" class=\"data row0 col33\" >2024-10-14</td>\n",
       "      <td id=\"T_bb77a_row0_col34\" class=\"data row0 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_bb77a_row0_col35\" class=\"data row0 col35\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col36\" class=\"data row0 col36\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_bb77a_row0_col37\" class=\"data row0 col37\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col38\" class=\"data row0 col38\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col39\" class=\"data row0 col39\" ></td>\n",
       "      <td id=\"T_bb77a_row0_col40\" class=\"data row0 col40\" >E14</td>\n",
       "      <td id=\"T_bb77a_row0_col41\" class=\"data row0 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_bb77a_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_bb77a_row1_col0\" class=\"data row1 col0\" >SRX3334824</td>\n",
       "      <td id=\"T_bb77a_row1_col1\" class=\"data row1 col1\" >SRP121892</td>\n",
       "      <td id=\"T_bb77a_row1_col2\" class=\"data row1 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_bb77a_row1_col3\" class=\"data row1 col3\" >SRS2637232</td>\n",
       "      <td id=\"T_bb77a_row1_col4\" class=\"data row1 col4\" >UBERON:0000203</td>\n",
       "      <td id=\"T_bb77a_row1_col5\" class=\"data row1 col5\" >pallium</td>\n",
       "      <td id=\"T_bb77a_row1_col6\" class=\"data row1 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_bb77a_row1_col7\" class=\"data row1 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_bb77a_row1_col8\" class=\"data row1 col8\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col9\" class=\"data row1 col9\" >brain - basorostralis</td>\n",
       "      <td id=\"T_bb77a_row1_col10\" class=\"data row1 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_bb77a_row1_col11\" class=\"data row1 col11\" >missing child term</td>\n",
       "      <td id=\"T_bb77a_row1_col12\" class=\"data row1 col12\" >not documented</td>\n",
       "      <td id=\"T_bb77a_row1_col13\" class=\"data row1 col13\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row1_col14\" class=\"data row1 col14\" >NA</td>\n",
       "      <td id=\"T_bb77a_row1_col15\" class=\"data row1 col15\" >White leghorn</td>\n",
       "      <td id=\"T_bb77a_row1_col16\" class=\"data row1 col16\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col17\" class=\"data row1 col17\" >9031</td>\n",
       "      <td id=\"T_bb77a_row1_col18\" class=\"data row1 col18\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col19\" class=\"data row1 col19\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col20\" class=\"data row1 col20\" >polyA</td>\n",
       "      <td id=\"T_bb77a_row1_col21\" class=\"data row1 col21\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col22\" class=\"data row1 col22\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col23\" class=\"data row1 col23\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_bb77a_row1_col24\" class=\"data row1 col24\" >SAMN07839989</td>\n",
       "      <td id=\"T_bb77a_row1_col25\" class=\"data row1 col25\" >14</td>\n",
       "      <td id=\"T_bb77a_row1_col26\" class=\"data row1 col26\" >embryonic day</td>\n",
       "      <td id=\"T_bb77a_row1_col27\" class=\"data row1 col27\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col28\" class=\"data row1 col28\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col29\" class=\"data row1 col29\" >nucleus basorostralis of the pallium (PMID: 15116397), annotated as pallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_bb77a_row1_col30\" class=\"data row1 col30\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col31\" class=\"data row1 col31\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col32\" class=\"data row1 col32\" >SAC</td>\n",
       "      <td id=\"T_bb77a_row1_col33\" class=\"data row1 col33\" >2024-10-14</td>\n",
       "      <td id=\"T_bb77a_row1_col34\" class=\"data row1 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_bb77a_row1_col35\" class=\"data row1 col35\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col36\" class=\"data row1 col36\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_bb77a_row1_col37\" class=\"data row1 col37\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col38\" class=\"data row1 col38\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col39\" class=\"data row1 col39\" ></td>\n",
       "      <td id=\"T_bb77a_row1_col40\" class=\"data row1 col40\" >E15</td>\n",
       "      <td id=\"T_bb77a_row1_col41\" class=\"data row1 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_bb77a_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_bb77a_row2_col0\" class=\"data row2 col0\" >SRX3334823</td>\n",
       "      <td id=\"T_bb77a_row2_col1\" class=\"data row2 col1\" >SRP121892</td>\n",
       "      <td id=\"T_bb77a_row2_col2\" class=\"data row2 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_bb77a_row2_col3\" class=\"data row2 col3\" >SRS2637229</td>\n",
       "      <td id=\"T_bb77a_row2_col4\" class=\"data row2 col4\" >UBERON:0014759</td>\n",
       "      <td id=\"T_bb77a_row2_col5\" class=\"data row2 col5\" >entopallium</td>\n",
       "      <td id=\"T_bb77a_row2_col6\" class=\"data row2 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_bb77a_row2_col7\" class=\"data row2 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_bb77a_row2_col8\" class=\"data row2 col8\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col9\" class=\"data row2 col9\" >brain - entopallium</td>\n",
       "      <td id=\"T_bb77a_row2_col10\" class=\"data row2 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_bb77a_row2_col11\" class=\"data row2 col11\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row2_col12\" class=\"data row2 col12\" >not documented</td>\n",
       "      <td id=\"T_bb77a_row2_col13\" class=\"data row2 col13\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row2_col14\" class=\"data row2 col14\" >NA</td>\n",
       "      <td id=\"T_bb77a_row2_col15\" class=\"data row2 col15\" >White leghorn</td>\n",
       "      <td id=\"T_bb77a_row2_col16\" class=\"data row2 col16\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col17\" class=\"data row2 col17\" >9031</td>\n",
       "      <td id=\"T_bb77a_row2_col18\" class=\"data row2 col18\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col19\" class=\"data row2 col19\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col20\" class=\"data row2 col20\" >polyA</td>\n",
       "      <td id=\"T_bb77a_row2_col21\" class=\"data row2 col21\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col22\" class=\"data row2 col22\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col23\" class=\"data row2 col23\" >Gga_entopallium</td>\n",
       "      <td id=\"T_bb77a_row2_col24\" class=\"data row2 col24\" >SAMN07839990</td>\n",
       "      <td id=\"T_bb77a_row2_col25\" class=\"data row2 col25\" >14</td>\n",
       "      <td id=\"T_bb77a_row2_col26\" class=\"data row2 col26\" >embryonic day</td>\n",
       "      <td id=\"T_bb77a_row2_col27\" class=\"data row2 col27\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col28\" class=\"data row2 col28\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col29\" class=\"data row2 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_bb77a_row2_col30\" class=\"data row2 col30\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col31\" class=\"data row2 col31\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col32\" class=\"data row2 col32\" >SAC</td>\n",
       "      <td id=\"T_bb77a_row2_col33\" class=\"data row2 col33\" >2024-10-14</td>\n",
       "      <td id=\"T_bb77a_row2_col34\" class=\"data row2 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_bb77a_row2_col35\" class=\"data row2 col35\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col36\" class=\"data row2 col36\" >Gga_entopallium</td>\n",
       "      <td id=\"T_bb77a_row2_col37\" class=\"data row2 col37\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col38\" class=\"data row2 col38\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col39\" class=\"data row2 col39\" ></td>\n",
       "      <td id=\"T_bb77a_row2_col40\" class=\"data row2 col40\" >E16</td>\n",
       "      <td id=\"T_bb77a_row2_col41\" class=\"data row2 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_bb77a_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_bb77a_row3_col0\" class=\"data row3 col0\" >SRX3334822</td>\n",
       "      <td id=\"T_bb77a_row3_col1\" class=\"data row3 col1\" >SRP121892</td>\n",
       "      <td id=\"T_bb77a_row3_col2\" class=\"data row3 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_bb77a_row3_col3\" class=\"data row3 col3\" >SRS2637228</td>\n",
       "      <td id=\"T_bb77a_row3_col4\" class=\"data row3 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_bb77a_row3_col5\" class=\"data row3 col5\" >nidopallium</td>\n",
       "      <td id=\"T_bb77a_row3_col6\" class=\"data row3 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_bb77a_row3_col7\" class=\"data row3 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_bb77a_row3_col8\" class=\"data row3 col8\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col9\" class=\"data row3 col9\" >brain - Field L</td>\n",
       "      <td id=\"T_bb77a_row3_col10\" class=\"data row3 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_bb77a_row3_col11\" class=\"data row3 col11\" >missing child term</td>\n",
       "      <td id=\"T_bb77a_row3_col12\" class=\"data row3 col12\" >not documented</td>\n",
       "      <td id=\"T_bb77a_row3_col13\" class=\"data row3 col13\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row3_col14\" class=\"data row3 col14\" >NA</td>\n",
       "      <td id=\"T_bb77a_row3_col15\" class=\"data row3 col15\" >White leghorn</td>\n",
       "      <td id=\"T_bb77a_row3_col16\" class=\"data row3 col16\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col17\" class=\"data row3 col17\" >9031</td>\n",
       "      <td id=\"T_bb77a_row3_col18\" class=\"data row3 col18\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col19\" class=\"data row3 col19\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col20\" class=\"data row3 col20\" >polyA</td>\n",
       "      <td id=\"T_bb77a_row3_col21\" class=\"data row3 col21\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col22\" class=\"data row3 col22\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col23\" class=\"data row3 col23\" >Gga_FieldL</td>\n",
       "      <td id=\"T_bb77a_row3_col24\" class=\"data row3 col24\" >SAMN07839991</td>\n",
       "      <td id=\"T_bb77a_row3_col25\" class=\"data row3 col25\" >14</td>\n",
       "      <td id=\"T_bb77a_row3_col26\" class=\"data row3 col26\" >embryonic day</td>\n",
       "      <td id=\"T_bb77a_row3_col27\" class=\"data row3 col27\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col28\" class=\"data row3 col28\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col29\" class=\"data row3 col29\" >Field L is the nidopallial region containing the primary auditory thalamorecipient zone (PMID: 15116397), annotated as nidopallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_bb77a_row3_col30\" class=\"data row3 col30\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col31\" class=\"data row3 col31\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col32\" class=\"data row3 col32\" >SAC</td>\n",
       "      <td id=\"T_bb77a_row3_col33\" class=\"data row3 col33\" >2024-10-14</td>\n",
       "      <td id=\"T_bb77a_row3_col34\" class=\"data row3 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_bb77a_row3_col35\" class=\"data row3 col35\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col36\" class=\"data row3 col36\" >Gga_FieldL</td>\n",
       "      <td id=\"T_bb77a_row3_col37\" class=\"data row3 col37\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col38\" class=\"data row3 col38\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col39\" class=\"data row3 col39\" ></td>\n",
       "      <td id=\"T_bb77a_row3_col40\" class=\"data row3 col40\" >E17</td>\n",
       "      <td id=\"T_bb77a_row3_col41\" class=\"data row3 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_bb77a_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_bb77a_row4_col0\" class=\"data row4 col0\" >SRX3334821</td>\n",
       "      <td id=\"T_bb77a_row4_col1\" class=\"data row4 col1\" >SRP121892</td>\n",
       "      <td id=\"T_bb77a_row4_col2\" class=\"data row4 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_bb77a_row4_col3\" class=\"data row4 col3\" >SRS2637227</td>\n",
       "      <td id=\"T_bb77a_row4_col4\" class=\"data row4 col4\" >UBERON:0014757</td>\n",
       "      <td id=\"T_bb77a_row4_col5\" class=\"data row4 col5\" >hyperpallium apicale</td>\n",
       "      <td id=\"T_bb77a_row4_col6\" class=\"data row4 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_bb77a_row4_col7\" class=\"data row4 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_bb77a_row4_col8\" class=\"data row4 col8\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col9\" class=\"data row4 col9\" >brain - HA (hyperpallium apicale)</td>\n",
       "      <td id=\"T_bb77a_row4_col10\" class=\"data row4 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_bb77a_row4_col11\" class=\"data row4 col11\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row4_col12\" class=\"data row4 col12\" >not documented</td>\n",
       "      <td id=\"T_bb77a_row4_col13\" class=\"data row4 col13\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row4_col14\" class=\"data row4 col14\" >NA</td>\n",
       "      <td id=\"T_bb77a_row4_col15\" class=\"data row4 col15\" >White leghorn</td>\n",
       "      <td id=\"T_bb77a_row4_col16\" class=\"data row4 col16\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col17\" class=\"data row4 col17\" >9031</td>\n",
       "      <td id=\"T_bb77a_row4_col18\" class=\"data row4 col18\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col19\" class=\"data row4 col19\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col20\" class=\"data row4 col20\" >polyA</td>\n",
       "      <td id=\"T_bb77a_row4_col21\" class=\"data row4 col21\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col22\" class=\"data row4 col22\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col23\" class=\"data row4 col23\" >Gga_HA</td>\n",
       "      <td id=\"T_bb77a_row4_col24\" class=\"data row4 col24\" >SAMN07839992</td>\n",
       "      <td id=\"T_bb77a_row4_col25\" class=\"data row4 col25\" >14</td>\n",
       "      <td id=\"T_bb77a_row4_col26\" class=\"data row4 col26\" >embryonic day</td>\n",
       "      <td id=\"T_bb77a_row4_col27\" class=\"data row4 col27\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col28\" class=\"data row4 col28\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col29\" class=\"data row4 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_bb77a_row4_col30\" class=\"data row4 col30\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col31\" class=\"data row4 col31\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col32\" class=\"data row4 col32\" >SAC</td>\n",
       "      <td id=\"T_bb77a_row4_col33\" class=\"data row4 col33\" >2024-10-14</td>\n",
       "      <td id=\"T_bb77a_row4_col34\" class=\"data row4 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_bb77a_row4_col35\" class=\"data row4 col35\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col36\" class=\"data row4 col36\" >Gga_HA</td>\n",
       "      <td id=\"T_bb77a_row4_col37\" class=\"data row4 col37\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col38\" class=\"data row4 col38\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col39\" class=\"data row4 col39\" ></td>\n",
       "      <td id=\"T_bb77a_row4_col40\" class=\"data row4 col40\" >E18</td>\n",
       "      <td id=\"T_bb77a_row4_col41\" class=\"data row4 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_bb77a_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_bb77a_row5_col0\" class=\"data row5 col0\" >SRX3334820</td>\n",
       "      <td id=\"T_bb77a_row5_col1\" class=\"data row5 col1\" >SRP121892</td>\n",
       "      <td id=\"T_bb77a_row5_col2\" class=\"data row5 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_bb77a_row5_col3\" class=\"data row5 col3\" >SRS2637226</td>\n",
       "      <td id=\"T_bb77a_row5_col4\" class=\"data row5 col4\" >UBERON:0007349</td>\n",
       "      <td id=\"T_bb77a_row5_col5\" class=\"data row5 col5\" >mesopallium</td>\n",
       "      <td id=\"T_bb77a_row5_col6\" class=\"data row5 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_bb77a_row5_col7\" class=\"data row5 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_bb77a_row5_col8\" class=\"data row5 col8\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col9\" class=\"data row5 col9\" >brain - anterior mesopallium</td>\n",
       "      <td id=\"T_bb77a_row5_col10\" class=\"data row5 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_bb77a_row5_col11\" class=\"data row5 col11\" >missing child term</td>\n",
       "      <td id=\"T_bb77a_row5_col12\" class=\"data row5 col12\" >not documented</td>\n",
       "      <td id=\"T_bb77a_row5_col13\" class=\"data row5 col13\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row5_col14\" class=\"data row5 col14\" >NA</td>\n",
       "      <td id=\"T_bb77a_row5_col15\" class=\"data row5 col15\" >White leghorn</td>\n",
       "      <td id=\"T_bb77a_row5_col16\" class=\"data row5 col16\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col17\" class=\"data row5 col17\" >9031</td>\n",
       "      <td id=\"T_bb77a_row5_col18\" class=\"data row5 col18\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col19\" class=\"data row5 col19\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col20\" class=\"data row5 col20\" >polyA</td>\n",
       "      <td id=\"T_bb77a_row5_col21\" class=\"data row5 col21\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col22\" class=\"data row5 col22\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col23\" class=\"data row5 col23\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_bb77a_row5_col24\" class=\"data row5 col24\" >SAMN07839993</td>\n",
       "      <td id=\"T_bb77a_row5_col25\" class=\"data row5 col25\" >14</td>\n",
       "      <td id=\"T_bb77a_row5_col26\" class=\"data row5 col26\" >embryonic day</td>\n",
       "      <td id=\"T_bb77a_row5_col27\" class=\"data row5 col27\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col28\" class=\"data row5 col28\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col29\" class=\"data row5 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_bb77a_row5_col30\" class=\"data row5 col30\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col31\" class=\"data row5 col31\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col32\" class=\"data row5 col32\" >SAC</td>\n",
       "      <td id=\"T_bb77a_row5_col33\" class=\"data row5 col33\" >2024-10-14</td>\n",
       "      <td id=\"T_bb77a_row5_col34\" class=\"data row5 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_bb77a_row5_col35\" class=\"data row5 col35\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col36\" class=\"data row5 col36\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_bb77a_row5_col37\" class=\"data row5 col37\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col38\" class=\"data row5 col38\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col39\" class=\"data row5 col39\" ></td>\n",
       "      <td id=\"T_bb77a_row5_col40\" class=\"data row5 col40\" >E19</td>\n",
       "      <td id=\"T_bb77a_row5_col41\" class=\"data row5 col41\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_bb77a_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_bb77a_row6_col0\" class=\"data row6 col0\" >SRX3334819</td>\n",
       "      <td id=\"T_bb77a_row6_col1\" class=\"data row6 col1\" >SRP121892</td>\n",
       "      <td id=\"T_bb77a_row6_col2\" class=\"data row6 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_bb77a_row6_col3\" class=\"data row6 col3\" >SRS2637225</td>\n",
       "      <td id=\"T_bb77a_row6_col4\" class=\"data row6 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_bb77a_row6_col5\" class=\"data row6 col5\" >nidopallium</td>\n",
       "      <td id=\"T_bb77a_row6_col6\" class=\"data row6 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_bb77a_row6_col7\" class=\"data row6 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_bb77a_row6_col8\" class=\"data row6 col8\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col9\" class=\"data row6 col9\" >brain - posterior nidopallium</td>\n",
       "      <td id=\"T_bb77a_row6_col10\" class=\"data row6 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_bb77a_row6_col11\" class=\"data row6 col11\" >missing child term</td>\n",
       "      <td id=\"T_bb77a_row6_col12\" class=\"data row6 col12\" >not documented</td>\n",
       "      <td id=\"T_bb77a_row6_col13\" class=\"data row6 col13\" >perfect match</td>\n",
       "      <td id=\"T_bb77a_row6_col14\" class=\"data row6 col14\" >NA</td>\n",
       "      <td id=\"T_bb77a_row6_col15\" class=\"data row6 col15\" >White leghorn</td>\n",
       "      <td id=\"T_bb77a_row6_col16\" class=\"data row6 col16\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col17\" class=\"data row6 col17\" >9031</td>\n",
       "      <td id=\"T_bb77a_row6_col18\" class=\"data row6 col18\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col19\" class=\"data row6 col19\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col20\" class=\"data row6 col20\" >polyA</td>\n",
       "      <td id=\"T_bb77a_row6_col21\" class=\"data row6 col21\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col22\" class=\"data row6 col22\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col23\" class=\"data row6 col23\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_bb77a_row6_col24\" class=\"data row6 col24\" >SAMN07839994</td>\n",
       "      <td id=\"T_bb77a_row6_col25\" class=\"data row6 col25\" >14</td>\n",
       "      <td id=\"T_bb77a_row6_col26\" class=\"data row6 col26\" >embryonic day</td>\n",
       "      <td id=\"T_bb77a_row6_col27\" class=\"data row6 col27\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col28\" class=\"data row6 col28\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col29\" class=\"data row6 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_bb77a_row6_col30\" class=\"data row6 col30\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col31\" class=\"data row6 col31\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col32\" class=\"data row6 col32\" >SAC</td>\n",
       "      <td id=\"T_bb77a_row6_col33\" class=\"data row6 col33\" >2024-10-14</td>\n",
       "      <td id=\"T_bb77a_row6_col34\" class=\"data row6 col34\" >RNA integrity was analyzed with a Bioanalyzer 2100; only samples with clean rRNA peaks and little to no degradation were used. Total RNA was polyA selected and directionally sequenced at the University of Chicago Genomics Facility on an Illumina HiSeq2000 per manufacturer's instructions, generating 100bp paired-end reads with an insert size of 300bp</td>\n",
       "      <td id=\"T_bb77a_row6_col35\" class=\"data row6 col35\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col36\" class=\"data row6 col36\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_bb77a_row6_col37\" class=\"data row6 col37\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col38\" class=\"data row6 col38\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col39\" class=\"data row6 col39\" ></td>\n",
       "      <td id=\"T_bb77a_row6_col40\" class=\"data row6 col40\" >E20</td>\n",
       "      <td id=\"T_bb77a_row6_col41\" class=\"data row6 col41\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library.loc[:,'annotatorId'] = 'SAC'\n",
    "library.loc[:,'lastModificationDate'] = '2024-10-14'\n",
    "\n",
    "# view\n",
    "display_df(library)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### comments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#library.loc[:,'comment'] = 'PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### save complete file with correct columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_a04f2\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_a04f2_level0_col0\" class=\"col_heading level0 col0\" >#libraryId</th>\n",
       "      <th id=\"T_a04f2_level0_col1\" class=\"col_heading level0 col1\" >experimentId</th>\n",
       "      <th id=\"T_a04f2_level0_col2\" class=\"col_heading level0 col2\" >platform</th>\n",
       "      <th id=\"T_a04f2_level0_col3\" class=\"col_heading level0 col3\" >SRSId</th>\n",
       "      <th id=\"T_a04f2_level0_col4\" class=\"col_heading level0 col4\" >anatId</th>\n",
       "      <th id=\"T_a04f2_level0_col5\" class=\"col_heading level0 col5\" >anatName</th>\n",
       "      <th id=\"T_a04f2_level0_col6\" class=\"col_heading level0 col6\" >stageId</th>\n",
       "      <th id=\"T_a04f2_level0_col7\" class=\"col_heading level0 col7\" >stageName</th>\n",
       "      <th id=\"T_a04f2_level0_col8\" class=\"col_heading level0 col8\" >url_GSM</th>\n",
       "      <th id=\"T_a04f2_level0_col9\" class=\"col_heading level0 col9\" >infoOrgan</th>\n",
       "      <th id=\"T_a04f2_level0_col10\" class=\"col_heading level0 col10\" >infoStage</th>\n",
       "      <th id=\"T_a04f2_level0_col11\" class=\"col_heading level0 col11\" >anatAnnotationStatus</th>\n",
       "      <th id=\"T_a04f2_level0_col12\" class=\"col_heading level0 col12\" >anatBiologicalStatus</th>\n",
       "      <th id=\"T_a04f2_level0_col13\" class=\"col_heading level0 col13\" >stageAnnotationStatus</th>\n",
       "      <th id=\"T_a04f2_level0_col14\" class=\"col_heading level0 col14\" >sex</th>\n",
       "      <th id=\"T_a04f2_level0_col15\" class=\"col_heading level0 col15\" >strain</th>\n",
       "      <th id=\"T_a04f2_level0_col16\" class=\"col_heading level0 col16\" >genotype</th>\n",
       "      <th id=\"T_a04f2_level0_col17\" class=\"col_heading level0 col17\" >speciesId</th>\n",
       "      <th id=\"T_a04f2_level0_col18\" class=\"col_heading level0 col18\" >protocol</th>\n",
       "      <th id=\"T_a04f2_level0_col19\" class=\"col_heading level0 col19\" >protocolType</th>\n",
       "      <th id=\"T_a04f2_level0_col20\" class=\"col_heading level0 col20\" >RNASelection</th>\n",
       "      <th id=\"T_a04f2_level0_col21\" class=\"col_heading level0 col21\" >globin_reduction</th>\n",
       "      <th id=\"T_a04f2_level0_col22\" class=\"col_heading level0 col22\" >replicate</th>\n",
       "      <th id=\"T_a04f2_level0_col23\" class=\"col_heading level0 col23\" >lib_name</th>\n",
       "      <th id=\"T_a04f2_level0_col24\" class=\"col_heading level0 col24\" >sampleName</th>\n",
       "      <th id=\"T_a04f2_level0_col25\" class=\"col_heading level0 col25\" >sampleAge_value</th>\n",
       "      <th id=\"T_a04f2_level0_col26\" class=\"col_heading level0 col26\" >sampleAge_unit</th>\n",
       "      <th id=\"T_a04f2_level0_col27\" class=\"col_heading level0 col27\" >PATOid</th>\n",
       "      <th id=\"T_a04f2_level0_col28\" class=\"col_heading level0 col28\" >PATOname</th>\n",
       "      <th id=\"T_a04f2_level0_col29\" class=\"col_heading level0 col29\" >comment</th>\n",
       "      <th id=\"T_a04f2_level0_col30\" class=\"col_heading level0 col30\" >condition</th>\n",
       "      <th id=\"T_a04f2_level0_col31\" class=\"col_heading level0 col31\" >physiologicalStatus</th>\n",
       "      <th id=\"T_a04f2_level0_col32\" class=\"col_heading level0 col32\" >annotatorId</th>\n",
       "      <th id=\"T_a04f2_level0_col33\" class=\"col_heading level0 col33\" >lastModificationDate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_a04f2_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_a04f2_row0_col0\" class=\"data row0 col0\" >SRX3334825</td>\n",
       "      <td id=\"T_a04f2_row0_col1\" class=\"data row0 col1\" >SRP121892</td>\n",
       "      <td id=\"T_a04f2_row0_col2\" class=\"data row0 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_a04f2_row0_col3\" class=\"data row0 col3\" >SRS2637233</td>\n",
       "      <td id=\"T_a04f2_row0_col4\" class=\"data row0 col4\" >UBERON:0007350</td>\n",
       "      <td id=\"T_a04f2_row0_col5\" class=\"data row0 col5\" >arcopallium</td>\n",
       "      <td id=\"T_a04f2_row0_col6\" class=\"data row0 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_a04f2_row0_col7\" class=\"data row0 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_a04f2_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col9\" class=\"data row0 col9\" >brain - arcopallium</td>\n",
       "      <td id=\"T_a04f2_row0_col10\" class=\"data row0 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_a04f2_row0_col11\" class=\"data row0 col11\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row0_col12\" class=\"data row0 col12\" >not documented</td>\n",
       "      <td id=\"T_a04f2_row0_col13\" class=\"data row0 col13\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row0_col14\" class=\"data row0 col14\" >NA</td>\n",
       "      <td id=\"T_a04f2_row0_col15\" class=\"data row0 col15\" >White leghorn</td>\n",
       "      <td id=\"T_a04f2_row0_col16\" class=\"data row0 col16\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col17\" class=\"data row0 col17\" >9031</td>\n",
       "      <td id=\"T_a04f2_row0_col18\" class=\"data row0 col18\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col19\" class=\"data row0 col19\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col20\" class=\"data row0 col20\" >polyA</td>\n",
       "      <td id=\"T_a04f2_row0_col21\" class=\"data row0 col21\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col22\" class=\"data row0 col22\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col23\" class=\"data row0 col23\" >Gga_arcopallium</td>\n",
       "      <td id=\"T_a04f2_row0_col24\" class=\"data row0 col24\" >SAMN07839988</td>\n",
       "      <td id=\"T_a04f2_row0_col25\" class=\"data row0 col25\" >14</td>\n",
       "      <td id=\"T_a04f2_row0_col26\" class=\"data row0 col26\" >embryonic day</td>\n",
       "      <td id=\"T_a04f2_row0_col27\" class=\"data row0 col27\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col28\" class=\"data row0 col28\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col29\" class=\"data row0 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_a04f2_row0_col30\" class=\"data row0 col30\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col31\" class=\"data row0 col31\" ></td>\n",
       "      <td id=\"T_a04f2_row0_col32\" class=\"data row0 col32\" >SAC</td>\n",
       "      <td id=\"T_a04f2_row0_col33\" class=\"data row0 col33\" >2024-10-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a04f2_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "      <td id=\"T_a04f2_row1_col0\" class=\"data row1 col0\" >SRX3334824</td>\n",
       "      <td id=\"T_a04f2_row1_col1\" class=\"data row1 col1\" >SRP121892</td>\n",
       "      <td id=\"T_a04f2_row1_col2\" class=\"data row1 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_a04f2_row1_col3\" class=\"data row1 col3\" >SRS2637232</td>\n",
       "      <td id=\"T_a04f2_row1_col4\" class=\"data row1 col4\" >UBERON:0000203</td>\n",
       "      <td id=\"T_a04f2_row1_col5\" class=\"data row1 col5\" >pallium</td>\n",
       "      <td id=\"T_a04f2_row1_col6\" class=\"data row1 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_a04f2_row1_col7\" class=\"data row1 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_a04f2_row1_col8\" class=\"data row1 col8\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col9\" class=\"data row1 col9\" >brain - basorostralis</td>\n",
       "      <td id=\"T_a04f2_row1_col10\" class=\"data row1 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_a04f2_row1_col11\" class=\"data row1 col11\" >missing child term</td>\n",
       "      <td id=\"T_a04f2_row1_col12\" class=\"data row1 col12\" >not documented</td>\n",
       "      <td id=\"T_a04f2_row1_col13\" class=\"data row1 col13\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row1_col14\" class=\"data row1 col14\" >NA</td>\n",
       "      <td id=\"T_a04f2_row1_col15\" class=\"data row1 col15\" >White leghorn</td>\n",
       "      <td id=\"T_a04f2_row1_col16\" class=\"data row1 col16\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col17\" class=\"data row1 col17\" >9031</td>\n",
       "      <td id=\"T_a04f2_row1_col18\" class=\"data row1 col18\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col19\" class=\"data row1 col19\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col20\" class=\"data row1 col20\" >polyA</td>\n",
       "      <td id=\"T_a04f2_row1_col21\" class=\"data row1 col21\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col22\" class=\"data row1 col22\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col23\" class=\"data row1 col23\" >Gga_basorostralis</td>\n",
       "      <td id=\"T_a04f2_row1_col24\" class=\"data row1 col24\" >SAMN07839989</td>\n",
       "      <td id=\"T_a04f2_row1_col25\" class=\"data row1 col25\" >14</td>\n",
       "      <td id=\"T_a04f2_row1_col26\" class=\"data row1 col26\" >embryonic day</td>\n",
       "      <td id=\"T_a04f2_row1_col27\" class=\"data row1 col27\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col28\" class=\"data row1 col28\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col29\" class=\"data row1 col29\" >nucleus basorostralis of the pallium (PMID: 15116397), annotated as pallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_a04f2_row1_col30\" class=\"data row1 col30\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col31\" class=\"data row1 col31\" ></td>\n",
       "      <td id=\"T_a04f2_row1_col32\" class=\"data row1 col32\" >SAC</td>\n",
       "      <td id=\"T_a04f2_row1_col33\" class=\"data row1 col33\" >2024-10-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a04f2_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "      <td id=\"T_a04f2_row2_col0\" class=\"data row2 col0\" >SRX3334823</td>\n",
       "      <td id=\"T_a04f2_row2_col1\" class=\"data row2 col1\" >SRP121892</td>\n",
       "      <td id=\"T_a04f2_row2_col2\" class=\"data row2 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_a04f2_row2_col3\" class=\"data row2 col3\" >SRS2637229</td>\n",
       "      <td id=\"T_a04f2_row2_col4\" class=\"data row2 col4\" >UBERON:0014759</td>\n",
       "      <td id=\"T_a04f2_row2_col5\" class=\"data row2 col5\" >entopallium</td>\n",
       "      <td id=\"T_a04f2_row2_col6\" class=\"data row2 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_a04f2_row2_col7\" class=\"data row2 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_a04f2_row2_col8\" class=\"data row2 col8\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col9\" class=\"data row2 col9\" >brain - entopallium</td>\n",
       "      <td id=\"T_a04f2_row2_col10\" class=\"data row2 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_a04f2_row2_col11\" class=\"data row2 col11\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row2_col12\" class=\"data row2 col12\" >not documented</td>\n",
       "      <td id=\"T_a04f2_row2_col13\" class=\"data row2 col13\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row2_col14\" class=\"data row2 col14\" >NA</td>\n",
       "      <td id=\"T_a04f2_row2_col15\" class=\"data row2 col15\" >White leghorn</td>\n",
       "      <td id=\"T_a04f2_row2_col16\" class=\"data row2 col16\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col17\" class=\"data row2 col17\" >9031</td>\n",
       "      <td id=\"T_a04f2_row2_col18\" class=\"data row2 col18\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col19\" class=\"data row2 col19\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col20\" class=\"data row2 col20\" >polyA</td>\n",
       "      <td id=\"T_a04f2_row2_col21\" class=\"data row2 col21\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col22\" class=\"data row2 col22\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col23\" class=\"data row2 col23\" >Gga_entopallium</td>\n",
       "      <td id=\"T_a04f2_row2_col24\" class=\"data row2 col24\" >SAMN07839990</td>\n",
       "      <td id=\"T_a04f2_row2_col25\" class=\"data row2 col25\" >14</td>\n",
       "      <td id=\"T_a04f2_row2_col26\" class=\"data row2 col26\" >embryonic day</td>\n",
       "      <td id=\"T_a04f2_row2_col27\" class=\"data row2 col27\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col28\" class=\"data row2 col28\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col29\" class=\"data row2 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_a04f2_row2_col30\" class=\"data row2 col30\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col31\" class=\"data row2 col31\" ></td>\n",
       "      <td id=\"T_a04f2_row2_col32\" class=\"data row2 col32\" >SAC</td>\n",
       "      <td id=\"T_a04f2_row2_col33\" class=\"data row2 col33\" >2024-10-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a04f2_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "      <td id=\"T_a04f2_row3_col0\" class=\"data row3 col0\" >SRX3334822</td>\n",
       "      <td id=\"T_a04f2_row3_col1\" class=\"data row3 col1\" >SRP121892</td>\n",
       "      <td id=\"T_a04f2_row3_col2\" class=\"data row3 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_a04f2_row3_col3\" class=\"data row3 col3\" >SRS2637228</td>\n",
       "      <td id=\"T_a04f2_row3_col4\" class=\"data row3 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_a04f2_row3_col5\" class=\"data row3 col5\" >nidopallium</td>\n",
       "      <td id=\"T_a04f2_row3_col6\" class=\"data row3 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_a04f2_row3_col7\" class=\"data row3 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_a04f2_row3_col8\" class=\"data row3 col8\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col9\" class=\"data row3 col9\" >brain - Field L</td>\n",
       "      <td id=\"T_a04f2_row3_col10\" class=\"data row3 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_a04f2_row3_col11\" class=\"data row3 col11\" >missing child term</td>\n",
       "      <td id=\"T_a04f2_row3_col12\" class=\"data row3 col12\" >not documented</td>\n",
       "      <td id=\"T_a04f2_row3_col13\" class=\"data row3 col13\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row3_col14\" class=\"data row3 col14\" >NA</td>\n",
       "      <td id=\"T_a04f2_row3_col15\" class=\"data row3 col15\" >White leghorn</td>\n",
       "      <td id=\"T_a04f2_row3_col16\" class=\"data row3 col16\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col17\" class=\"data row3 col17\" >9031</td>\n",
       "      <td id=\"T_a04f2_row3_col18\" class=\"data row3 col18\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col19\" class=\"data row3 col19\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col20\" class=\"data row3 col20\" >polyA</td>\n",
       "      <td id=\"T_a04f2_row3_col21\" class=\"data row3 col21\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col22\" class=\"data row3 col22\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col23\" class=\"data row3 col23\" >Gga_FieldL</td>\n",
       "      <td id=\"T_a04f2_row3_col24\" class=\"data row3 col24\" >SAMN07839991</td>\n",
       "      <td id=\"T_a04f2_row3_col25\" class=\"data row3 col25\" >14</td>\n",
       "      <td id=\"T_a04f2_row3_col26\" class=\"data row3 col26\" >embryonic day</td>\n",
       "      <td id=\"T_a04f2_row3_col27\" class=\"data row3 col27\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col28\" class=\"data row3 col28\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col29\" class=\"data row3 col29\" >Field L is the nidopallial region containing the primary auditory thalamorecipient zone (PMID: 15116397), annotated as nidopallium since this doesn't appear to be in uberon and homologous structure is not known; PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_a04f2_row3_col30\" class=\"data row3 col30\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col31\" class=\"data row3 col31\" ></td>\n",
       "      <td id=\"T_a04f2_row3_col32\" class=\"data row3 col32\" >SAC</td>\n",
       "      <td id=\"T_a04f2_row3_col33\" class=\"data row3 col33\" >2024-10-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a04f2_level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "      <td id=\"T_a04f2_row4_col0\" class=\"data row4 col0\" >SRX3334821</td>\n",
       "      <td id=\"T_a04f2_row4_col1\" class=\"data row4 col1\" >SRP121892</td>\n",
       "      <td id=\"T_a04f2_row4_col2\" class=\"data row4 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_a04f2_row4_col3\" class=\"data row4 col3\" >SRS2637227</td>\n",
       "      <td id=\"T_a04f2_row4_col4\" class=\"data row4 col4\" >UBERON:0014757</td>\n",
       "      <td id=\"T_a04f2_row4_col5\" class=\"data row4 col5\" >hyperpallium apicale</td>\n",
       "      <td id=\"T_a04f2_row4_col6\" class=\"data row4 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_a04f2_row4_col7\" class=\"data row4 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_a04f2_row4_col8\" class=\"data row4 col8\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col9\" class=\"data row4 col9\" >brain - HA (hyperpallium apicale)</td>\n",
       "      <td id=\"T_a04f2_row4_col10\" class=\"data row4 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_a04f2_row4_col11\" class=\"data row4 col11\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row4_col12\" class=\"data row4 col12\" >not documented</td>\n",
       "      <td id=\"T_a04f2_row4_col13\" class=\"data row4 col13\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row4_col14\" class=\"data row4 col14\" >NA</td>\n",
       "      <td id=\"T_a04f2_row4_col15\" class=\"data row4 col15\" >White leghorn</td>\n",
       "      <td id=\"T_a04f2_row4_col16\" class=\"data row4 col16\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col17\" class=\"data row4 col17\" >9031</td>\n",
       "      <td id=\"T_a04f2_row4_col18\" class=\"data row4 col18\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col19\" class=\"data row4 col19\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col20\" class=\"data row4 col20\" >polyA</td>\n",
       "      <td id=\"T_a04f2_row4_col21\" class=\"data row4 col21\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col22\" class=\"data row4 col22\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col23\" class=\"data row4 col23\" >Gga_HA</td>\n",
       "      <td id=\"T_a04f2_row4_col24\" class=\"data row4 col24\" >SAMN07839992</td>\n",
       "      <td id=\"T_a04f2_row4_col25\" class=\"data row4 col25\" >14</td>\n",
       "      <td id=\"T_a04f2_row4_col26\" class=\"data row4 col26\" >embryonic day</td>\n",
       "      <td id=\"T_a04f2_row4_col27\" class=\"data row4 col27\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col28\" class=\"data row4 col28\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col29\" class=\"data row4 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_a04f2_row4_col30\" class=\"data row4 col30\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col31\" class=\"data row4 col31\" ></td>\n",
       "      <td id=\"T_a04f2_row4_col32\" class=\"data row4 col32\" >SAC</td>\n",
       "      <td id=\"T_a04f2_row4_col33\" class=\"data row4 col33\" >2024-10-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a04f2_level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "      <td id=\"T_a04f2_row5_col0\" class=\"data row5 col0\" >SRX3334820</td>\n",
       "      <td id=\"T_a04f2_row5_col1\" class=\"data row5 col1\" >SRP121892</td>\n",
       "      <td id=\"T_a04f2_row5_col2\" class=\"data row5 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_a04f2_row5_col3\" class=\"data row5 col3\" >SRS2637226</td>\n",
       "      <td id=\"T_a04f2_row5_col4\" class=\"data row5 col4\" >UBERON:0007349</td>\n",
       "      <td id=\"T_a04f2_row5_col5\" class=\"data row5 col5\" >mesopallium</td>\n",
       "      <td id=\"T_a04f2_row5_col6\" class=\"data row5 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_a04f2_row5_col7\" class=\"data row5 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_a04f2_row5_col8\" class=\"data row5 col8\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col9\" class=\"data row5 col9\" >brain - anterior mesopallium</td>\n",
       "      <td id=\"T_a04f2_row5_col10\" class=\"data row5 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_a04f2_row5_col11\" class=\"data row5 col11\" >missing child term</td>\n",
       "      <td id=\"T_a04f2_row5_col12\" class=\"data row5 col12\" >not documented</td>\n",
       "      <td id=\"T_a04f2_row5_col13\" class=\"data row5 col13\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row5_col14\" class=\"data row5 col14\" >NA</td>\n",
       "      <td id=\"T_a04f2_row5_col15\" class=\"data row5 col15\" >White leghorn</td>\n",
       "      <td id=\"T_a04f2_row5_col16\" class=\"data row5 col16\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col17\" class=\"data row5 col17\" >9031</td>\n",
       "      <td id=\"T_a04f2_row5_col18\" class=\"data row5 col18\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col19\" class=\"data row5 col19\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col20\" class=\"data row5 col20\" >polyA</td>\n",
       "      <td id=\"T_a04f2_row5_col21\" class=\"data row5 col21\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col22\" class=\"data row5 col22\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col23\" class=\"data row5 col23\" >Gga_mesopallium</td>\n",
       "      <td id=\"T_a04f2_row5_col24\" class=\"data row5 col24\" >SAMN07839993</td>\n",
       "      <td id=\"T_a04f2_row5_col25\" class=\"data row5 col25\" >14</td>\n",
       "      <td id=\"T_a04f2_row5_col26\" class=\"data row5 col26\" >embryonic day</td>\n",
       "      <td id=\"T_a04f2_row5_col27\" class=\"data row5 col27\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col28\" class=\"data row5 col28\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col29\" class=\"data row5 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_a04f2_row5_col30\" class=\"data row5 col30\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col31\" class=\"data row5 col31\" ></td>\n",
       "      <td id=\"T_a04f2_row5_col32\" class=\"data row5 col32\" >SAC</td>\n",
       "      <td id=\"T_a04f2_row5_col33\" class=\"data row5 col33\" >2024-10-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th id=\"T_a04f2_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "      <td id=\"T_a04f2_row6_col0\" class=\"data row6 col0\" >SRX3334819</td>\n",
       "      <td id=\"T_a04f2_row6_col1\" class=\"data row6 col1\" >SRP121892</td>\n",
       "      <td id=\"T_a04f2_row6_col2\" class=\"data row6 col2\" >Illumina HiSeq 2000</td>\n",
       "      <td id=\"T_a04f2_row6_col3\" class=\"data row6 col3\" >SRS2637225</td>\n",
       "      <td id=\"T_a04f2_row6_col4\" class=\"data row6 col4\" >UBERON:0007334</td>\n",
       "      <td id=\"T_a04f2_row6_col5\" class=\"data row6 col5\" >nidopallium</td>\n",
       "      <td id=\"T_a04f2_row6_col6\" class=\"data row6 col6\" >GgalDv:0000054</td>\n",
       "      <td id=\"T_a04f2_row6_col7\" class=\"data row6 col7\" >Hamburger Hamilton stage 40</td>\n",
       "      <td id=\"T_a04f2_row6_col8\" class=\"data row6 col8\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col9\" class=\"data row6 col9\" >brain - posterior nidopallium</td>\n",
       "      <td id=\"T_a04f2_row6_col10\" class=\"data row6 col10\" >Embryonic day 14</td>\n",
       "      <td id=\"T_a04f2_row6_col11\" class=\"data row6 col11\" >missing child term</td>\n",
       "      <td id=\"T_a04f2_row6_col12\" class=\"data row6 col12\" >not documented</td>\n",
       "      <td id=\"T_a04f2_row6_col13\" class=\"data row6 col13\" >perfect match</td>\n",
       "      <td id=\"T_a04f2_row6_col14\" class=\"data row6 col14\" >NA</td>\n",
       "      <td id=\"T_a04f2_row6_col15\" class=\"data row6 col15\" >White leghorn</td>\n",
       "      <td id=\"T_a04f2_row6_col16\" class=\"data row6 col16\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col17\" class=\"data row6 col17\" >9031</td>\n",
       "      <td id=\"T_a04f2_row6_col18\" class=\"data row6 col18\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col19\" class=\"data row6 col19\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col20\" class=\"data row6 col20\" >polyA</td>\n",
       "      <td id=\"T_a04f2_row6_col21\" class=\"data row6 col21\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col22\" class=\"data row6 col22\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col23\" class=\"data row6 col23\" >Gga_Nidopallium</td>\n",
       "      <td id=\"T_a04f2_row6_col24\" class=\"data row6 col24\" >SAMN07839994</td>\n",
       "      <td id=\"T_a04f2_row6_col25\" class=\"data row6 col25\" >14</td>\n",
       "      <td id=\"T_a04f2_row6_col26\" class=\"data row6 col26\" >embryonic day</td>\n",
       "      <td id=\"T_a04f2_row6_col27\" class=\"data row6 col27\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col28\" class=\"data row6 col28\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col29\" class=\"data row6 col29\" >PMID: 29456143, emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "      <td id=\"T_a04f2_row6_col30\" class=\"data row6 col30\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col31\" class=\"data row6 col31\" ></td>\n",
       "      <td id=\"T_a04f2_row6_col32\" class=\"data row6 col32\" >SAC</td>\n",
       "      <td id=\"T_a04f2_row6_col33\" class=\"data row6 col33\" >2024-10-14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library_file_complete = library[library_cols]\n",
    "library_file_complete.to_csv(library_to_add_path, sep=\"\\t\", index=False, quoting=csv.QUOTE_ALL)\n",
    "\n",
    "# view\n",
    "display_df(library_file_complete)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### experiment annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_9b8e2\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_9b8e2_level0_col0\" class=\"col_heading level0 col0\" >#experimentId</th>\n",
       "      <th id=\"T_9b8e2_level0_col1\" class=\"col_heading level0 col1\" >experimentName</th>\n",
       "      <th id=\"T_9b8e2_level0_col2\" class=\"col_heading level0 col2\" >experimentDescription</th>\n",
       "      <th id=\"T_9b8e2_level0_col3\" class=\"col_heading level0 col3\" >experimentSource</th>\n",
       "      <th id=\"T_9b8e2_level0_col4\" class=\"col_heading level0 col4\" >experimentStatus</th>\n",
       "      <th id=\"T_9b8e2_level0_col5\" class=\"col_heading level0 col5\" >projectTags</th>\n",
       "      <th id=\"T_9b8e2_level0_col6\" class=\"col_heading level0 col6\" >numberOfAnnotatedLibraries</th>\n",
       "      <th id=\"T_9b8e2_level0_col7\" class=\"col_heading level0 col7\" >protocol</th>\n",
       "      <th id=\"T_9b8e2_level0_col8\" class=\"col_heading level0 col8\" >protocolType</th>\n",
       "      <th id=\"T_9b8e2_level0_col9\" class=\"col_heading level0 col9\" >GSE</th>\n",
       "      <th id=\"T_9b8e2_level0_col10\" class=\"col_heading level0 col10\" >Bioproject</th>\n",
       "      <th id=\"T_9b8e2_level0_col11\" class=\"col_heading level0 col11\" >PMID</th>\n",
       "      <th id=\"T_9b8e2_level0_col12\" class=\"col_heading level0 col12\" >reference_url</th>\n",
       "      <th id=\"T_9b8e2_level0_col13\" class=\"col_heading level0 col13\" >DOI</th>\n",
       "      <th id=\"T_9b8e2_level0_col14\" class=\"col_heading level0 col14\" >xrefs</th>\n",
       "      <th id=\"T_9b8e2_level0_col15\" class=\"col_heading level0 col15\" >comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_9b8e2_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_9b8e2_row0_col0\" class=\"data row0 col0\" >SRP121892</td>\n",
       "      <td id=\"T_9b8e2_row0_col1\" class=\"data row0 col1\" >Chicken telencephalon RNAseq</td>\n",
       "      <td id=\"T_9b8e2_row0_col2\" class=\"data row0 col2\" >Embryonic chicken telencephalon nuclei were isolated for RNAseq to identify transcripts differentially expressed across different brain regions.</td>\n",
       "      <td id=\"T_9b8e2_row0_col3\" class=\"data row0 col3\" >SRA</td>\n",
       "      <td id=\"T_9b8e2_row0_col4\" class=\"data row0 col4\" ></td>\n",
       "      <td id=\"T_9b8e2_row0_col5\" class=\"data row0 col5\" ></td>\n",
       "      <td id=\"T_9b8e2_row0_col6\" class=\"data row0 col6\" ></td>\n",
       "      <td id=\"T_9b8e2_row0_col7\" class=\"data row0 col7\" ></td>\n",
       "      <td id=\"T_9b8e2_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_9b8e2_row0_col9\" class=\"data row0 col9\" ></td>\n",
       "      <td id=\"T_9b8e2_row0_col10\" class=\"data row0 col10\" >PRJNA416004</td>\n",
       "      <td id=\"T_9b8e2_row0_col11\" class=\"data row0 col11\" ></td>\n",
       "      <td id=\"T_9b8e2_row0_col12\" class=\"data row0 col12\" ></td>\n",
       "      <td id=\"T_9b8e2_row0_col13\" class=\"data row0 col13\" >E,r,r,o,r,:, ,U,n,a,b,l,e, ,t,o, ,r,e,t,r,i,e,v,e, ,d,a,t,a,,, ,S,t,a,t,u,s, ,c,o,d,e, ,4,0,4</td>\n",
       "      <td id=\"T_9b8e2_row0_col14\" class=\"data row0 col14\" ></td>\n",
       "      <td id=\"T_9b8e2_row0_col15\" class=\"data row0 col15\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "experiment = pd.read_csv(experiment_path_from_script, sep='\\t', index_col=False, keep_default_na=False, na_values=['NULL','null', 'nan','NaN'], dtype=object)\n",
    "display_df(experiment)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### experiment and protocol details"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# this will give you the number of rows in the complete library file \n",
    "# this should be the number of annotated libraries\n",
    "ann_lib = len(library_file_complete.index)\n",
    "len(library_file_complete.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_3dc43\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_3dc43_level0_col0\" class=\"col_heading level0 col0\" >#experimentId</th>\n",
       "      <th id=\"T_3dc43_level0_col1\" class=\"col_heading level0 col1\" >experimentName</th>\n",
       "      <th id=\"T_3dc43_level0_col2\" class=\"col_heading level0 col2\" >experimentDescription</th>\n",
       "      <th id=\"T_3dc43_level0_col3\" class=\"col_heading level0 col3\" >experimentSource</th>\n",
       "      <th id=\"T_3dc43_level0_col4\" class=\"col_heading level0 col4\" >experimentStatus</th>\n",
       "      <th id=\"T_3dc43_level0_col5\" class=\"col_heading level0 col5\" >projectTags</th>\n",
       "      <th id=\"T_3dc43_level0_col6\" class=\"col_heading level0 col6\" >numberOfAnnotatedLibraries</th>\n",
       "      <th id=\"T_3dc43_level0_col7\" class=\"col_heading level0 col7\" >protocol</th>\n",
       "      <th id=\"T_3dc43_level0_col8\" class=\"col_heading level0 col8\" >protocolType</th>\n",
       "      <th id=\"T_3dc43_level0_col9\" class=\"col_heading level0 col9\" >GSE</th>\n",
       "      <th id=\"T_3dc43_level0_col10\" class=\"col_heading level0 col10\" >Bioproject</th>\n",
       "      <th id=\"T_3dc43_level0_col11\" class=\"col_heading level0 col11\" >PMID</th>\n",
       "      <th id=\"T_3dc43_level0_col12\" class=\"col_heading level0 col12\" >reference_url</th>\n",
       "      <th id=\"T_3dc43_level0_col13\" class=\"col_heading level0 col13\" >DOI</th>\n",
       "      <th id=\"T_3dc43_level0_col14\" class=\"col_heading level0 col14\" >xrefs</th>\n",
       "      <th id=\"T_3dc43_level0_col15\" class=\"col_heading level0 col15\" >comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_3dc43_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_3dc43_row0_col0\" class=\"data row0 col0\" >SRP121892</td>\n",
       "      <td id=\"T_3dc43_row0_col1\" class=\"data row0 col1\" >Chicken telencephalon RNAseq</td>\n",
       "      <td id=\"T_3dc43_row0_col2\" class=\"data row0 col2\" >Embryonic chicken telencephalon nuclei were isolated for RNAseq to identify transcripts differentially expressed across different brain regions.</td>\n",
       "      <td id=\"T_3dc43_row0_col3\" class=\"data row0 col3\" >SRA</td>\n",
       "      <td id=\"T_3dc43_row0_col4\" class=\"data row0 col4\" >total</td>\n",
       "      <td id=\"T_3dc43_row0_col5\" class=\"data row0 col5\" ></td>\n",
       "      <td id=\"T_3dc43_row0_col6\" class=\"data row0 col6\" >7</td>\n",
       "      <td id=\"T_3dc43_row0_col7\" class=\"data row0 col7\" ></td>\n",
       "      <td id=\"T_3dc43_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_3dc43_row0_col9\" class=\"data row0 col9\" ></td>\n",
       "      <td id=\"T_3dc43_row0_col10\" class=\"data row0 col10\" >PRJNA416004</td>\n",
       "      <td id=\"T_3dc43_row0_col11\" class=\"data row0 col11\" ></td>\n",
       "      <td id=\"T_3dc43_row0_col12\" class=\"data row0 col12\" ></td>\n",
       "      <td id=\"T_3dc43_row0_col13\" class=\"data row0 col13\" >E,r,r,o,r,:, ,U,n,a,b,l,e, ,t,o, ,r,e,t,r,i,e,v,e, ,d,a,t,a,,, ,S,t,a,t,u,s, ,c,o,d,e, ,4,0,4</td>\n",
       "      <td id=\"T_3dc43_row0_col14\" class=\"data row0 col14\" ></td>\n",
       "      <td id=\"T_3dc43_row0_col15\" class=\"data row0 col15\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# partial or total\n",
    "experiment.loc[:,'experimentStatus'] = 'total'\n",
    "#experiment.loc[:,'projectTags'] = '' \n",
    "# see above cell, also can add as free text\n",
    "experiment.loc[:,'numberOfAnnotatedLibraries'] = ann_lib\n",
    "\n",
    "# these variables should already exist from above but if not can just add as free text\n",
    "#experiment.loc[:,'protocol'] = my_protocol\n",
    "#experiment.loc[:,'protocolType'] = my_protocolType\n",
    "\n",
    "display_df(experiment)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### paper and xrefs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_29a84\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_29a84_level0_col0\" class=\"col_heading level0 col0\" >#experimentId</th>\n",
       "      <th id=\"T_29a84_level0_col1\" class=\"col_heading level0 col1\" >experimentName</th>\n",
       "      <th id=\"T_29a84_level0_col2\" class=\"col_heading level0 col2\" >experimentDescription</th>\n",
       "      <th id=\"T_29a84_level0_col3\" class=\"col_heading level0 col3\" >experimentSource</th>\n",
       "      <th id=\"T_29a84_level0_col4\" class=\"col_heading level0 col4\" >experimentStatus</th>\n",
       "      <th id=\"T_29a84_level0_col5\" class=\"col_heading level0 col5\" >projectTags</th>\n",
       "      <th id=\"T_29a84_level0_col6\" class=\"col_heading level0 col6\" >numberOfAnnotatedLibraries</th>\n",
       "      <th id=\"T_29a84_level0_col7\" class=\"col_heading level0 col7\" >protocol</th>\n",
       "      <th id=\"T_29a84_level0_col8\" class=\"col_heading level0 col8\" >protocolType</th>\n",
       "      <th id=\"T_29a84_level0_col9\" class=\"col_heading level0 col9\" >GSE</th>\n",
       "      <th id=\"T_29a84_level0_col10\" class=\"col_heading level0 col10\" >Bioproject</th>\n",
       "      <th id=\"T_29a84_level0_col11\" class=\"col_heading level0 col11\" >PMID</th>\n",
       "      <th id=\"T_29a84_level0_col12\" class=\"col_heading level0 col12\" >reference_url</th>\n",
       "      <th id=\"T_29a84_level0_col13\" class=\"col_heading level0 col13\" >DOI</th>\n",
       "      <th id=\"T_29a84_level0_col14\" class=\"col_heading level0 col14\" >xrefs</th>\n",
       "      <th id=\"T_29a84_level0_col15\" class=\"col_heading level0 col15\" >comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_29a84_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_29a84_row0_col0\" class=\"data row0 col0\" >SRP121892</td>\n",
       "      <td id=\"T_29a84_row0_col1\" class=\"data row0 col1\" >Chicken telencephalon RNAseq</td>\n",
       "      <td id=\"T_29a84_row0_col2\" class=\"data row0 col2\" >Embryonic chicken telencephalon nuclei were isolated for RNAseq to identify transcripts differentially expressed across different brain regions.</td>\n",
       "      <td id=\"T_29a84_row0_col3\" class=\"data row0 col3\" >SRA</td>\n",
       "      <td id=\"T_29a84_row0_col4\" class=\"data row0 col4\" >total</td>\n",
       "      <td id=\"T_29a84_row0_col5\" class=\"data row0 col5\" ></td>\n",
       "      <td id=\"T_29a84_row0_col6\" class=\"data row0 col6\" >7</td>\n",
       "      <td id=\"T_29a84_row0_col7\" class=\"data row0 col7\" ></td>\n",
       "      <td id=\"T_29a84_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_29a84_row0_col9\" class=\"data row0 col9\" ></td>\n",
       "      <td id=\"T_29a84_row0_col10\" class=\"data row0 col10\" >PRJNA416004</td>\n",
       "      <td id=\"T_29a84_row0_col11\" class=\"data row0 col11\" >29456143</td>\n",
       "      <td id=\"T_29a84_row0_col12\" class=\"data row0 col12\" >https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11098552/</td>\n",
       "      <td id=\"T_29a84_row0_col13\" class=\"data row0 col13\" >10.1016/j.cub.2018.01.036</td>\n",
       "      <td id=\"T_29a84_row0_col14\" class=\"data row0 col14\" ></td>\n",
       "      <td id=\"T_29a84_row0_col15\" class=\"data row0 col15\" ></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#experiment.loc[:,'GSE'] = ''\n",
    "#experiment.loc[:,'Bioproject'] = '' \n",
    "experiment.loc[:,'PMID'] = '29456143'\n",
    "experiment.loc[:,'reference_url'] = 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11098552/'\n",
    "experiment.loc[:,'DOI'] = '10.1016/j.cub.2018.01.036'\n",
    "#experiment.loc[:,'xrefs'] = ''\n",
    "\n",
    "display_df(experiment)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### comments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style='height: 300px; overflow: auto; width: fit-content'><style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_8a152\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th class=\"blank level0\" >&nbsp;</th>\n",
       "      <th id=\"T_8a152_level0_col0\" class=\"col_heading level0 col0\" >#experimentId</th>\n",
       "      <th id=\"T_8a152_level0_col1\" class=\"col_heading level0 col1\" >experimentName</th>\n",
       "      <th id=\"T_8a152_level0_col2\" class=\"col_heading level0 col2\" >experimentDescription</th>\n",
       "      <th id=\"T_8a152_level0_col3\" class=\"col_heading level0 col3\" >experimentSource</th>\n",
       "      <th id=\"T_8a152_level0_col4\" class=\"col_heading level0 col4\" >experimentStatus</th>\n",
       "      <th id=\"T_8a152_level0_col5\" class=\"col_heading level0 col5\" >projectTags</th>\n",
       "      <th id=\"T_8a152_level0_col6\" class=\"col_heading level0 col6\" >numberOfAnnotatedLibraries</th>\n",
       "      <th id=\"T_8a152_level0_col7\" class=\"col_heading level0 col7\" >protocol</th>\n",
       "      <th id=\"T_8a152_level0_col8\" class=\"col_heading level0 col8\" >protocolType</th>\n",
       "      <th id=\"T_8a152_level0_col9\" class=\"col_heading level0 col9\" >GSE</th>\n",
       "      <th id=\"T_8a152_level0_col10\" class=\"col_heading level0 col10\" >Bioproject</th>\n",
       "      <th id=\"T_8a152_level0_col11\" class=\"col_heading level0 col11\" >PMID</th>\n",
       "      <th id=\"T_8a152_level0_col12\" class=\"col_heading level0 col12\" >reference_url</th>\n",
       "      <th id=\"T_8a152_level0_col13\" class=\"col_heading level0 col13\" >DOI</th>\n",
       "      <th id=\"T_8a152_level0_col14\" class=\"col_heading level0 col14\" >xrefs</th>\n",
       "      <th id=\"T_8a152_level0_col15\" class=\"col_heading level0 col15\" >comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th id=\"T_8a152_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "      <td id=\"T_8a152_row0_col0\" class=\"data row0 col0\" >SRP121892</td>\n",
       "      <td id=\"T_8a152_row0_col1\" class=\"data row0 col1\" >Chicken telencephalon RNAseq</td>\n",
       "      <td id=\"T_8a152_row0_col2\" class=\"data row0 col2\" >Embryonic chicken telencephalon nuclei were isolated for RNAseq to identify transcripts differentially expressed across different brain regions.</td>\n",
       "      <td id=\"T_8a152_row0_col3\" class=\"data row0 col3\" >SRA</td>\n",
       "      <td id=\"T_8a152_row0_col4\" class=\"data row0 col4\" >total</td>\n",
       "      <td id=\"T_8a152_row0_col5\" class=\"data row0 col5\" ></td>\n",
       "      <td id=\"T_8a152_row0_col6\" class=\"data row0 col6\" >7</td>\n",
       "      <td id=\"T_8a152_row0_col7\" class=\"data row0 col7\" ></td>\n",
       "      <td id=\"T_8a152_row0_col8\" class=\"data row0 col8\" ></td>\n",
       "      <td id=\"T_8a152_row0_col9\" class=\"data row0 col9\" ></td>\n",
       "      <td id=\"T_8a152_row0_col10\" class=\"data row0 col10\" >PRJNA416004</td>\n",
       "      <td id=\"T_8a152_row0_col11\" class=\"data row0 col11\" >29456143</td>\n",
       "      <td id=\"T_8a152_row0_col12\" class=\"data row0 col12\" >https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11098552/</td>\n",
       "      <td id=\"T_8a152_row0_col13\" class=\"data row0 col13\" >10.1016/j.cub.2018.01.036</td>\n",
       "      <td id=\"T_8a152_row0_col14\" class=\"data row0 col14\" ></td>\n",
       "      <td id=\"T_8a152_row0_col15\" class=\"data row0 col15\" >emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "experiment.loc[:,'comment'] = 'emailed stevendb@uchicago.edu to clarify developmental stage - all E14, SRA is incorrect'\n",
    "\n",
    "display_df(experiment)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### save complete file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "experiment.to_csv(experiment_to_add_path, sep=\"\\t\", index=False, quoting=csv.QUOTE_ALL)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### added to git separately\n",
    "used this annotation as a demo example with anne, QA and add to git was done together and not documented"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "scRNA",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
